<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>AlphaFold</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="theme3.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Home</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-microscope"></span>
     
    Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="orthop-genomes-project.html">Comparative Genomics of Schistocerca gregaria</a>
    </li>
    <li class="dropdown-header">Comparative Genomics of Mormon Cricket subgroups -- UNDER CONSTRUCTION</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-code"></span>
     
    Program Information
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">TOGA &amp; Related Pipelines</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="toga.html">TOGA</a>
        </li>
        <li>
          <a href="mlc.html">make_lastz_chains</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="genespace.html">GENESPACE</a>
    </li>
    <li>
      <a href="barkdull.html">FormicidaeMolecularEvolution (OrthoFinder)</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Annotation Tools</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="Apollo.html">Apollo</a>
        </li>
        <li>
          <a href="augustus.html">Augustus</a>
        </li>
        <li>
          <a href="braker.html">Braker</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="rep-model-mask.html">RepeatModeler &amp; RepeatMasker</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Proteomics</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="alphafold.html">AlphaFold</a>
        </li>
        <li>
          <a href="chimerax.html">Chimera X</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-book-open-reader"></span>
     
    Reference
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="slurm-comm.html">Linux and Slurm Commands</a>
    </li>
    <li>
      <a href="definitions.html">Glossary of Terminology</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">AlphaFold</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-07-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>song-lab/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomecbaker7tamusonglabtreebf7765b893698fba805f5c3263c00881f04c8413targetblankbf7765ba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/ecbaker7-tamu/song-lab/tree/bf7765b893698fba805f5c3263c00881f04c8413" target="_blank">bf7765b</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomecbaker7tamusonglabtreebf7765b893698fba805f5c3263c00881f04c8413targetblankbf7765ba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/ecbaker7-tamu/song-lab/tree/bf7765b893698fba805f5c3263c00881f04c8413" target="_blank">bf7765b</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store

Unstaged changes:
    Modified:   analysis/Publish all files
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/alphafold.Rmd</code>) and HTML
(<code>docs/alphafold.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/7a5615778c5c2cf842b2ae1b8c5969112f4bef73/docs/alphafold.html" target="_blank">7a56157</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-05-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/fecf69695ef32ed4f732693ce7fb0749bedc3a3e/docs/alphafold.html" target="_blank">fecf696</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-05-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/9ed28350316f5760a28dc7c92480d718b24a7c5b/docs/alphafold.html" target="_blank">9ed2835</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-05-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/05539da9915300f741c6fda620038213ded63e5b/docs/alphafold.html" target="_blank">05539da</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-05-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/638b4d2eb0c93f48a1f1679f39b19dd83557ada3/analysis/alphafold.Rmd" target="_blank">638b4d2</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-05-20
</td>
<td>
wflow_publish(c("analysis/alphafold.Rmd", "analysis/Apollo.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/ad2ac0606d6ada9812f82d214d88171a2d69673a/docs/alphafold.html" target="_blank">ad2ac06</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-04-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/a302b2806831239bbdea852edd453c41d6b89408/analysis/alphafold.Rmd" target="_blank">a302b28</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-04-15
</td>
<td>
wflow_publish(c("analysis/alphafold.Rmd", "analysis/Apollo.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/729de9688ab12a7636a75643ec35e40f91cadddd/docs/alphafold.html" target="_blank">729de96</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-04-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/b2b35b2d503db89b116e8155fd40983daf557fbf/analysis/alphafold.Rmd" target="_blank">b2b35b2</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-04-11
</td>
<td>
wflow_publish(c("analysis/alphafold.Rmd", "analysis/Apollo.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="in-silico-protein-structure-prediction"
class="section level2 tabset">
<h2 class="tabset">In-silico Protein Structure Prediction</h2>
<div id="what-is-alphafold" class="section level3">
<h3>What is AlphaFold?</h3>
<p>AlphaFold is a software created by Google DeepMind that predicts
protein structure based solely off of the primary structure of the
protein (amino acid sequence). Additionally, you may predict multi-mer
structures and proteins with the addition of certain ligands available
in the AlphaFold data set.</p>
<p>Currently, the third release of the program, aptly named AlphaFold3,
is the latest edition. You may view the paper, published in the journal
<em>Nature</em>, <a
href="https://www.nature.com/articles/s41586-024-07487-w">here</a>.</p>
<div id="limitations" class="section level4">
<h4>Limitations</h4>
<p>As of April 2025, the daily limit for the number of jobs per account
is 30. If you require a higher threshhold of jobs, you will have to
obtain a contract with DeepMind for this level of access. However, 30
jobs is plenty for a large portion of researchers.</p>
<p>One of AlphaFold’s biggest limitations is the software’s
unreliability in predicting highly dynamic regions of a protein.
Commonly, when you put in a protein with large areas of dynamic
activity, AlphaFold will tend to force the structure into an alpha-helix
shape. I tend to see it in my analysis when trying to model a protein
with some sort of anchoring mechanism, as these regions are inherently
dynamic.</p>
<p>This is of less concern when not working with proteins with majority
highly dynamic structures, but it is certainly a property to be aware
of.</p>
</div>
</div>
<div id="how-to-use" class="section level3">
<h3>How to Use</h3>
The main function of AlphaFold is the prediction of protein structures.
To perform this prediction, simply visit the website <a
href="alphafoldserver.com">alphafoldserver.com</a> and copy-paste your
protein’s sequence into the interface provided. Below is an image of
this interface when you first enter the website:
<center>
<img src="assets/alphafold-gui-empty.png" width="600" />
</center>
and once you enter your protein sequence is should look like this:
<center>
<img src="assets/alphafold-GUI.png" width="600" />
</center>
<p>Above the space where you enter your protein sequence, you will see 2
boxes offering specifications of your AlphaFold run.</p>
<p>The topmost box is labeled “Entity type”. This allows you to select
from 5 options: Protein, DNA, RNA, Ligand, or Ion. Select whichever best
describes the structure you’re trying to predict.</p>
<p>The middle box is labeled “Copies”. This allows you to produce
multiple copies of your entity for the generation of structures such as
dimers and multimers.</p>
<p>You may also add multiple proteins to one run by adding multiple
sequences using the “Add entity” option below your previously-entered
sequence.</p>
Once you’ve specified all necessary steps for your run, you will hit
either “Save job” if you would like to run it at a later date, or
“Continue and preview job” if you are ready to run the job. Once you
select continue and preview, a pop-up will appear:
<center>
<img src="assets/alphafold-GUI-submit.png" width="600" />
</center>
<p>You should change your job name to something meaningful, as AlphaFold
will automatically name it using the date and time of the run
submission. Additionally, since AlphaFold is a predictive model, it may
generate differing results between runs. If you wish to be able to
reproduce identical results to your run, add a seed to your run. This
will allow this run to be replicable. Once you’re ready, hit “Confirm
and submit job.”</p>
<p>This run will produce 2 figures: a 3-D prediction color-coded by
pIDTT scores and a pAE graph. Interpretation of these 2 graphs are
discussed under the <em>Analysis of Predictions</em> tab of this
page.</p>
<div id="downloading-results" class="section level4">
<h4>Downloading Results</h4>
<p>By hitting the download button above your AlphaFold run, your browser
will download a .zip file. this file contains 5 .json files for data, 5
.cif files, 5 .json files for summary of confidences, a .json for the
job request, and one file for the terms of use. For each run, AlphaFold
generates 5 predictions that show 5 different configurations of your
protein. Additionally, the labeling of the models follows zero-based
indexing, so the first model is labeled YourRunName_model_0.</p>
</div>
</div>
<div id="analysis-of-predictions" class="section level3">
<h3>Analysis of Predictions</h3>
<div id="plddt" class="section level4">
<h4>pLDDT</h4>
<p>The main figure that AlphaFold produces is a generated image of the
predicted structure, along with color-coordinated scores of confidence
called predicted local distance difference test (pLDDT) scores. Dark
blue indicated a very high pLDDT score, light blue a moderately high
score, yellow is low, and orange is extremely low.</p>
<p>The pLDDT score gives an estimate of how well AlphaFold’s predicted
strutcure would agree with an experimentally-derived structure. In the
example below, you can see region of high certainty within the center of
the protein, but lower certainty along the alpha-heix that is along the
back part of the loop extended out from the main structure.</p>
<center>
<p><img src="assets/alphafold-ex-pred.png" width="600" /></p>
<p><img src="assets/alphafold-key.png" width="600" /></p>
</center>
</div>
<div id="pae-plots" class="section level4">
<h4>pAE Plots</h4>
<p>The predicted aligned error (pAE) plots produced by AlphaFold
indicate the model’s confidence that of the relative position of 2
residues within the structure. Along the x- and y-axes we have the
nucleotide bases of the protein from start to end. The intersection of 2
points indicates AlphaFold’s certainty of those 2 regions positions
relative to each other.</p>
This plot is important because there are many groups extending off of
the main structure of the protein, and these groups can interact and
cause the protein to change shape.
<center>
<img src="assets/alphafold-ex-pAE.png" width="500" />
</center>
<p>Commonly, regions that are further from each other have lower scores
of certainty, and regions that are close together score highly. Notice
the dark line going from the top left corner of the graph to the bottom
right. This is the relative position of each bases’ position relative to
itself (which is always certain), so this dark green line will
<em>always</em> be in the pAE plot.</p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
