<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Augustus</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="theme3.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Home</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-microscope"></span>
     
    Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="orthop-genomes-project.html">Comparative Genomics of Schistocerca gregaria</a>
    </li>
    <li class="dropdown-header">Comparative Genomics of Mormon Cricket subgroups -- UNDER CONSTRUCTION</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-code"></span>
     
    Workflows
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="barkdull.html">FormicidaeMolecularEvolution - OrthoFinder Workflow for Longest Isoform</a>
    </li>
    <li>
      <a href="rep-model-mask.html">RepeatModeler &amp; RepeatMasker</a>
    </li>
    <li>
      <a href="psmc.html">PSMC Model Analysis</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Proteomics</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="alphafold.html">AlphaFold</a>
        </li>
        <li>
          <a href="chimerax.html">Chimera X</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-palette"></span>
     
    Visualization
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="kimura.html">Kimura Plots (Repeat Content)</a>
    </li>
    <li>
      <a href="genespace.html">GENESPACE (Synteny)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-book-open-reader"></span>
     
    Reference
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="slurm-comm.html">Linux and Slurm Commands</a>
    </li>
    <li>
      <a href="definitions.html">Glossary of Terminology</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Augustus</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-07-11
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>song-lab/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240605code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240605)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240605code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240605)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomecbaker7tamusonglabtreec9a13b52d09f5b1ebb5643266c1580673a143a42targetblankc9a13b5a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/ecbaker7-tamu/song-lab/tree/c9a13b52d09f5b1ebb5643266c1580673a143a42" target="_blank">c9a13b5</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomecbaker7tamusonglabtreec9a13b52d09f5b1ebb5643266c1580673a143a42targetblankc9a13b5a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/ecbaker7-tamu/song-lab/tree/c9a13b52d09f5b1ebb5643266c1580673a143a42" target="_blank">c9a13b5</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/augustus.Rmd</code>) and HTML
(<code>docs/augustus.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/a1b980b1cdd4fe50a1a3d6b10949e589e8f1b825/docs/augustus.html" target="_blank">a1b980b</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-07-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/4e146c4576f9c498f75eb81aa9193b039c9d4fcc/docs/augustus.html" target="_blank">4e146c4</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-07-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/7a5615778c5c2cf842b2ae1b8c5969112f4bef73/docs/augustus.html" target="_blank">7a56157</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-05-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/fecf69695ef32ed4f732693ce7fb0749bedc3a3e/docs/augustus.html" target="_blank">fecf696</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-05-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/9ed28350316f5760a28dc7c92480d718b24a7c5b/docs/augustus.html" target="_blank">9ed2835</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-05-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/05539da9915300f741c6fda620038213ded63e5b/docs/augustus.html" target="_blank">05539da</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-05-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/ad2ac0606d6ada9812f82d214d88171a2d69673a/docs/augustus.html" target="_blank">ad2ac06</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-04-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/729de9688ab12a7636a75643ec35e40f91cadddd/docs/augustus.html" target="_blank">729de96</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-04-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/63d1ad06520e7f288ebaa6562d63fe2865527bf7/docs/augustus.html" target="_blank">63d1ad0</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/aaa78c46da9d439cbb501e9591ecc971e6283817/docs/augustus.html" target="_blank">aaa78c4</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-01-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/a9233f82b9e3a659c6612c7f6f8279fb2e79e21d/docs/augustus.html" target="_blank">a9233f8</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/b96dd10e745964b9491094cb342a421b6affab80/docs/augustus.html" target="_blank">b96dd10</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-10-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/3731e9e5fc895d0bb1d501e825e40d8f92ec9e6d/docs/augustus.html" target="_blank">3731e9e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/0c3ed6b76968749b6ad01b7a66f347fc1d7b8be3/docs/augustus.html" target="_blank">0c3ed6b</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/79140076f6780bbbd8c216112c863ca77cdafb1a/docs/augustus.html" target="_blank">7914007</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-08-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/98122725125fb3eac2898505b1869c1b94c26204/docs/augustus.html" target="_blank">9812272</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/f2bc75a7088b701660c636f42bc845588b644ad2/docs/augustus.html" target="_blank">f2bc75a</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/c93686e4e53271171e03b8c09ccd48906b416205/docs/augustus.html" target="_blank">c93686e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/9a95e4e4daf346e551af00fb5642b117cba37e1e/analysis/augustus.Rmd" target="_blank">9a95e4e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-15
</td>
<td>
wflow_publish(c("analysis/cafe5.Rmd", "analysis/augustus.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/783cca38f0df082611321d7be930f88510b823c9/docs/augustus.html" target="_blank">783cca3</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/a60be8ed2b9562f92be143c0e35927110463bfcf/analysis/augustus.Rmd" target="_blank">a60be8e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-03
</td>
<td>
wflow_publish(c("analysis/barkdull.Rmd", "analysis/compare-genomes.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="gene-prediction" class="section level2 tabset">
<h2 class="tabset">Gene Prediction</h2>
<div id="what-is-augustus" class="section level3">
<h3>What is Augustus?</h3>
<p>From the <a
href="https://github.com/Gaius-Augustus/Augustus/blob/master/docs/ABOUT.md">Augustus
GitHub introduction page</a>:</p>
<p>AUGUSTUS is a gene prediction program written or maintained by Mario
Stanke, Oliver Keller, Stefanie König, Lizzy Gerischer, Katharina Hoff,
Giovanna Migliorelli, Lars Gabriel, Anica Hoppe, Tonatiuh Peña Centeno,
Henry Mehlan, Daniel Honsel and Steffen Herbold. It can be used as an ab
initio program, which means it bases its prediction purely on the
sequence. AUGUSTUS may also incorporate hints on the gene structure
coming from extrinsic sources such as EST, MS/MS, protein alignments and
syntenic genomic alignments. Since version 3.0 AUGUSTUS can also predict
the genes simultaneously in several aligned genomes (see README-cgp.md).
AUGUSTUS has been trained for predicting genes, among other species,
in:</p>
<style>
.two-columns {
  column-count: 2; /* Number of columns */
  column-gap: 20px; /* Space between columns */
}

.two-columns p, .two-columns ul {
  margin: 0;
  padding-left: 20px; /* Ensure consistent indentation */
}

ul {
  padding-left: 20px; /* Adjust as needed */
}

ul li {
  margin-bottom: 5px; /* Adjust spacing between items */
}
</style>
<div class="two-columns">
<p>
<ul>
<li>Homo sapiens (human),
<ul>
<li>Drosophila melanogaster (fruit fly),</li>
<li>Arabidopsis thaliana (plant),</li>
<li>Brugia malayi (nematode),</li>
<li>Aedes aegypti (mosquito),</li>
<li>Coprinus cinereus (fungus),</li>
<li>Tribolium castaneum (beetle)</li>
<li>Schistosoma mansoni (worm)</li>
<li>Tetrahymena thermophila (ciliate)</li>
<li>Galdieria sulphuraria (red algae)</li>
<li>Zea mays (maize)</li>
<li>Toxoplasma gondii (parasitic protozoa)</li>
<li>Caenorhabditis elegans (worm)</li>
<li>Aspergillus fumigatus</li>
<li>Aspergillus nidulans</li>
<li>Aspergillus oryzae</li>
<li>Aspergillus terreus</li>
<li>Botrytis cinerea</li>
<li>Callorhinchus milii</li>
<li>Candida albicans</li>
<li>Candida guilliermondii</li>
<li>Candida tropicalis</li>
<li>Chaetomium globosum</li>
<li>Coccidioides immitis</li>
<li>Cryptococcus neoformans gattii</li>
<li>Cryptococcus neoformans neoformans</li>
<li>Danio rerio</li>
<li>Debaryomyces hansenii</li>
<li>Encephalitozoon cuniculi</li>
<li>Eremothecium gossypii</li>
<li>Fusarium graminearum</li>
<li>Gallus gallus</li>
<li>Histoplasma capsulatum</li>
<li>Kluyveromyces lactis</li>
<li>Laccaria bicolor</li>
<li>Lodderomyces elongisporus</li>
<li>Magnaporthe grisea</li>
<li>Neurospora crassa</li>
<li>Nicotiana attenuata (coyote tobacco)</li>
<li>Petromyzon marinus (sea lamprey)</li>
<li>Phanerochaete chrysosporium</li>
<li>Pichia stipitis</li>
<li>Rhizopus oryzae</li>
<li>Saccharomyces cerevisiae</li>
<li>Schizosaccharomyces pombe</li>
<li>Staphylococcus aureus</li>
<li>Ustilago maydis</li>
<li>Yarrowia lipolytica</li>
<li>Nasonia vitripennis (wasp)</li>
<li>Solanum lycopersicum (tomato)</li>
<li>Chlamydomonas reinhardtii (green algae)</li>
<li>Amphimedon queenslandica (sponge)</li>
<li>Acyrthosiphon pisum (pea aphid)</li>
<li>Leishmania tarentolae (protozoa, intronless)</li>
<li>Trichinella spiralis</li>
<li>Theobroma cacao (cacao)</li>
<li>Escherichia coli</li>
<li>Thermoanaerobacter tengcongensis (a bacterium)</li>
<li>Triticum aestivum (wheat)</li>
<li>Ancylostoma ceylanicum</li>
<li>Volvox carteri</li>
<li>Mnemiopsis leidyi (Ctenophora)</li>
<li>Nematostella vectensis (Cnidaria)</li>
<li>Ciona intestinalis (Chordata)</li>
<li>Strongylocentrotus purpuratus (Echinodermata)</li>
<li>Pisaster ochraceus (starfish)</li>
<li>Chiloscyllium punctatum (bamboo shark)</li>
<li>Scyliorhinus torazame (cat shark)</li>
<li>Rhincodon typus (whale shark)</li>
</ul>
</p>
</div></li>
</ul>
<div id="installation-running" class="section level3">
<h3>Installation &amp; Running</h3>
<ol style="list-style-type: decimal">
<li>Load Augustus</li>
</ol>
<p>The Augustus module is available on the TAMU Grace cluster, you can
load it using the command</p>
<pre class="r"><code> ml GCC/10.2.0  OpenMPI/4.0.5 GCC/11.2.0  OpenMPI/4.1.1 AUGUSTUS/3.4.0</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>After loading Augustus, you are ready to run the program. Augustus
only requires 2 input files, the fasta file of the query species and the
species you want to reference from the AUGUSTUS database (see the list
<a
href="https://github.com/Gaius-Augustus/Augustus/blob/master/docs/RUNNING-AUGUSTUS.md">here</a>).
The command is as follows:</li>
</ol>
<pre class="r"><code>augustus [parameters] --species=SPECIES queryfilename</code></pre>
<p>Below is a table of optional flags you can add to the above command
to customize your Augustus run.</p>
<table class="table table-striped table-hover table-condensed table-responsive" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
</caption>
<thead>
<tr>
<th style="empty-cells: hide;border-bottom:hidden;" colspan="1">
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: right; " colspan="1">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Augustus Commands
</div>
</th>
<th style="empty-cells: hide;border-bottom:hidden;" colspan="1">
</th>
</tr>
<tr>
<th style="text-align:left;font-weight: bold;background-color: rgba(191, 162, 176, 255) !important;">
Command
</th>
<th style="text-align:left;font-weight: bold;background-color: rgba(191, 162, 176, 255) !important;">
Options
</th>
<th style="text-align:left;font-weight: bold;background-color: rgba(191, 162, 176, 255) !important;">
Function
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;width: 20em; ">
–strand
</td>
<td style="text-align:left;width: 30em; ">
both, forward, backward
</td>
<td style="text-align:left;width: 50em; ">
Report predicted genes on both strands, just the forward or just the
backward. Default is both
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–genemodel
</td>
<td style="text-align:left;width: 30em; ">
partial, intronless, or complete
</td>
<td style="text-align:left;width: 50em; ">
Allow prediction of incomplete genes at the sequence boundaries
(default) intronless : only predict single-exon genes like in
prokaryotes and some eukaryotes complete : only predict complete genes
atleastone : predict at least one complete gene exactlyone : predict
exactly one complete gene
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–singlestrand
</td>
<td style="text-align:left;width: 30em; ">
true
</td>
<td style="text-align:left;width: 50em; ">
Predict genes independently on each strand, allow overlapping genes on
opposite strands This option is turned off by default.
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–hintsfile
</td>
<td style="text-align:left;width: 30em; ">
hintsfilename
</td>
<td style="text-align:left;width: 50em; ">
When this option is used the prediction considering hints (extrinsic
information) is turned on. hintsfilename contains the hints in gff
format.
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–extrinsicCfgFile
</td>
<td style="text-align:left;width: 30em; ">
cfg-file-name
</td>
<td style="text-align:left;width: 50em; ">
Optional. This file contains the list of used sources for the hints and
their boni and mali. If not specified the file ‘extrinsic.cfg’ in the
config directory $AUGUSTUS_CONFIG_PATH is used.
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–maxDNAPieceSize
</td>
<td style="text-align:left;width: 30em; ">
Number
</td>
<td style="text-align:left;width: 50em; ">
This value specifies the maximal length of the pieces that the sequence
is cut into for the core algorithm (Viterbi) to be run. Default is
–maxDNAPieceSize=200000. AUGUSTUS tries to place the boundaries of these
pieces in the intergenic region, which is inferred by a preliminary
prediction. GC-content dependent parameters are chosen for each piece of
DNA if /Constant/decomp_num_steps &gt; 1 for that species. This is why
this value should not be set very large, even if you have plenty of
memory.
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–protein
</td>
<td style="text-align:left;width: 30em; ">
on, off
</td>
<td style="text-align:left;width: 50em; ">
Includes protein fasta file
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–codingseq
</td>
<td style="text-align:left;width: 30em; ">
on, off
</td>
<td style="text-align:left;width: 50em; ">
Include coding sequences
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–introns
</td>
<td style="text-align:left;width: 30em; ">
on, off
</td>
<td style="text-align:left;width: 50em; ">
Include introns
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–start
</td>
<td style="text-align:left;width: 30em; ">
on, off
</td>
<td style="text-align:left;width: 50em; ">
Look for start codons
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–stop
</td>
<td style="text-align:left;width: 30em; ">
on, off
</td>
<td style="text-align:left;width: 50em; ">
Look for stop codons
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–cds
</td>
<td style="text-align:left;width: 30em; ">
on, off
</td>
<td style="text-align:left;width: 50em; ">
Looks for coding sequences
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–exonnames
</td>
<td style="text-align:left;width: 30em; ">
on, off
</td>
<td style="text-align:left;width: 50em; ">
Output options. Output predicted amino acid sequences or coding
sequences. Or toggle the display of the GFF features/lines of type
intron, start codon, stop codon, CDS or ‘initial’, ‘internal’,
‘terminal’ and ‘single’ exon type names. The CDS excludes the stop codon
(unless stopCodonExcludedFromCDS=false) whereas the terminal and single
exon include the stop codon.
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–AUGUSTUS_CONFIG_PATH
</td>
<td style="text-align:left;width: 30em; ">
path
</td>
<td style="text-align:left;width: 50em; ">
Path to config directory (if not specified as environment variable)
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–alternatives-from-evidence
</td>
<td style="text-align:left;width: 30em; ">
true, false
</td>
<td style="text-align:left;width: 50em; ">
Report alternative transcripts when they are suggested by hints
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–alternatives-from-sampling
</td>
<td style="text-align:left;width: 30em; ">
true, false
</td>
<td style="text-align:left;width: 50em; ">
Report alternative transcripts generated through probabilistic sampling
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–sample
</td>
<td style="text-align:left;width: 30em; ">
n
</td>
<td style="text-align:left;width: 50em; ">
Set number of data points in sample
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–minexonintronprob
</td>
<td style="text-align:left;width: 30em; ">
p
</td>
<td style="text-align:left;width: 50em; ">
The posterior probability of every exon and every intron in a transcript
must be at least ‘minexonintronprob’
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–minmeanexonintronprob
</td>
<td style="text-align:left;width: 30em; ">
p
</td>
<td style="text-align:left;width: 50em; ">
The geometric mean of the probabilities of all exons and introns must be
at least ‘minmeanexonintronprob’.
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–maxtracks
</td>
<td style="text-align:left;width: 30em; ">
n
</td>
<td style="text-align:left;width: 50em; ">
The maximum number of tracks when displayed in a genome browser is
‘maxtracks’ (unless maxtracks=-1, then it is unbounded). In cases where
all transcripts of a gene overlap at some position this is also the
maximal number of transcripts for that gene… I recommend increasing the
parameter ‘maxtracks’ for improving sensitivity and setting ‘maxtracks’
to 1 and increasing minmeanexonintronprob and/or minexonintronprob in
order to improve specificity.
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–proteinprofile
</td>
<td style="text-align:left;width: 30em; ">
filename
</td>
<td style="text-align:left;width: 50em; ">
Read a protein profile from filename
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–predictionStart / –predictionEnd
</td>
<td style="text-align:left;width: 30em; ">
A / B
</td>
<td style="text-align:left;width: 50em; ">
Define the range of the sequence for which predictions should be found
(from A to B). Quicker if you only need predictions for a small part of
the genome.
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–gff3
</td>
<td style="text-align:left;width: 30em; ">
on, off
</td>
<td style="text-align:left;width: 50em; ">
Output in gff3 format
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–UTR
</td>
<td style="text-align:left;width: 30em; ">
on, off
</td>
<td style="text-align:left;width: 50em; ">
Predict the untranslated regions in addition to the coding sequence.
This currently works only for human, galdieria, toxoplasma and
caenorhabditis.
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–outfile
</td>
<td style="text-align:left;width: 30em; ">
filename
</td>
<td style="text-align:left;width: 50em; ">
print output to filename instead to standard output. This is useful for
computing environments, e.g. parasol jobs, which do not allow shell
redirection.
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–noInFrameStop
</td>
<td style="text-align:left;width: 30em; ">
on, off
</td>
<td style="text-align:left;width: 50em; ">
Don’t report transcripts with in-frame stop codons. Otherwise,
intron-spanning stop codons could occur. Default: false
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–noprediction
</td>
<td style="text-align:left;width: 30em; ">
on, off
</td>
<td style="text-align:left;width: 50em; ">
If true and input is in genbank format, no prediction is made. Useful
for getting the annotated protein sequences.
</td>
</tr>
<tr>
<td style="text-align:left;width: 20em; ">
–contentmodels
</td>
<td style="text-align:left;width: 30em; ">
on, off
</td>
<td style="text-align:left;width: 50em; ">
If ‘off’ the content models are disabled (all emissions uniformly 1/4).
The content models are; coding region Markov chain (emiprobs), initial
k-mers in coding region (Pls), intron and intergenic regin Markov chain.
This option is intended for special applications that require judging
gene structures from the signal models only, e.g. for predicting the
effect of SNPs or mutations on splicing. For all typical gene
predictions, this should be true. Default: on
</td>
</tr>
</tbody>
</table>
</div>
<div id="troubleshooting" class="section level3">
<h3>Troubleshooting</h3>
</div>
<div id="citations" class="section level3">
<h3>Citations</h3>
<p>Mario Stanke, Mark Diekhans, Robert Baertsch, David Haussler (2008).
<em>Using native and syntenically mapped cDNA alignments to improve de
novo gene finding</em>. Bioinformatics, 24(5), pages 637–644, doi:
10.1093/bioinformatics/btn013</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] kableExtra_1.4.0 knitr_1.50       workflowr_1.7.1 

loaded via a namespace (and not attached):
 [1] jsonlite_2.0.0    compiler_4.4.2    promises_1.3.2    Rcpp_1.0.14      
 [5] xml2_1.3.8        stringr_1.5.1     git2r_0.36.2      callr_3.7.6      
 [9] later_1.4.2       jquerylib_0.1.4   systemfonts_1.2.2 scales_1.3.0     
[13] yaml_2.3.10       fastmap_1.2.0     R6_2.6.1          tibble_3.2.1     
[17] munsell_0.5.1     rprojroot_2.0.4   svglite_2.1.3     bslib_0.9.0      
[21] pillar_1.10.2     rlang_1.1.6       cachem_1.1.0      stringi_1.8.7    
[25] httpuv_1.6.16     xfun_0.52         getPass_0.2-4     fs_1.6.6         
[29] sass_0.4.10       viridisLite_0.4.2 cli_3.6.4         magrittr_2.0.3   
[33] ps_1.9.1          digest_0.6.37     processx_3.8.6    rstudioapi_0.17.1
[37] lifecycle_1.0.4   vctrs_0.6.5       evaluate_1.0.3    glue_1.8.0       
[41] whisker_0.4.1     colorspace_2.1-1  rmarkdown_2.29    httr_1.4.7       
[45] tools_4.4.2       pkgconfig_2.0.3   htmltools_0.5.8.1</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
