<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>FormicidaeMolecularEvolution - Comparative Genomic Analysis Workflow</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="theme3.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Home</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-note-sticky"></span>
     
    Notes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">June</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="week1-notes.html">6/4 - 6/7</a>
        </li>
        <li>
          <a href="week2-notes.html">6/10 - 6/14</a>
        </li>
        <li>
          <a href="week3-notes.html">6/17 - 6/21</a>
        </li>
        <li>
          <a href="week4-notes.html">6/24 - 6/28</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">July</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="week5-notes.html">6/31 - 7/5</a>
        </li>
        <li>
          <a href="week6-notes.html">7/8 - 7/12</a>
        </li>
        <li>
          <a href="week7-notes.html">7/15 - 7/19</a>
        </li>
        <li>
          <a href="week9-notes.html">7/29 - 8/2</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">August</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="week12-notes.html">8/19 - 8/30</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">September</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="week14-notes.html">9/9 - 9/20</a>
        </li>
        <li>
          <a href="week16-notes.html">9/30 - 10/4</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-microscope"></span>
     
    Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="orthop-genomes-project.html">Comparative Genomics of Large Orthoptera Genomes</a>
    </li>
    <li class="dropdown-header">Comparative Genomics of Mormon Cricket subgroups -- UNDER CONSTRUCTION</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-code"></span>
     
    Program Information
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">TOGA &amp; Related Pipelines</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="toga.html">TOGA</a>
        </li>
        <li>
          <a href="mlc.html">make_lastz_chains</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="genespace.html">GENESPACE</a>
    </li>
    <li>
      <a href="compare-genomes.html">Compare_Genomes</a>
    </li>
    <li>
      <a href="barkdull.html">FormicidaeMolecularEvolution (OrthoFinder)</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Annotation Tools</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="Apollo.html">Apollo</a>
        </li>
        <li>
          <a href="augustus.html">Augustus</a>
        </li>
        <li>
          <a href="braker.html">Braker</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="rep-model-mask.html">RepeatModeler &amp; RepeatMasker</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-book-open-reader"></span>
     
    Reference
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="definitions.html">Glossary of Terminology</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">FormicidaeMolecularEvolution - Comparative
Genomic Analysis Workflow</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-01-16
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>song-lab/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240605code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240605)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240605code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240605)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomecbaker7tamusonglabtree388a8c2cf3a642caae516686cd45bfe3bd68b434targetblank388a8c2a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/ecbaker7-tamu/song-lab/tree/388a8c2cf3a642caae516686cd45bfe3bd68b434" target="_blank">388a8c2</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomecbaker7tamusonglabtree388a8c2cf3a642caae516686cd45bfe3bd68b434targetblank388a8c2a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/ecbaker7-tamu/song-lab/tree/388a8c2cf3a642caae516686cd45bfe3bd68b434" target="_blank">388a8c2</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store

Unstaged changes:
    Modified:   analysis/Publish all files
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/barkdull.Rmd</code>) and HTML
(<code>docs/barkdull.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/a9233f82b9e3a659c6612c7f6f8279fb2e79e21d/docs/barkdull.html" target="_blank">a9233f8</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/1cc30508b4a4fc9f2d8ea498d78360d880f3f2c8/analysis/barkdull.Rmd" target="_blank">1cc3050</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-10-17
</td>
<td>
Update OrthoFinder instructions
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/1cc30508b4a4fc9f2d8ea498d78360d880f3f2c8/docs/barkdull.html" target="_blank">1cc3050</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-10-17
</td>
<td>
Update OrthoFinder instructions
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/b96dd10e745964b9491094cb342a421b6affab80/docs/barkdull.html" target="_blank">b96dd10</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-10-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/421a156f8f4480f7d696f3b31547ac9ff0bf6385/docs/barkdull.html" target="_blank">421a156</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-10-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/e42e9f4dff38837ee65db845b9e5c7c9c955a9bc/analysis/barkdull.Rmd" target="_blank">e42e9f4</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-10-01
</td>
<td>
wflow_publish("analysis/barkdull.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/3731e9e5fc895d0bb1d501e825e40d8f92ec9e6d/docs/barkdull.html" target="_blank">3731e9e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/0c3ed6b76968749b6ad01b7a66f347fc1d7b8be3/docs/barkdull.html" target="_blank">0c3ed6b</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/79140076f6780bbbd8c216112c863ca77cdafb1a/docs/barkdull.html" target="_blank">7914007</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-08-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/98122725125fb3eac2898505b1869c1b94c26204/docs/barkdull.html" target="_blank">9812272</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/f2bc75a7088b701660c636f42bc845588b644ad2/docs/barkdull.html" target="_blank">f2bc75a</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/c93686e4e53271171e03b8c09ccd48906b416205/docs/barkdull.html" target="_blank">c93686e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/9a95e4e4daf346e551af00fb5642b117cba37e1e/analysis/barkdull.Rmd" target="_blank">9a95e4e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-15
</td>
<td>
wflow_publish(c("analysis/cafe5.Rmd", "analysis/augustus.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/783cca38f0df082611321d7be930f88510b823c9/docs/barkdull.html" target="_blank">783cca3</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/521747a727d407e3232e8bdb9db449cef883fc91/docs/barkdull.html" target="_blank">521747a</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/1126462a2adfb69005596da7dc536113a5fd9d75/docs/barkdull.html" target="_blank">1126462</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/94e9d6309b6decb954ef83ea7672e983e76b1e62/docs/barkdull.html" target="_blank">94e9d63</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/8b6255e939113ec6f1cee7981648a90f75c15417/docs/barkdull.html" target="_blank">8b6255e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/20907b2408f7dadb613d6dd05a69e7e8419df70b/analysis/barkdull.Rmd" target="_blank">20907b2</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-26
</td>
<td>
wflow_publish("analysis/barkdull.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/dd8cba58fffef3e461e9215365aa0d52e8ba757d/docs/barkdull.html" target="_blank">dd8cba5</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/857c1c6bac11546efd7c9889d7062f62328c47cd/analysis/barkdull.Rmd" target="_blank">857c1c6</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-25
</td>
<td>
wflow_publish("analysis/barkdull.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/7795b0e69e694beef0a09eb706b779cce048b02f/docs/barkdull.html" target="_blank">7795b0e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/2a916884a32bd5e25a54942186eb8db5f803fa91/analysis/barkdull.Rmd" target="_blank">2a91688</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-25
</td>
<td>
wflow_publish("analysis/barkdull.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/89fcf96dca7327ec45edc6a6fd7b87c2c7312094/docs/barkdull.html" target="_blank">89fcf96</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/b527cd257ba10139b51d58f50fd560762f179f19/analysis/barkdull.Rmd" target="_blank">b527cd2</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-24
</td>
<td>
wflow_publish("analysis/barkdull.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/08bae20c1abec5ad65336cb854a3cfea5156067f/docs/barkdull.html" target="_blank">08bae20</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/5652092132871a86f2cc66560fd387117de56799/analysis/barkdull.Rmd" target="_blank">5652092</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-20
</td>
<td>
updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/5652092132871a86f2cc66560fd387117de56799/docs/barkdull.html" target="_blank">5652092</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-20
</td>
<td>
updates
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="scripts-and-how-tos-for-comparative-genomics-work."
class="section level2 tabset">
<h2 class="tabset">Scripts and how-tos for comparative genomics
work.</h2>
<div id="what-is-formicidaemolecularevolution" class="section level3">
<h3>What is FormicidaeMolecularEvolution?</h3>
<p>This workflow was developed by Megan Barkdull of the Moreau lab at
Cornell University to find orthogroups between a reference and at least
1 query species. In particular, this pipeline allows for the user to
select for the longest <a
href="https://elifesciences.org/articles/34477">isoform</a>. This
feature allows us to skip analyzing repeating sequences in the
<em>schistocerca</em> genomes.</p>
<p>On her GitHub, Barkdull provides a flowchart that summarizes the
process of this pipeline and the purpose of each step. Pictured below is
the simplified version of this flowchart.</p>
<center>
<img src="assets/barkdull-flowchart-simple.png">
</center>
<center>
<i><font size="-1.5">Simplified FormicidaeMolecularEvolution Pipeline
Flowchart</font></i>
</center>
</div>
<div id="installation-and-running" class="section level3">
<h3>Installation and Running</h3>
<div id="clone-the-github-repository" class="section level4">
<h4>Clone the GitHub repository</h4>
<p>For both local and cluster installation, you just need to clone
Barkdull’s GitHub repository into your own environment using the
command</p>
<pre class="r"><code>git clone git@github.com:mbarkdull/FormicidaeMolecularEvolution.git</code></pre>
</div>
<div id="preparing-your-data-files" class="section level4">
<h4>Preparing your data files</h4>
<p>Barkdull provides a script that downloads all the data you need from
an FTP address, but the formatting must be within a .csv file, with the
URLs formatted within the columns as follows</p>
<table class="table table-striped table-hover table-condensed table-responsive" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
</caption>
<thead>
<tr>
<th style="text-align:left;font-weight: bold;background-color: rgba(191, 162, 176, 255) !important;">
Column_1
</th>
<th style="text-align:left;font-weight: bold;background-color: rgba(191, 162, 176, 255) !important;">
Column_2
</th>
<th style="text-align:left;font-weight: bold;background-color: rgba(191, 162, 176, 255) !important;">
Column_3
</th>
<th style="text-align:left;font-weight: bold;background-color: rgba(191, 162, 176, 255) !important;">
Column_4
</th>
<th style="text-align:left;font-weight: bold;background-color: rgba(191, 162, 176, 255) !important;">
Column_5
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;width: 30em; ">
Coding Sequence URL
</td>
<td style="text-align:left;width: 30em; ">
Protein Sequence URL
</td>
<td style="text-align:left;width: 30em; ">
Species abbreviation code
</td>
<td style="text-align:left;width: 30em; ">
Longest isoform? (yes/no)
</td>
<td style="text-align:left;width: 30em; ">
Comments
</td>
</tr>
</tbody>
</table>
<p>This should be done for each species, where the data for each species
is on its own row. So the format would be:</p>
<pre><code>Species 1: Coding seq. URL, Protein seq. URL, ...  
Species 2: Coding seq. URL, Protein seq. URL, ...  
Species 3: Coding seq. URL, Protein seq. URL, ...  </code></pre>
<p>and so on.</p>
<p>These files are easily created in Microsoft Excel, Google Sheets, or
any spreadsheet software by entering your data in the columns as shown
above and selecting the .csv option under “Save as”.</p>
<p>Then, move this file into the scripts folder within the cloned GitHub
repository. Finally, from the FormicidaeMolecularEvolution directory run
the command</p>
<pre class="r"><code>./scripts/DataDownload scripts/YourInputFile.csv</code></pre>
<p>This will download all the data you need to run this workflow.</p>
</div>
<div id="running-the-pipeline" class="section level4">
<h4>Running the Pipeline</h4>
<div id="step-1-set-up-the-environment" class="section level5">
<h5>Step 1: Set Up the Environment</h5>
<p>First we need to download PanDoc in order to update our packages as
needed without interference. We do this with the command</p>
<pre class="r"><code>ml GCC/12.2.0  OpenMPI/4.1.4 R_tamu/4.3.1

export R_LIBS=$SCRATCH/R_LIBS_USER/
  
ml Pandoc/2.13</code></pre>
<p>Then, we need to load the the necessary modules onto your terminal.
The command is as follows:</p>
<pre class="r"><code>R

library(&quot;orthologr&quot;)
library(&quot;biomartr&quot;)

quit()</code></pre>
<p><strong>Is orthologr not available?</strong> Copy and paste the code
from <a href="https://github.com/drostlab/orthologr">this link</a> under
<em>Install orthologr</em>.</p>
</div>
<div id="step-2-select-the-longest-isoforms" class="section level5">
<h5>Step 2: Select the Longest Isoforms</h5>
<p>The next step is the command</p>
<pre class="r"><code>./scripts/GeneRetrieval.R scripts/YourInputFile.csv</code></pre>
<p>Running this script selects for the single longest isoform in each
gene then matches protein names to gene names in order to subset the
coding sequence file to just the longest isoforms. This produces 2 files
for each species, one containing the longest protein isoforms and
another with the longest coding sequence isoforms in a directory called
<em>2_LongestIsoforms</em>.</p>
</div>
<div id="step-3-clean-the-raw-data" class="section level5">
<h5>Step 3: Clean the Raw Data</h5>
<p>The next step is to clean the data using the command</p>
<pre class="r"><code>./scripts/DataCleaning scripts/YourInputFile.csv</code></pre>
<p>This script removes any special characters from gene names and adds
the 4-letter taxon abbreviation to the beginning of each gene name. This
creates a new directory called <em>3_CleanedData</em>.</p>
</div>
<div id="step-4-translate-nucleotide-sequences-to-amino-acid-sequences"
class="section level5">
<h5>Step 4: Translate Nucleotide Sequences to Amino Acid Sequences</h5>
<p>For this next step we will need to alter one of Barkdull’s files with
some of our own lines.</p>
<ol style="list-style-type: lower-roman">
<li><p>Create a file in <em>scripts</em> called
<em>DataTranslating_modif</em>.</p></li>
<li><p>Copy and paste the code from the <em>scripts/DataTranslating</em>
file.</p></li>
<li><p>Replace lines 25-28 in <em>DataTranslating_modif</em> with the
following code</p></li>
</ol>
<pre class="r"><code>     # Now we can run Transdecoder on the cleaned file:
  echo &quot;First, attempting TransDecoder run on $cleanName&quot;
  ml GCC/10.2.0 OpenMPI/4.0.5 TransDecoder/5.5.0
  TransDecoder.LongOrfs -t $cleanName
  TransDecoder.Predict -t $cleanName --single_best_only</code></pre>
<ol start="4" style="list-style-type: lower-roman">
<li>Replace the header in <em>DataTranslating_modif</em> with the
following lines</li>
</ol>
<pre class="r"><code>  #!/bin/bash

  ##NECESSARY JOB SPECIFICATIONS
  #SBATCH --job-name=Transdecoder         #Set the job name to &quot;JobExample4&quot;
  #SBATCH --time=02:00:00         #Set the wall clock limit to 1hr and 30min
  #SBATCH --ntasks=2              #Request 2 task
  #SBATCH --cpus-per-task=4       #Request 8 task
  #SBATCH --mem=20G              #Request 50GB per node

  ml GCC/10.2.0 OpenMPI/4.0.5 TransDecoder/5.5.0 Biopython/1.78</code></pre>
<ol start="22" style="list-style-type: lower-alpha">
<li>Run the script using the command</li>
</ol>
<pre class="r"><code>sbatch ./scripts/DataTranslating_modif ./scripts/YourInputFile.csv</code></pre>
</div>
<div id="step-5-running-orthofinder" class="section level5">
<h5>Step 5: Running OrthoFinder</h5>
<p>For this step we will not be using anything from the original
pipeline’s steps, instead we will create our own directory and copy over
the translated fasta files as follows:</p>
<pre class="r"><code>  mkdir LocustsGenomeEvolution
  mkdir LocustsGenomeEvolution/tmp
  mkdir 5_OrthoFinder
  mkdir 5_OrthoFinder/fasta
  cp ./4_1_TranslatedData/OutputFiles/translated* ./5_OrthoFinder/fasta
  cd ./5_OrthoFinder/fasta
  rename translated &#39;&#39; translated*
  cd ..
  cd ..</code></pre>
<p>Then to run OrthoFinder on the Grace cluster we will create a file
within the <em>scripts</em> directory called <em>run-ortho.sh</em>. Copy
and paste the following code into <em>run-ortho.sh</em></p>
<pre class="r"><code>#!/bin/bash

##NECESSARY JOB SPECIFICATIONS
#SBATCH --job-name=orthofinder1         #Set the job name to &quot;JobExample4&quot;
#SBATCH --time=06:00:00         #Set the wall clock limit to 1hr and 30min
#SBATCH --ntasks=2              #Request 1 task
#SBATCH --cpus-per-task=4       #Request 1 task
#SBATCH --mem=30G              #Request 100GB per node

ml iccifort/2019.5.281  impi/2018.5.288 OrthoFinder/2.3.11-Python-3.7.4 
ml IQ-TREE/1.6.12 FastTree/2.1.11
ml MAFFT/7.453-with-extensions

orthofinder -S diamond -T iqtree -A mafft -I 5 -t 32 -a 4 -M msa -f ${file path to 5_OrthoFinder/fasta} -p ${file path to LocustsGenomeEvolution/tmp}</code></pre>
<details>
<summary>
What do these flags mean? (-S, -T, etc.)
</summary>
<p>-S diamond: Use diamond as the sequence search program.</p>
<p>-T iqtree: Use iqtree as the tree inference program.</p>
<p>-A mafft: Use mafft as the multiple sequence alignment (MSA)
program.</p>
<p>-I 5: Sets the MCL inflation parameter to 5.</p>
<p>-t 32: Sets the number of threads to 32.</p>
</details>
<p>Now we want to run OrthoFinder using BLAST to improve accuracy
(around 2% increase). To load this we use the command</p>
<pre class="r"><code>ml GCC/8.2.0-2.31.1  OpenMPI/3.1.3 iccifort/2019.5.281  impi/2018.5.288 BLAST+/2.9.0</code></pre>
<p>and then we can run our slurm script using</p>
<pre class="r"><code>sbatch scripts/run-ortho.sh</code></pre>
</div>
<div id="step-6-orthogroups-to-geneid" class="section level5">
<h5>Step 6: Orthogroups to GeneID</h5>
<p>First we must remove the suffixes that the pipeline puts in front of
our protein coding sequences, so create a file called
<em>remove-suffix.py</em> in the directory
5_OrthoFinder/fasta/OrthoFinder/Results_{DATE-OF-RUN}/Orthogroups and
copy the following code into it</p>
<pre class="r"><code>import re

# Define the input and output file names
input_file = &#39;Orthogroups.txt&#39;
output_file = &#39;Orthogroups_reprocessed.txt&#39;

# Read the input file
with open(input_file, &#39;r&#39;) as file:
    data = file.read()

# Remove all prefixes before any occurrence of &quot;_XP&quot;, &quot;_NP&quot;, and &quot;_YP&quot;
result = re.sub(r&#39;\b\w+_(XP|NP|YP)&#39;, r&#39;\1&#39;, data)

# Write the result to the output file
with open(output_file, &#39;w&#39;) as file:
    file.write(result)

print(f&quot;Processed data has been written to {output_file}&quot;)</code></pre>
</div>
<div id="step-7-create-a-table-and-clean-the-data"
class="section level5">
<h5>Step 7: Create a Table and Clean the Data</h5>
<p>We then need to create a table with each protein and gene ID for all
species present, so we must use the the script
<em>protein2geneid_loop.py</em> by David Emms, a software developer for
OrthoFinder.</p>
<p>Create a file called <em>protein2geneid_loop.py</em> in your
<em>1_RawData</em> directory and copy and paste the following code into
it</p>
<pre class="r"><code>import os
import re

# Get the current working directory
gff_directory = os.getcwd()

# Create a directory to store the output files
output_directory = os.path.join(gff_directory, &quot;output_files&quot;)
os.makedirs(output_directory, exist_ok=True)

# List GFF files with &quot;_GFF.gff&quot; extension
species_list = [filename for filename in os.listdir(gff_directory) if filename.endswith(&quot;_GFF.gff&quot;)]

# Process each species&#39; GFF file
for species_filename in species_list:
    # Extract the species name from the file name
    species_name = re.sub(r&quot;_GFF\.gff$&quot;, &quot;&quot;, species_filename)
    
    # Construct input and output file paths
    input_path = os.path.join(gff_directory, species_filename)
    output_filename = f&quot;xp{species_name}.gff&quot;  # Remove the dot before species_name
    output_path = os.path.join(output_directory, output_filename)
    
    # Use &#39;grep&#39; to filter lines containing &quot;XP&quot; and save to the output file
    grep_command = f&#39;grep &quot;XP&quot; &quot;{input_path}&quot; &gt; &quot;{output_path}&quot;&#39;
    os.system(grep_command)
    
    # Print a message indicating the filtering process
    print(f&quot;Filtered {species_filename} to {output_filename}&quot;)

    # Construct the output TSV file path
    tsv_output_filename = f&#39;gffKey{species_name}.tsv&#39;  # Remove the dot before species_name
    tsv_output_path = os.path.join(output_directory, tsv_output_filename)
    lol = {}

    # Read and process the contents of the GFF file
    with open(output_path) as gffFile:
        for line in gffFile:
            if re.search(r&quot;product=[^;]+&quot;, line):
                gene_match = re.search(r&quot;gene=[^;=]+&quot;, line)
                product_match = re.search(r&quot;product=[^;]+&quot;, line)
                proteinID_match = re.search(r&quot;protein_id=(.+)&quot;, line)
                
                if gene_match and product_match and proteinID_match:
                    gene = gene_match.group(0)
                    if gene.endswith(&#39;product&#39;):
                        gene = gene[5:-7]
                    else:
                        gene = gene[5:]
                    
                    product = product_match.group(0)
                    product = product.replace(&quot;product=&quot;, &quot;&quot;)
                    product = product.rstrip()
                    
                    proteinID = proteinID_match.group(1)
                    
                    if proteinID not in lol.keys():
                        lol[proteinID] = [gene, product, species_name]  # Add species name

    # Write the processed data to the output TSV file
    with open(tsv_output_path, &#39;w&#39;) as output:
        for proID, lis in lol.items():
            output.writelines(proID + &#39;\t&#39; + lis[0] + &#39;\t&#39; + lis[1] + &#39;\t&#39; + lis[2] + &#39;\n&#39;)  # Include species name

    # Print a message indicating the processing of the GFF file
    print(f&quot;Processed {output_filename} to {tsv_output_filename}&quot;)

# Concatenate all final files into a single file
final_output_filename = &quot;allspecies_protein2geneid.tsv&quot;
final_output_path = os.path.join(output_directory, final_output_filename)

with open(final_output_path, &#39;w&#39;) as final_output:
    for species_filename in species_list:
        species_name = re.sub(r&quot;_GFF\.gff$&quot;, &quot;&quot;, species_filename)
        tsv_output_filename = f&#39;gffKey{species_name}.tsv&#39;  # Remove the dot before species_name
        tsv_output_path = os.path.join(output_directory, tsv_output_filename)
        
        with open(tsv_output_path, &#39;r&#39;) as species_output:
            final_output.write(species_output.read())

print(f&quot;Concatenated all species files into {final_output_filename}&quot;)</code></pre>
<p>Then run this script in the folder <em>1_RawData</em> using</p>
<pre class="r"><code>cd {PATH TO 1_RawData}

python3 protein2geneid_loop.py</code></pre>
</div>
<div id="step-8-correct-possible-errors-in-table"
class="section level5">
<h5>Step 8: Correct Possible Errors in Table</h5>
<p>After we get this table, we need to run this script to clean some
small errors that may occur in the table. Create a file named
<em>clean-table.R</em> in your main OrthoFinder directory and copy and
paste the following code into it</p>
<pre class="r"><code>library(readr)
library(tidyverse)

orthogroup &lt;- read_table(&quot;data/orthologs/Orthogroups_reprocessed.txt&quot;, col_names = FALSE)

# We reshape the table so that we have one orthogroup line associated with one protein id
transformed_orthogroup &lt;- orthogroup %&gt;%
  pivot_longer(cols = -X1, names_to = &quot;Protein_Column&quot;, values_to = &quot;protein_id&quot;) %&gt;%
  replace_na(list(X1 = &quot;N/A&quot;)) %&gt;%
  select(orthogroup = X1, protein_id) %&gt;%
  filter(!is.na(protein_id))

# Export the table to  tab-separated text file (you can change the delimiter if needed)
write.table(transformed_orthogroup, file = &quot;data/orthologs/Orthogroups_11species_May2024.txt&quot;, sep = &quot;\t&quot;, quote = FALSE, row.names = FALSE)

proteingeneid &lt;- read_tsv(&quot;data/orthologs/allspecies_protein2geneid.tsv&quot;, col_names = TRUE ) 

final_orthotable &lt;- left_join(transformed_orthogroup, proteingeneid, by = &quot;protein_id&quot;)
write.table(final_orthotable, file = &quot;data/orthologs/Orthogroups_genesprotein_11species_May2024.csv&quot;, sep = &quot;,&quot;, quote = FALSE, row.names = FALSE)</code></pre>
<p>We can run this script in the main directory using the following
commands:</p>
<pre class="r"><code>ml GCC/12.2.0  OpenMPI/4.1.4 R_tamu/4.3.1

export R_LIBS=$SCRATCH/R_LIBS_USER/ # Can skip reloading these if still present from previous steps

Rscript clean-table.R</code></pre>
<p>Finally, we have finished and gotten a table of all species and their
corresponding IDs.</p>
</div>
</div>
</div>
<div id="troubleshooting" class="section level3">
<h3>Troubleshooting</h3>
<div id="no-module-named-bio" class="section level5">
<h5>No Module named Bio</h5>
<p>This error occurs when the program can’t find BioPython. Before the
“ml” line in your script add the command</p>
<pre class="r"><code>module purge</code></pre>
<p>This will get rid of all previously loaded packages before you reload
the ones you need for this script. Not quite sure why this works but it
does.</p>
</div>
</div>
<div id="citations" class="section level3">
<h3>Citations</h3>
<p>Megan Barkdull. (2021). mbarkdull/FormicidaeMolecularEvolution:
v1.0.0-beta (v1.0.0-beta). Zenodo. <a
href="https://doi.org/10.5281/zenodo.5668329">https://doi.org/10.5281/zenodo.5668329</a></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] kableExtra_1.4.0 knitr_1.49       workflowr_1.7.1 

loaded via a namespace (and not attached):
 [1] jsonlite_1.8.9    compiler_4.4.2    promises_1.3.2    Rcpp_1.0.14      
 [5] xml2_1.3.6        stringr_1.5.1     git2r_0.35.0      callr_3.7.6      
 [9] later_1.4.1       jquerylib_0.1.4   systemfonts_1.1.0 scales_1.3.0     
[13] yaml_2.3.10       fastmap_1.2.0     R6_2.5.1          tibble_3.2.1     
[17] munsell_0.5.1     rprojroot_2.0.4   svglite_2.1.3     bslib_0.8.0      
[21] pillar_1.10.1     rlang_1.1.4       cachem_1.1.0      stringi_1.8.4    
[25] httpuv_1.6.15     xfun_0.50         getPass_0.2-4     fs_1.6.5         
[29] sass_0.4.9        viridisLite_0.4.2 cli_3.6.3         magrittr_2.0.3   
[33] ps_1.8.1          digest_0.6.37     processx_3.8.5    rstudioapi_0.17.1
[37] lifecycle_1.0.4   vctrs_0.6.5       evaluate_1.0.3    glue_1.8.0       
[41] whisker_0.4.1     colorspace_2.1-1  rmarkdown_2.29    httr_1.4.7       
[45] tools_4.4.2       pkgconfig_2.0.3   htmltools_0.5.8.1</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
