<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>BRAKER</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="theme3.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Home</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-note-sticky"></span>
     
    Notes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">June</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="week1-notes.html">6/4 - 6/7</a>
        </li>
        <li>
          <a href="week2-notes.html">6/10 - 6/14</a>
        </li>
        <li>
          <a href="week3-notes.html">6/17 - 6/21</a>
        </li>
        <li>
          <a href="week4-notes.html">6/24 - 6/28</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">July</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="week5-notes.html">6/31 - 7/5</a>
        </li>
        <li>
          <a href="week6-notes.html">7/8 - 7/12</a>
        </li>
        <li>
          <a href="week7-notes.html">7/15 - 7/19</a>
        </li>
        <li>
          <a href="week9-notes.html">7/29 - 8/2</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">August</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="week12-notes.html">8/19 - 8/30</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">September</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="week14-notes.html">9/9 - 9/20</a>
        </li>
        <li>
          <a href="week16-notes.html">9/30 - 10/4</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-microscope"></span>
     
    Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="orthop-genomes-project.html">Comparative Genomics of Large Orthoptera Genomes</a>
    </li>
    <li class="dropdown-header">Comparative Genomics of Mormon Cricket subgroups -- UNDER CONSTRUCTION</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-code"></span>
     
    Program Information
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">TOGA &amp; Related Pipelines</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="toga.html">TOGA</a>
        </li>
        <li>
          <a href="mlc.html">make_lastz_chains</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="genespace.html">GENESPACE</a>
    </li>
    <li>
      <a href="compare-genomes.html">Compare_Genomes</a>
    </li>
    <li>
      <a href="barkdull.html">FormicidaeMolecularEvolution (OrthoFinder)</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Annotation Pipelines</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="augustus.html">Augustus</a>
        </li>
        <li>
          <a href="braker.html">Braker</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="rep-model-mask.html">RepeatModeler &amp; RepeatMasker</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-book-open-reader"></span>
     
    Reference
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="definitions.html">Glossary of Terminology</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">BRAKER</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-12-16
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>bioinformatics-notes/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240605code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240605)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240605code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240605)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomecbaker7tamusonglabtreee05dc1e7a4db86c6623bffe2a768bd6066d68ecbtargetblanke05dc1ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/ecbaker7-tamu/song-lab/tree/e05dc1e7a4db86c6623bffe2a768bd6066d68ecb" target="_blank">e05dc1e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomecbaker7tamusonglabtreee05dc1e7a4db86c6623bffe2a768bd6066d68ecbtargetblanke05dc1ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/ecbaker7-tamu/song-lab/tree/e05dc1e7a4db86c6623bffe2a768bd6066d68ecb" target="_blank">e05dc1e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/shiny/
    Ignored:    bootswatch/
    Ignored:    images/

Unstaged changes:
    Modified:   analysis/Publish all files
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/braker.Rmd</code>) and HTML
(<code>docs/braker.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/b96dd10e745964b9491094cb342a421b6affab80/docs/braker.html" target="_blank">b96dd10</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-10-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/3731e9e5fc895d0bb1d501e825e40d8f92ec9e6d/docs/braker.html" target="_blank">3731e9e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/0c3ed6b76968749b6ad01b7a66f347fc1d7b8be3/docs/braker.html" target="_blank">0c3ed6b</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/79140076f6780bbbd8c216112c863ca77cdafb1a/docs/braker.html" target="_blank">7914007</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-08-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/98122725125fb3eac2898505b1869c1b94c26204/docs/braker.html" target="_blank">9812272</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/b3c792c1aa348b9dd4988261e7498987ab7ac7f7/docs/braker.html" target="_blank">b3c792c</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/3a0177b9bfb5074f3d6490e2858e7fb417dc8a1c/analysis/braker.Rmd" target="_blank">3a0177b</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-31
</td>
<td>
wflow_publish(c("analysis/week9-notes.Rmd", "analysis/braker.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/85f95aaaf8530227409404255c659884125f7da8/docs/braker.html" target="_blank">85f95aa</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/57f9dfc90faef39e1fc465c3cf05a589cfc71cca/analysis/braker.Rmd" target="_blank">57f9dfc</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-19
</td>
<td>
wflow_publish(c("analysis/braker.Rmd", "analysis/week7-notes.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/9519d8ced3438ff21d05f90fc7718b531d72973d/docs/braker.html" target="_blank">9519d8c</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/282feaec18e7be39c002be61a79349a6eabc33fa/analysis/braker.Rmd" target="_blank">282feae</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-17
</td>
<td>
wflow_publish(c("analysis/week7-notes.Rmd", "analysis/braker.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/f2bc75a7088b701660c636f42bc845588b644ad2/docs/braker.html" target="_blank">f2bc75a</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/a46293c8b5b7fde2e10ed7b988f6549132f4004d/docs/braker.html" target="_blank">a46293c</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/421a51a0b955fda4308180791468d99ffbbfd06d/analysis/braker.Rmd" target="_blank">421a51a</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-15
</td>
<td>
wflow_publish(c("analysis/week7-notes.Rmd", "analysis/braker.Rmd"))
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="fully-automated-prediction-of-protein-coding-gene-structure"
class="section level2 tabset">
<h2 class="tabset">Fully automated prediction of protein coding gene
structure</h2>
<div id="what-is-braker" class="section level3">
<h3>What is BRAKER?</h3>
<p>From the Gaius-Augustus GitHub:</p>
<p><b><i>“The rapidly growing number of sequenced genomes requires fully
automated methods for accurate gene structure annotation. With this goal
in mind, we have developed BRAKER1R1R0, a combination of GeneMark-ET R2
and AUGUSTUS R3, R4, that uses genomic and RNA-Seq data to automatically
generate full gene structure annotations in novel genome.</p>
<p>However, the quality of RNA-Seq data that is available for annotating
a novel genome is variable, and in some cases, RNA-Seq data is not
available, at all.</p>
<p>BRAKER2 is an extension of BRAKER1 which allows for fully automated
training of the gene prediction tools GeneMark-ES/ET/EP/ETP R14, R15,
R17, F1 and AUGUSTUS from RNA-Seq and/or protein homology information,
and that integrates the extrinsic evidence from RNA-Seq and protein
homology information into the prediction.</p>
<p>In contrast to other available methods that rely on protein homology
information, BRAKER2 reaches high gene prediction accuracy even in the
absence of the annotation of very closely related species and in the
absence of RNA-Seq data.</p>
<p>BRAKER3 is the latest pipeline in the BRAKER suite. It enables the
usage of RNA-seq and protein data in a fully automated pipeline to train
and predict highly reliable genes with GeneMark-ETP and AUGUSTUS. The
result of the pipeline is the combined gene set of both gene prediction
tools, which only contains genes with very high support from extrinsic
evidence.”</b></i></p>
</div>
<div id="keys-to-sucessful-gene-prediction" class="section level3">
<h3>Keys to sucessful gene prediction</h3>
<ul>
<li><p>Use a high quality genome assembly. If you have a huge number of
very short scaffolds in your genome assembly, those short scaffolds will
likely increase runtime dramatically but will not increase prediction
accuracy.</p></li>
<li><p>Use simple scaffold names in the genome file
(e.g. <em>&gt;contig1</em> will work better than <em>&gt;contig1my
custom species namesome putative function /more/information/ and lots of
special characters %&amp;!(){}</em>). Make the scaffold names in all
your fasta files simple before running any alignment program.</p></li>
<li><p>In order to predict genes accurately in a novel genome, the
genome should be masked for repeats. This will avoid the prediction of
false positive gene structures in repetitive and low complexitiy
regions. Repeat masking is also essential for mapping RNA-Seq data to a
genome with some tools (other RNA-Seq mappers, such as HISAT2, ignore
masking information). In case of GeneMark-ES/ET/EP/ETP and AUGUSTUS,
softmasking (i.e. putting repeat regions into lower case letters and all
other regions into upper case letters) leads to better results than
hardmasking (i.e. replacing letters in repetitive regions by the letter
N for unknown nucleotide).</p></li>
<li><p>Many genomes have gene structures that will be predicted
accurately with standard parameters of GeneMark-ES/ET/EP/ETP and
AUGUSTUS within BRAKER. However, some genomes have clade-specific
features, i.e. special branch point model in fungi, or non-standard
splice-site patterns. Please read the options section [options] in order
to determine whether any of the custom options may improve gene
prediction accuracy in the genome of your target species.</p></li>
<li><p>Always check gene prediction results before further usage! You
can e.g. use a genome browser for visual inspection of gene models in
context with extrinsic evidence data. BRAKER supports the generation of
track data hubs for the UCSC Genome Browser with MakeHub for this
purpose.</p></li>
</ul>
</div>
<div id="installation-running" class="section level3">
<h3>Installation &amp; Running</h3>
<p>BRAKER is available on the Grace cluster, you can load it using the
command</p>
<pre class="r"><code>ml GCC/9.3.0  OpenMPI/4.0.3  BRAKER/3.0.3-Python-3.8.2</code></pre>
<p>Then, there are many ways you can run BRAKER, but for our purposes we
want to run with long-read RNA-Seq data and protein data. To do this we
need to use the following steps that are based on the poster from
Dr. Hoff <a
href="https://github.com/Gaius-Augustus/BRAKER/blob/master/docs/posters/poster_PAG2024.pdf">here</a>:</p>
<p><strong>1.</strong> Download the iso-seq data by creating a file
called <em>download-data.sh</em> and copy and paste the following code
into it:</p>
<pre class="r"><code>#!/bin/bash

##NECESSARY JOB SPECIFICATIONS
#SBATCH --job-name=prefetch1          #Set the job name to &quot;prefetch1&quot;
#SBATCH --time=2-00:00:00             #Set the wall clock limit to 2 days
#SBATCH --ntasks=2                    #Request 2 task
#SBATCH --cpus-per-task=12            #Request 12 cpus for each task
#SBATCH --mem=50G                     #Request 50GB per node

# Load necessary modules
ml GCC/11.3.0  OpenMPI/4.1.4 SRA-Toolkit/3.0.3

# enable proxy to allow compute node connection to internet
module load WebProxy

# Define variables
input_file=&quot;SRR_Acc_List_isoseq_2024.txt&quot;
output_dir=&quot;${Directory to BRAKER files}&quot;

# Prefetch the data
prefetch --option-file &quot;$input_file&quot;

# Ensure output directory exists
mkdir -p &quot;$output_dir&quot;

# Read accession numbers and process with fasterq-dump and gzip
while IFS= read -r accession; do
fastq-dump --split-3 --outdir &quot;$output_dir&quot; &quot;$accession&quot;
if [[ $? -ne 0 ]]; then
  echo &quot;Error: Processing accession $accession encountered an error.&quot;
  exit 1
fi

# Compress the resulting fastq files
for file in &quot;$output_dir&quot;/*; do
  if [[ -f $file &amp;&amp; $file == *.fastq ]]; then
    gzip &quot;$file&quot;
    if [[ $? -ne 0 ]]; then
      echo &quot;Error: Compressing file $file encountered an error.&quot;
      exit 1
    fi
  fi
done
done &lt; &quot;$input_file&quot;

echo &quot;fasterq-dump and gzip completed successfully.&quot;</code></pre>
<p>Then visit <a href="https://www.ncbi.nlm.nih.gov/sra/">this site</a>
to get the IsoSeq data SRR access numbers. Select all available files by
checking the box next to all the files and selecting Send &gt; File &gt;
Accession List to in the upper right hand corner. This will download a
.csv file with all of the accession numbers for your data. You then need
to create a file called <em>SRR_Acc_List_isoseq.txt</em> and copy and
paste the accession numbers into it. You can then run the
<em>download-data.sh</em> file.</p>
<p><br> <strong>2.</strong> Now you must combine all of the IsoSeq data,
map it to your genome of choice, and then convert the sam file to a BAM
file by copy and pasting the following script into a file called
<em>run-alignment.sh</em>:</p>
<pre class="r"><code>#!/bin/bash
#SBATCH --job-name=braker-step2           # Job name
#SBATCH --output=braker-step2.out         # Output file
#SBATCH --error=braker-step2.err          # Error file
#SBATCH --nodes=1                         # Number of nodes
#SBATCH --ntasks=1                        # Number of tasks
#SBATCH --cpus-per-task=12                # Number of CPU cores per task
#SBATCH --mem=200G                        # Total memory limit
#SBATCH --time=48:00:00                   # Time limit hrs:min:sec

# Load necessary modules
module load GCC/13.2.0 GCCcore/12.2.0 minimap2/2.26 SAMtools/13.2.0

# Concatenate FASTQ files
cat file-1.fastq file-2.fastq ... file-n.fastq &gt; all-subreads.fa

# Run minimap2 for alignment w/ increased memory allotment
minimap2 -t 12 -ax splice:hq -I4Gb --split-prefix=tmp GENOME.fna all-subreads.fa &gt; long_reads.sam

# Use Samtools to convert sam to BAM
samtools view -bS --threads 48 long_reads.sam -o isoseq.bam</code></pre>
<p><br> <strong>3.</strong> Build the Singularity image</p>
<pre class="r"><code>singularity build braker3_lr.sif docker://teambraker/braker3:isoseq</code></pre>
<p><br> <strong>4.</strong> Call BRAKER3 with the BAM file</p>
<pre class="r"><code>singularity exec -B ${PWD}:${PWD} braker3_lr.sif braker.pl --genome=genome.fa --prot_seq=protein_db.fa --bam=isoseq.bam --threads=48</code></pre>
</div>
<div id="troubleshooting" class="section level3">
<h3>Troubleshooting</h3>
<div id="samtools-esam_parse1-no-sq-lines-present-in-the-header"
class="section level4">
<h4>Samtools: [E::sam_parse1] no SQ lines present in the header</h4>
<p>SQ lines are used by Samtools to align sequences in .sam and BAM
files. This error occurs when SQ lines are not included in the .sam file
and the you try to convert it to a BAM file. To solve this issue, add
the flag –split-prefix=tmp to split up the alignments in order to retain
these SQ lines in the headers (I have now included this in my Step 2
command by default).</p>
</div>
</div>
<div id="citations" class="section level3">
<h3>Citations</h3>
<p>Stanke, M., Diekhans, M., Baertsch, R. and Haussler, D. (2008).
<em>Using native and syntenically mapped cDNA alignments to improve de
novo gene finding</em>. Bioinformatics, doi:
10.1093/bioinformatics/btn013.</p>
<p>Stanke. M., Schöffmann, O., Morgenstern, B. and Waack, S. (2006).
<em>Gene prediction in eukaryotes with a generalized hidden Markov model
that uses hints from external sources</em>. BMC Bioinformatics 7,
62.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 22631)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.7.1

loaded via a namespace (and not attached):
 [1] vctrs_0.6.5       httr_1.4.7        cli_3.6.2         knitr_1.47       
 [5] rlang_1.1.3       xfun_0.45         stringi_1.8.4     processx_3.8.4   
 [9] promises_1.3.0    jsonlite_1.8.8    glue_1.7.0        rprojroot_2.0.4  
[13] git2r_0.33.0      htmltools_0.5.8.1 httpuv_1.6.15     ps_1.7.6         
[17] sass_0.4.9        fansi_1.0.6       rmarkdown_2.27    jquerylib_0.1.4  
[21] tibble_3.2.1      evaluate_0.24.0   fastmap_1.2.0     yaml_2.3.8       
[25] lifecycle_1.0.4   whisker_0.4.1     stringr_1.5.1     compiler_4.4.0   
[29] fs_1.6.4          pkgconfig_2.0.3   Rcpp_1.0.12       rstudioapi_0.16.0
[33] later_1.3.2       digest_0.6.35     R6_2.5.1          utf8_1.2.4       
[37] pillar_1.9.0      callr_3.7.6       magrittr_2.0.3    bslib_0.7.0      
[41] tools_4.4.0       cachem_1.1.0      getPass_0.2-4    </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
