<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>GENESPACE</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="theme3.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Home</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-note-sticky"></span>
     
    Notes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">June</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="week1-notes.html">6/4 - 6/7</a>
        </li>
        <li>
          <a href="week2-notes.html">6/10 - 6/14</a>
        </li>
        <li>
          <a href="week3-notes.html">6/17 - 6/21</a>
        </li>
        <li>
          <a href="week4-notes.html">6/24 - 6/28</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">July</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="week5-notes.html">6/31 - 7/5</a>
        </li>
        <li>
          <a href="week6-notes.html">7/8 - 7/12</a>
        </li>
        <li>
          <a href="week7-notes.html">7/15 - 7/19</a>
        </li>
        <li>
          <a href="week9-notes.html">7/29 - 8/2</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">August</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="week12-notes.html">8/19 - 8/30</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">September</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="week14-notes.html">9/9 - 9/20</a>
        </li>
        <li>
          <a href="week16-notes.html">9/30 - 10/4</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-microscope"></span>
     
    Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="orthop-genomes-project.html">Comparative Genomics of Schistocerca gregaria</a>
    </li>
    <li class="dropdown-header">Comparative Genomics of Mormon Cricket subgroups -- UNDER CONSTRUCTION</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-code"></span>
     
    Program Information
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">TOGA &amp; Related Pipelines</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="toga.html">TOGA</a>
        </li>
        <li>
          <a href="mlc.html">make_lastz_chains</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="genespace.html">GENESPACE</a>
    </li>
    <li>
      <a href="compare-genomes.html">Compare_Genomes</a>
    </li>
    <li>
      <a href="barkdull.html">FormicidaeMolecularEvolution (OrthoFinder)</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Annotation Tools</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="Apollo.html">Apollo</a>
        </li>
        <li>
          <a href="augustus.html">Augustus</a>
        </li>
        <li>
          <a href="braker.html">Braker</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="rep-model-mask.html">RepeatModeler &amp; RepeatMasker</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-book-open-reader"></span>
     
    Reference
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="definitions.html">Glossary of Terminology</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">GENESPACE</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-04-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>song-lab/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240605code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240605)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240605code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240605)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomecbaker7tamusonglabtreea5e996b567a8526569a17d2dfdb2a6c20e6ae56atargetblanka5e996ba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/ecbaker7-tamu/song-lab/tree/a5e996b567a8526569a17d2dfdb2a6c20e6ae56a" target="_blank">a5e996b</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomecbaker7tamusonglabtreea5e996b567a8526569a17d2dfdb2a6c20e6ae56atargetblanka5e996ba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/ecbaker7-tamu/song-lab/tree/a5e996b567a8526569a17d2dfdb2a6c20e6ae56a" target="_blank">a5e996b</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/genespace.Rmd</code>) and HTML
(<code>docs/genespace.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/aaa78c46da9d439cbb501e9591ecc971e6283817/docs/genespace.html" target="_blank">aaa78c4</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-01-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/a9233f82b9e3a659c6612c7f6f8279fb2e79e21d/docs/genespace.html" target="_blank">a9233f8</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/b96dd10e745964b9491094cb342a421b6affab80/docs/genespace.html" target="_blank">b96dd10</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-10-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/3731e9e5fc895d0bb1d501e825e40d8f92ec9e6d/docs/genespace.html" target="_blank">3731e9e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/0c3ed6b76968749b6ad01b7a66f347fc1d7b8be3/docs/genespace.html" target="_blank">0c3ed6b</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/79140076f6780bbbd8c216112c863ca77cdafb1a/docs/genespace.html" target="_blank">7914007</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-08-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/98122725125fb3eac2898505b1869c1b94c26204/docs/genespace.html" target="_blank">9812272</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/f2bc75a7088b701660c636f42bc845588b644ad2/docs/genespace.html" target="_blank">f2bc75a</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/c93686e4e53271171e03b8c09ccd48906b416205/docs/genespace.html" target="_blank">c93686e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/9a95e4e4daf346e551af00fb5642b117cba37e1e/analysis/genespace.Rmd" target="_blank">9a95e4e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-15
</td>
<td>
wflow_publish(c("analysis/cafe5.Rmd", "analysis/augustus.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/f408e01e63de149efc5e0a070432d3763974b95c/docs/genespace.html" target="_blank">f408e01</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/b4a0fb0b33ecec7273c0ce69798b31c87dba8554/analysis/genespace.Rmd" target="_blank">b4a0fb0</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-03
</td>
<td>
wflow_publish("analysis/genespace.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/783cca38f0df082611321d7be930f88510b823c9/docs/genespace.html" target="_blank">783cca3</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/a60be8ed2b9562f92be143c0e35927110463bfcf/analysis/genespace.Rmd" target="_blank">a60be8e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-03
</td>
<td>
wflow_publish(c("analysis/barkdull.Rmd", "analysis/compare-genomes.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div
id="an-r-package-for-synteny--and-orthology-constrained-comparative-genomics"
class="section level2 tabset">
<h2 class="tabset">An R package for synteny- and orthology-constrained
comparative genomics</h2>
<div id="what-is-genespace" class="section level3">
<h3>What is GENESPACE?</h3>
<div class="img-with-text">
<style>
p {
  text-indent: 2em; /* Adjust the value as needed to simulate a tab */
}
</style>
<div class="text">
<p>
<b> “The development of multiple chromosome-scale reference genome
sequences in many taxonomic groups has yielded a high-resolution view of
the patterns and processes of molecular evolution. Nonetheless,
leveraging information across multiple genomes remains a significant
challenge in nearly all eukaryotic systems… Here, we present GENESPACE,
which addresses these challenges by integrating conserved gene order and
orthology to define the expected physical position of all genes across
multiple genomes.”</b>
</p>
<p>– Lovell <i> et. al.</i>, <i> GENESPACE tracks regions of interest
and gene copy number variation across multiple genomes </i></p>
</div>
<p><img src="assets/riparian-example.png" alt="Riparian Plot Example"></p>
</div>
<style>
.img-with-text {
  display: flex;
  margin-top: 5px;
  align-items: flex-start;
}

.img-with-text img {
  margin-right: 20px;
  margin-top: 10px;
  max-width: 550px;
  height: auto;
}

.img-with-text .text {
  flex: 3 0 400px;
  display: flex;
  flex-direction: column;
}
</style>
</div>
<div id="installation-running" class="section level3">
<h3>Installation &amp; Running</h3>
<div id="install-in-a-conda-environment" class="section level4">
<h4>Install in a Conda Environment</h4>
<p>These instructions are for installing GENESPACE within a conda
environment on the TAMU Grace cluster.</p>
<p>1.Install OrthoFinder by setting up a conda environment via the
procedure described <a
href="https://hprc.tamu.edu/kb/Software/ANACONDA/">here</a></p>
<p>After you’ve set up a conda environment, you should be able to
install orthofinder using the following command:</p>
<pre class="r"><code>conda install -c bioconda orthofinder</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Install MCScanX using the following commands in your scratch
directory:</li>
</ol>
<pre class="r"><code>git clone https://github.com/wyp1125/MCScanX
cd MCScanX
make</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>Finally, to install GENESPACE itself, you first need to load R and
specify path for local packages as described <a
href="https://hprc.tamu.edu/kb/Software/R-tamu/">here</a></li>
</ol>
<p>Here is an example of how to install R:</p>
<pre class="r"><code>module load GCC/12.2.0
module load OpenMPI/4.1.4
module load R_tamu/4.3.1

mkdir $SCRATCH/R_LIBS_USER/
export R_LIBS=$SCRATCH/R_LIBS_USER/
  ```

4. After you have installed R, run the following command in the conda environment
with orthofinder installed:

  ```{r, eval=FALSE}
R

if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)
BiocManager::install(c(&quot;Biostrings&quot;, &quot;rtracklayer&quot;))

if (!requireNamespace(&quot;devtools&quot;, quietly = TRUE))
    install.packages(&quot;devtools&quot;)
devtools::install_github(&quot;jtlovell/GENESPACE&quot;)</code></pre>
<ol start="5" style="list-style-type: decimal">
<li>Verify that GENESPACE have been installed by running</li>
</ol>
<pre class="r"><code>library(GENESPACE)</code></pre>
<p>If no error message appears, then GENESPACE has been installed
correctly.</p>
</div>
<div id="run-a-test-case" class="section level4">
<h4>Run a Test Case</h4>
<p>The following commands can be edited afterwards to run other data,
this is just the steps to create the example Riparian plot on the
GENESPACE GitHub.</p>
<p>The code provided below downloads the files, unzips them, check that
they’re formatted correctly, and then runs GENESPACE in preparation for
your analysis. You can remove the downloading parts of the script if yuo
already have the data downloaded, just make sure your directories are
set up correctly. If you look under <a
href="https://htmlpreview.github.io/?https://github.com/jtlovell/tutorials/blob/main/genespaceGuide.html">section
5.2 here</a> it will show the file structure you need to follow.</p>
<ol style="list-style-type: decimal">
<li>Create a file called <em>genespace-script.R</em> and copy and paste
the following code into it:</li>
</ol>
<pre class="r"><code>#####################  CHANGE THESE TO YOUR DIRECTORIES  ####################### 

genomeRepo &lt;- &quot;~/path/to/store/rawGenomes&quot;
wd &lt;- &quot;~/path/to/genespace/workingDirectory&quot;
path2mcscanx &lt;- &quot;~/path/to/MCScanX/&quot;

################################################################################



####################  CHANGE THESE TO YOUR GENOME URLS  ########################

urls &lt;- c(
  human =&quot;000/001/405/GCF_000001405.40_GRCh38.p14/GCF_000001405.40_GRCh38.p14_&quot;,
  mouse = &quot;000/001/635/GCF_000001635.27_GRCm39/GCF_000001635.27_GRCm39_&quot;,
  platypus = &quot;004/115/215/GCF_004115215.2_mOrnAna1.pri.v4/GCF_004115215.2_mOrnAna1.pri.v4_&quot;,
  chicken = &quot;016/699/485/GCF_016699485.2_bGalGal1.mat.broiler.GRCg7b/GCF_016699485.2_bGalGal1.mat.broiler.GRCg7b_&quot;,
  sandLizard = &quot;009/819/535/GCF_009819535.1_rLacAgi1.pri/GCF_009819535.1_rLacAgi1.pri_&quot;)

################################################################################




##############################  LEAVE THIS ALONE  ##############################

genomes2run &lt;- names(urls)
urls &lt;- file.path(&quot;https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF&quot;, urls)
translatedCDS &lt;- sprintf(&quot;%stranslated_cds.faa.gz&quot;, urls)
geneGff &lt;- sprintf(&quot;%sgenomic.gff.gz&quot;, urls)

names(translatedCDS) &lt;- genomes2run
names(geneGff) &lt;- genomes2run
writeDirs &lt;- file.path(genomeRepo, genomes2run)
names(writeDirs) &lt;- genomes2run
for(i in genomes2run){
  print(i)
  if(!dir.exists(writeDirs[i]))
    dir.create(writeDirs[i])
  download.file(
    url = geneGff[i], 
    destfile = file.path(writeDirs[i], basename(geneGff[i])))
  download.file(
    url = translatedCDS[i], 
    destfile = file.path(writeDirs[i], basename(translatedCDS[i])))
}
################################################################################



##########################  CHANGE TO YOUR GENOME NAMES  #######################

genomes2run &lt;- c(&quot;human&quot;, &quot;mouse&quot;, &quot;platypus&quot;, &quot;chicken&quot;, &quot;sandLizard&quot;)
parsedPaths &lt;- parse_annotations(
  rawGenomeRepo = genomeRepo,
  genomeDirs = genomes2run,
  genomeIDs = genomes2run,
  presets = &quot;ncbi&quot;,
  genespaceWd = wd)

################################################################################



# Check if input data is okay
gpar &lt;- init_genespace(
  wd = wd,
  path2mcscanx = path2mcscanx)


# Run GENESPACE
out &lt;- run_genespace(gpar, overwrite = T)

### INSERT DESIRED FUNCTIONS HERE ###</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>You can then run this using the command</li>
</ol>
<pre class="r"><code>source(&quot;genespace-script.R&quot;)</code></pre>
</div>
</div>
<div id="visualization-tools" class="section level3">
<h3>Visualization Tools</h3>
<div id="riparian-plots" class="section level4">
<h4>Riparian Plots</h4>
<center>
<img src="assets/riparian-example.png" width="400" />
</center>
<p>Riparian plots are used to visualize synteny between 2 or more
species. The command to create these plots in GENESPACE is</p>
<pre class="r"><code>ripd &lt;- plot_riparian(
  gsParam = out,
  refGenome = &quot;human&quot;, 
  useRegions = FALSE)</code></pre>
</div>
<div id="dotplots" class="section level4">
<h4>Dotplots</h4>
<center>
<img src="assets/dotplots.png" width="600" />
</center>
<p>Dot plots are used to visualize global pattern of synteny between a
pair of genomes. The following command produces 3 dot plots: (1) All
hits above a score threshold, (2) All hits where the query and target
genomes are in the same orthogroup, and (3) Syntenic anchor hits colored
by syntenic block ID. The command to do this is</p>
<pre class="r"><code>hits &lt;- read_allBlast(
  filepath = file.path(out$paths$syntenicHits, 
                       &quot;mouse_vs_human.allBlast.txt.gz&quot;))
ggdotplot(hits = hits, type = &quot;all&quot;, verbose = FALSE)</code></pre>
</div>
</div>
<div id="troubleshooting" class="section level3">
<h3>Troubleshooting</h3>
</div>
<div id="citations" class="section level3">
<h3>Citations</h3>
<p>John T Lovell, Avinash Sreedasyam, M Eric Schranz, Melissa Wilson,
Joseph W Carlson, Alex Harkess, David Emms, David M Goodstein, Jeremy
Schmutz (2022) <em>GENESPACE tracks regions of interest and gene copy
number variation across multiple genomes</em> eLife 11:e78526</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.7.1

loaded via a namespace (and not attached):
 [1] vctrs_0.6.5       httr_1.4.7        cli_3.6.4         knitr_1.49       
 [5] rlang_1.1.5       xfun_0.51         stringi_1.8.4     processx_3.8.5   
 [9] promises_1.3.2    jsonlite_1.8.9    glue_1.8.0        rprojroot_2.0.4  
[13] git2r_0.35.0      htmltools_0.5.8.1 httpuv_1.6.15     ps_1.8.1         
[17] sass_0.4.9        rmarkdown_2.29    jquerylib_0.1.4   tibble_3.2.1     
[21] evaluate_1.0.3    fastmap_1.2.0     yaml_2.3.10       lifecycle_1.0.4  
[25] whisker_0.4.1     stringr_1.5.1     compiler_4.4.2    fs_1.6.5         
[29] pkgconfig_2.0.3   Rcpp_1.0.14       rstudioapi_0.17.1 later_1.4.1      
[33] digest_0.6.37     R6_2.6.1          pillar_1.10.1     callr_3.7.6      
[37] magrittr_2.0.3    bslib_0.9.0       tools_4.4.2       cachem_1.1.0     
[41] getPass_0.2-4    </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
