<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>GENESPACE</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="theme3.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Home</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-microscope"></span>
     
    Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Comparative Genomics of Schistocerca gregaria -- In prep</li>
    <li class="dropdown-header">Comparative Genomics of Mormon Cricket subgroups -- UNDER CONSTRUCTION</li>
    <li class="dropdown-header">Ensifera Ear Transcriptomics -- UNDER CONSTRUCTION</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Comparative Genomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">FormicidaeMolecularEvolution - OrthoFinder Workflow for Longest Isoform</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="barkdull.html">TAMU Workflow</a>
        </li>
        <li>
          <a href="barkdull_asu.html">ASU Workflow</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">RepeatModeler &amp; RepeatMasker</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="rep-model-mask.html">TAMU Workflow</a>
        </li>
        <li>
          <a href="rep-model-mask_asu.html">ASU Workflow</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">PSMC Model Analysis</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="psmc.html">TAMU Workflow</a>
        </li>
        <li>
          <a href="psmc_asu.html">ASU Workflow</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Transcriptomics</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="tf_mapping.html">Transcription Factor Mapping (ASU Only)</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Proteomics</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="alphafold.html">AlphaFold</a>
        </li>
        <li>
          <a href="chimerax.html">Chimera X</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-palette"></span>
     
    Visualization
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="kimura.html">Kimura Plots (Repeat Content)</a>
    </li>
    <li>
      <a href="genespace.html">GENESPACE (Synteny)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-book-open-reader"></span>
     
    Reference
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="slurm-comm.html">Linux and Slurm Commands</a>
    </li>
    <li>
      <a href="definitions.html">Glossary of Terminology</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">GENESPACE</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-09-17
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>song-lab/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240605code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240605)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240605code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240605)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomemilycbakersonglabtree36abff94147c225664d99cf5ce238f07b4dc794atargetblank36abff9a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/emily-cbaker/song-lab/tree/36abff94147c225664d99cf5ce238f07b4dc794a" target="_blank">36abff9</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomemilycbakersonglabtree36abff94147c225664d99cf5ce238f07b4dc794atargetblank36abff9a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/emily-cbaker/song-lab/tree/36abff94147c225664d99cf5ce238f07b4dc794a" target="_blank">36abff9</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store

Unstaged changes:
    Modified:   analysis/Publish all files
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/genespace.Rmd</code>) and HTML
(<code>docs/genespace.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/791a625b1fac52a4b4467889a7f51664d0b334da/docs/genespace.html" target="_blank">791a625</a>
</td>
<td>
emilyc-baker
</td>
<td>
2025-09-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/2dff33cea2a8edc040224e29433a47bb56f26835/docs/genespace.html" target="_blank">2dff33c</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-07-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/97c5e6ce376a27871cdc61b162a81d1d314c3409/docs/genespace.html" target="_blank">97c5e6c</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-07-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/a1b980b1cdd4fe50a1a3d6b10949e589e8f1b825/docs/genespace.html" target="_blank">a1b980b</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-07-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/4e146c4576f9c498f75eb81aa9193b039c9d4fcc/docs/genespace.html" target="_blank">4e146c4</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-07-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/7a5615778c5c2cf842b2ae1b8c5969112f4bef73/docs/genespace.html" target="_blank">7a56157</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-05-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/fecf69695ef32ed4f732693ce7fb0749bedc3a3e/docs/genespace.html" target="_blank">fecf696</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-05-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/9ed28350316f5760a28dc7c92480d718b24a7c5b/docs/genespace.html" target="_blank">9ed2835</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-05-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/emily-cbaker/song-lab/blob/6c7409ad7dbd0803cc23cedb29d86249853e21cc/analysis/genespace.Rmd" target="_blank">6c7409a</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-05-20
</td>
<td>
wflow_publish(c("analysis/alphafold.Rmd", "analysis/Apollo.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/05539da9915300f741c6fda620038213ded63e5b/docs/genespace.html" target="_blank">05539da</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-05-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/emily-cbaker/song-lab/blob/638b4d2eb0c93f48a1f1679f39b19dd83557ada3/analysis/genespace.Rmd" target="_blank">638b4d2</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-05-20
</td>
<td>
wflow_publish(c("analysis/alphafold.Rmd", "analysis/Apollo.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/ad2ac0606d6ada9812f82d214d88171a2d69673a/docs/genespace.html" target="_blank">ad2ac06</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-04-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/729de9688ab12a7636a75643ec35e40f91cadddd/docs/genespace.html" target="_blank">729de96</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-04-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/emily-cbaker/song-lab/blob/b2b35b2d503db89b116e8155fd40983daf557fbf/analysis/genespace.Rmd" target="_blank">b2b35b2</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-04-11
</td>
<td>
wflow_publish(c("analysis/alphafold.Rmd", "analysis/Apollo.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/ce9e325bbd248cb113399d97cc81b34d75ef4935/docs/genespace.html" target="_blank">ce9e325</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/emily-cbaker/song-lab/blob/28bf89b8327c9f022c69ba353fff00ec944c99b3/analysis/genespace.Rmd" target="_blank">28bf89b</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-04-10
</td>
<td>
wflow_publish("analysis/genespace.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/63d1ad06520e7f288ebaa6562d63fe2865527bf7/docs/genespace.html" target="_blank">63d1ad0</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/aaa78c46da9d439cbb501e9591ecc971e6283817/docs/genespace.html" target="_blank">aaa78c4</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-01-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/a9233f82b9e3a659c6612c7f6f8279fb2e79e21d/docs/genespace.html" target="_blank">a9233f8</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/b96dd10e745964b9491094cb342a421b6affab80/docs/genespace.html" target="_blank">b96dd10</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-10-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/3731e9e5fc895d0bb1d501e825e40d8f92ec9e6d/docs/genespace.html" target="_blank">3731e9e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/0c3ed6b76968749b6ad01b7a66f347fc1d7b8be3/docs/genespace.html" target="_blank">0c3ed6b</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/79140076f6780bbbd8c216112c863ca77cdafb1a/docs/genespace.html" target="_blank">7914007</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-08-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/98122725125fb3eac2898505b1869c1b94c26204/docs/genespace.html" target="_blank">9812272</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/f2bc75a7088b701660c636f42bc845588b644ad2/docs/genespace.html" target="_blank">f2bc75a</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/c93686e4e53271171e03b8c09ccd48906b416205/docs/genespace.html" target="_blank">c93686e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/emily-cbaker/song-lab/blob/9a95e4e4daf346e551af00fb5642b117cba37e1e/analysis/genespace.Rmd" target="_blank">9a95e4e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-15
</td>
<td>
wflow_publish(c("analysis/cafe5.Rmd", "analysis/augustus.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/f408e01e63de149efc5e0a070432d3763974b95c/docs/genespace.html" target="_blank">f408e01</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/emily-cbaker/song-lab/blob/b4a0fb0b33ecec7273c0ce69798b31c87dba8554/analysis/genespace.Rmd" target="_blank">b4a0fb0</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-03
</td>
<td>
wflow_publish("analysis/genespace.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/783cca38f0df082611321d7be930f88510b823c9/docs/genespace.html" target="_blank">783cca3</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/emily-cbaker/song-lab/blob/a60be8ed2b9562f92be143c0e35927110463bfcf/analysis/genespace.Rmd" target="_blank">a60be8e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-03
</td>
<td>
wflow_publish(c("analysis/barkdull.Rmd", "analysis/compare-genomes.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<!-- This code gets rid of the auto-indent -->
<style>
/* Reset margin/padding/indentation for text elements, but NOT code blocks */
body, p, h1, h2, h3, h4, h5, h6, ol, ul, li, blockquote {
  margin-left: 0 !important;
  padding-left: 0 !important;
  text-indent: 0 !important;
}
</style>
<div
id="an-r-package-for-synteny--and-orthology-constrained-comparative-genomics"
class="section level2 tabset">
<h2 class="tabset">An R package for synteny- and orthology-constrained
comparative genomics</h2>
<div id="what-is-genespace" class="section level3">
<h3>What is GENESPACE?</h3>
<div class="img-with-text">
<style>
p {
  text-indent: 2em; /* Adjust the value as needed to simulate a tab */
}
</style>
<div class="text">
<p>
<b> “The development of multiple chromosome-scale reference genome
sequences in many taxonomic groups has yielded a high-resolution view of
the patterns and processes of molecular evolution. Nonetheless,
leveraging information across multiple genomes remains a significant
challenge in nearly all eukaryotic systems… Here, we present GENESPACE,
which addresses these challenges by integrating conserved gene order and
orthology to define the expected physical position of all genes across
multiple genomes.”</b>
</p>
<p>– Lovell <i> et. al.</i>, <i> GENESPACE tracks regions of interest
and gene copy number variation across multiple genomes </i></p>
</div>
<p><img src="assets/riparian-example.png" alt="Riparian Plot Example"></p>
</div>
<style>
.img-with-text {
  display: flex;
  margin-top: 5px;
  align-items: flex-start;
}

.img-with-text img {
  margin-right: 20px;
  margin-top: 10px;
  max-width: 550px;
  height: auto;
}

.img-with-text .text {
  flex: 3 0 400px;
  display: flex;
  flex-direction: column;
}
</style>
</div>
<div id="installation-running" class="section level3">
<h3>Installation &amp; Running</h3>
<p><mark> This page has been updated to run on the Grace cluster
following the March 2025 update, as well as the new GENESPACE 1.4
workflow.</mark></p>
<div id="load-packages-prepare-the-r-environment"
class="section level4">
<h4>Load packages &amp; Prepare the R environment</h4>
<p>These instructions are for installing GENESPACE within a conda
environment on the TAMU Grace cluster.</p>
<p><strong>1.</strong>To install GENESPACE itself, you first need to
load R and specify path for local packages as described <a
href="https://hprc.tamu.edu/kb/Software/R-tamu/">here</a></p>
<pre class="r"><code>ml GCC/12.3.0
ml OpenMPI/4.1.5
ml R_tamu/4.3.2
ml OrthoFinder/2.5.5
ml MCScanX/2024.12.19
export R_LIBS=$SCRATCH/R_LIBS_USER/4.3.2-foss-2023a:$R_LIBS</code></pre>
<p><strong>2.</strong> After you have installed R, run the following
command in the R terminal:</p>
<pre class="r"><code>R # Launches R terminal 

if (!requireNamespace(&quot;devtools&quot;, quietly = TRUE))
    install.packages(&quot;devtools&quot;)
devtools::install_github(&quot;jtlovell/GENESPACE&quot;)</code></pre>
<p><strong>3.</strong> Verify that GENESPACE have been installed by
running</p>
<pre class="r"><code>library(GENESPACE)</code></pre>
<p>If no error message appears, then GENESPACE has been installed
correctly. After this step, you may exit the R terminal with the
command</p>
<pre class="r"><code>q()</code></pre>
<br>
<hr style="border:2px solid lightgray">
</div>
<div id="running-with-your-data" class="section level4">
<h4>Running with your data</h4>
<p>The code provided below is meant to generate GENESPACE results from a
previous OrthoFinder run. Slight modifications can be made to run
OrthoFinder within the GENESPACE pipeline.</p>
<p><strong>1.</strong> Create the genome repository</p>
<p>GENESPACE expects a directory called the “genomeRepo” with a specific
file structure for each species as follows:</p>
<pre class="ascii"><code>└── genomeRepo
    └── species_gene_id
        ├── species_gene_id_genomic_gff.gz
        └── species_gene_id_translated_cds.faa.gz</code></pre>
<p>You must create a folder under the genomeRepo for each species you
want to run GENESPACE on and <strong>be sure to name the folder the
<em>same</em> as the gene id you assigned to the species in
OrthoFinder!</strong></p>
<p><strong>2.</strong> Create a file called <em>genespace-script.R</em>
and copy and paste the following code into it:</p>
<pre class="r"><code>library(GENESPACE)


library(GENESPACE)

genomeRepo &lt;- &quot;/scratch/group/songlab/emily/comparative-genomics/genespace/mormon-ID/genomeRepo&quot;
wd &lt;- &quot;/scratch/group/songlab/emily/comparative-genomics/SCMB/5_OrthoFinder/fasta/genespace&quot;
path2mcscanx &lt;- &quot;/sw/eb/sw/MCScanX/2024.12.19-foss-2023a/bin/&quot;

##########################  CHANGE TO YOUR GENOME NAMES  ####################### 

genomes2run &lt;- c(&quot;Sgreg&quot;, &quot;Scanc&quot;, &quot;Samer&quot;, &quot;Snite&quot;, &quot;Spice&quot;, &quot;Sscub&quot;, &quot;Csecu&quot;, &quot;Brsri&quot;, &quot;Amel&quot;, &quot;Sinvi&quot;, &quot;Apisu&quot;, &quot;Dmela&quot;, &quot;Ansimp&quot;)

################################################################################



parsedPaths &lt;- parse_annotations(
  rawGenomeRepo = genomeRepo,
  genomeDirs = genomes2run,
  genomeIDs = genomes2run,
  presets = &quot;ncbi&quot;,
  genespaceWd = wd)



gpar &lt;- init_genespace(
    wd = wd,
    path2mcscanx = path2mcscanx,
    rawOrthofinderDir = &quot;/Path/to/OrthoFinder/run&quot;) # Note: If you do not have a previous OrthoFinder run you want to run GENESPACE on, 
                                                    # just leave this option and only include the wd and path2mcscanx options and 
                                                    # GENESPACE will automatically run OrthoFinder for you

out &lt;- run_genespace(gpar)</code></pre>
<p><strong>3.</strong> You can then run this using the command</p>
<pre class="r"><code>Rscript genespace-script.R</code></pre>
<p>This generates the riparian and dot plots from the GENESPACE
pipeline.</p>
</div>
</div>
<div id="visualization-tools" class="section level3">
<h3>Visualization Tools</h3>
<div id="riparian-plots" class="section level4">
<h4>Riparian Plots</h4>
<p>Riparian plots are used to visualize synteny between 2 or more
species. The command to create these plots in GENESPACE is</p>
<pre class="r"><code>ripd &lt;- plot_riparian(
  gsParam = out,
  refGenome = &quot;human&quot;, 
  useRegions = FALSE)</code></pre>
<p>This code produces a graph that looks like this:</p>
<center>
<img src="assets/riparian-example.png" />
</center>
<div id="customizing-riparian-plots" class="section level5">
<h5>Customizing Riparian Plots</h5>
<p>We can also generate riparian plots for regions of interest using the
following commands:</p>
<p>To plot against the whole genome:</p>
<pre class="r"><code>roibed &lt;- roi[,c(&quot;genome&quot;, &quot;chr&quot;)]
roibed$color &lt;- c(&quot;pink&quot;, &quot;cyan&quot;, &quot;green&quot;)
ripd &lt;- plot_riparian(
  gsParam = out, 
  useRegions = FALSE, 
  highlightBed = roibed)</code></pre>
<p>This generates a plot that looks like this:</p>
<p><img
src="assets/riparian-whole-genome-regions-of-interest.png" /></p>
<p>To plot against only the regions of interest:</p>
<pre class="r"><code>ripd &lt;- plot_riparian(
  gsParam = out,  
  useRegions = FALSE,
  highlightBed = roibed, 
  backgroundColor = NULL)</code></pre>
<p>This generates a plot that looks like this:</p>
<p><img src="assets/riparian-only-regions-of-interest.png" /></p>
</div>
</div>
<div id="dotplots" class="section level4">
<h4>Dotplots</h4>
<p>Dot plots are used to visualize global pattern of synteny between a
pair of genomes. The following command produces 3 dot plots: (1) All
hits above a score threshold, (2) All hits where the query and target
genomes are in the same orthogroup, and (3) Syntenic anchor hits colored
by syntenic block ID. The command to do this is</p>
<pre class="r"><code>hits &lt;- read_allBlast(
  filepath = file.path(out$paths$syntenicHits, 
                       &quot;mouse_vs_human.allBlast.txt.gz&quot;))
ggdotplot(hits = hits, type = &quot;all&quot;, verbose = FALSE)</code></pre>
<p>This generates a plot that looks like this:</p>
<p><img src="assets/dotplots.png" /></p>
<center>
Image credits: Lovell et al. 
</center>
</div>
</div>
<div id="troubleshooting" class="section level3">
<h3>Troubleshooting</h3>
</div>
<div id="citations" class="section level3">
<h3>Citations</h3>
<p>John T Lovell, Avinash Sreedasyam, M Eric Schranz, Melissa Wilson,
Joseph W Carlson, Alex Harkess, David Emms, David M Goodstein, Jeremy
Schmutz (2022) <em>GENESPACE tracks regions of interest and gene copy
number variation across multiple genomes</em> eLife 11:e78526</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Phoenix
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.7.2

loaded via a namespace (and not attached):
 [1] vctrs_0.6.5       httr_1.4.7        cli_3.6.5         knitr_1.50       
 [5] rlang_1.1.6       xfun_0.53         stringi_1.8.7     processx_3.8.6   
 [9] promises_1.3.3    jsonlite_2.0.0    glue_1.8.0        rprojroot_2.1.1  
[13] git2r_0.36.2      htmltools_0.5.8.1 httpuv_1.6.16     ps_1.9.1         
[17] sass_0.4.10       rmarkdown_2.29    jquerylib_0.1.4   tibble_3.3.0     
[21] evaluate_1.0.5    fastmap_1.2.0     yaml_2.3.10       lifecycle_1.0.4  
[25] whisker_0.4.1     stringr_1.5.2     compiler_4.4.2    fs_1.6.6         
[29] pkgconfig_2.0.3   Rcpp_1.1.0        rstudioapi_0.17.1 later_1.4.4      
[33] digest_0.6.37     R6_2.6.1          pillar_1.11.0     callr_3.7.6      
[37] magrittr_2.0.4    bslib_0.9.0       tools_4.4.2       cachem_1.1.0     
[41] getPass_0.2-4    </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
