<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>make_lastz_chains</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="theme3.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Home</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-note-sticky"></span>
     
    Notes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">June</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="week1-notes.html">6/4 - 6/7</a>
        </li>
        <li>
          <a href="week2-notes.html">6/10 - 6/14</a>
        </li>
        <li>
          <a href="week3-notes.html">6/17 - 6/21</a>
        </li>
        <li>
          <a href="week4-notes.html">6/24 - 6/28</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">July</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="week5-notes.html">6/31 - 7/5</a>
        </li>
        <li>
          <a href="week6-notes.html">7/8 - 7/12</a>
        </li>
        <li>
          <a href="week7-notes.html">7/15 - 7/19</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-microscope"></span>
     
    Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="orthop-genomes-project.html">Comparative Genomics of Large Orthoptera Genomes</a>
    </li>
    <li>
      <a href="pop-genomics-morm-crick.html">Population Genomics of Mormon Crickets, Central American Locusts, Eremidium Grasshoppers</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-code"></span>
     
    Program Information
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">TOGA &amp; Related Pipelines</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="toga.html">TOGA</a>
        </li>
        <li>
          <a href="mlc.html">make_lastz_chains</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="genespace.html">GENESPACE</a>
    </li>
    <li>
      <a href="compare-genomes.html">Compare_Genomes</a>
    </li>
    <li>
      <a href="barkdull.html">FormicidaeMolecularEvolution (OrthoFinder)</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Annotation Pipelines</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="augustus.html">Augustus</a>
        </li>
        <li>
          <a href="braker.html">Braker</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-book-open-reader"></span>
     
    Reference
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="definitions.html">Glossary of Terminology</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">make_lastz_chains</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-07-15
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>bioinformatics-notes/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240605code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240605)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240605code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240605)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomecbaker7tamusonglabtree377496f45c0c0ad90e233f05c3554297fa81105btargetblank377496fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/ecbaker7-tamu/song-lab/tree/377496f45c0c0ad90e233f05c3554297fa81105b" target="_blank">377496f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomecbaker7tamusonglabtree377496f45c0c0ad90e233f05c3554297fa81105btargetblank377496fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/ecbaker7-tamu/song-lab/tree/377496f45c0c0ad90e233f05c3554297fa81105b" target="_blank">377496f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/shiny/
    Ignored:    bootswatch/
    Ignored:    images/

Unstaged changes:
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/mlc.Rmd</code>) and HTML
(<code>docs/mlc.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/c93686e4e53271171e03b8c09ccd48906b416205/docs/mlc.html" target="_blank">c93686e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/9a95e4e4daf346e551af00fb5642b117cba37e1e/analysis/mlc.Rmd" target="_blank">9a95e4e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-15
</td>
<td>
wflow_publish(c("analysis/cafe5.Rmd", "analysis/augustus.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/783cca38f0df082611321d7be930f88510b823c9/docs/mlc.html" target="_blank">783cca3</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/a60be8ed2b9562f92be143c0e35927110463bfcf/analysis/mlc.Rmd" target="_blank">a60be8e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-07-03
</td>
<td>
wflow_publish(c("analysis/barkdull.Rmd", "analysis/compare-genomes.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/521747a727d407e3232e8bdb9db449cef883fc91/docs/mlc.html" target="_blank">521747a</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/1126462a2adfb69005596da7dc536113a5fd9d75/docs/mlc.html" target="_blank">1126462</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/ad8ffae6b7fadda9860fff7d5f5b2d3bfe93339f/docs/mlc.html" target="_blank">ad8ffae</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/f23c37e754eb27304dc42564f4f61df0e3c82606/analysis/mlc.Rmd" target="_blank">f23c37e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-28
</td>
<td>
wflow_publish(c("analysis/week4-notes.Rmd",
"analysis/orthop-genomes-project.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/94e9d6309b6decb954ef83ea7672e983e76b1e62/docs/mlc.html" target="_blank">94e9d63</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/8a621bdf242f76e9ab93716d2c690dd7bfae85d5/analysis/mlc.Rmd" target="_blank">8a621bd</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-27
</td>
<td>
wflow_publish(c("analysis/index.Rmd", "analysis/week4-notes.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/a64d72711660f0baffa8eca1e5f5bb5ae3f4aaf1/docs/mlc.html" target="_blank">a64d727</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-26
</td>
<td>
commit images
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/7ab1aa198291b18c7f74ce5b49291eda6110df35/docs/mlc.html" target="_blank">7ab1aa1</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/a87b0a82eefad50f339a689d0f1d7c9e2806e691/analysis/mlc.Rmd" target="_blank">a87b0a8</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-24
</td>
<td>
wflow_publish("analysis/mlc.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/11fefb7898fe90d3464bcb5f4d530a4232781ed2/docs/mlc.html" target="_blank">11fefb7</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/e7e5c1a7e64ada1d372c7cd8610874cacf21eb8e/analysis/mlc.Rmd" target="_blank">e7e5c1a</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-19
</td>
<td>
commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/e7e5c1a7e64ada1d372c7cd8610874cacf21eb8e/docs/mlc.html" target="_blank">e7e5c1a</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-19
</td>
<td>
commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/a53134bcb7eea19f0ab5254c99f102c17c849f89/docs/mlc.html" target="_blank">a53134b</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/bb84281b2b6c35bb5f2577709bc03509a4881e66/analysis/mlc.Rmd" target="_blank">bb84281</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-18
</td>
<td>
wflow_publish("analysis/mlc.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="creating-genome-alignments-for-toga"
class="section level2 tabset">
<h2 class="tabset">Creating Genome Alignments for TOGA</h2>
<div id="what-is-make_lastz_chains" class="section level3">
<h3>What is make_lastz_chains?</h3>
<blockquote>
<p>This program is also from Hiller Lab (see <a
href="toga.html">TOGA</a>) and it is a pipeline that takes in a
reference and a query genome either as .fna or .2bit files and creates
pairwise genome alignment pairs from them for use in TOGA.</p>
</blockquote>
<p>Read more in depth on their GitHub <a
href="https://github.com/hillerlab/make_lastz_chains">here</a>.</p>
</div>
<div id="installation-and-running" class="section level3">
<h3>Installation and Running</h3>
<center>
<mark style="background: #f9d9da">This program is supported on MacOS,
but the Hiller Lab team <em>highly<br />
recommends</em> running it on a Linux-based machine.</mark>
</center>
<div id="prepare-files" class="section level4">
<h4>Prepare Files</h4>
<p>To acquire files to put into this pipeline, find your genome(s) of
interest on NCBI, click on the FTP tab, and download the file ending in
genomic.fna.gz. Then create a folder called <em>data</em> and copy your
files from your system into this folder. Then run the command</p>
<pre class="r"><code>gunzip YOUR-GENOME_genomic.fna.gz</code></pre>
<p>This will turn your file from a .fna.gz to just .fna. Then create a
file called <em>remove-headers.sh</em> in your <em>data</em> folder and
copy the following code into it.</p>
<pre class="r"><code>#!/bin/bash

# Define the input and output files
input_file=&quot;input-name.fna&quot;
output_file=&quot;input-name_only_contig_name.fna&quot;

# Use sed to process both headers and sequences in one pass
sed -e &#39;s/^\(&gt;\S*\).*/\1/&#39; &quot;$input_file&quot; &gt; &quot;$output_file&quot;

echo &quot;Processing complete. Output written to $output_file.&quot;</code></pre>
<p>This script removes everything except the contig name from the
headers in the .fna file. We do this because make_lastz_chains will
through a spacing error otherwise. <u> You must do this for each file
you intend to create chains for.</u> You are now ready to run this
pipeline!</p>
<p>You can also run RepeatMasker and RepeatModeler on your genome files.
The compatible versions are RepeatMasker/4.1.4 and
RepeatModeler/2.0.5.</p>
</div>
<div id="running-on-grace" class="section level4">
<h4>Running on Grace</h4>
<p>To run the make_lastz_chains pipeline on Grace, create a file in your
folder for make_lastz_chains called <em>run-mlc.sh</em> and copy and
paste the following code into it:</p>
<pre class="r"><code>#!/bin/sh
#SBATCH --time=0-1:00:00
#SBATCH --ntasks=48
#SBATCH --mem=50G
#SBATCH --nodes=1
#SBATCH --job-name=MLC-TEST

# module purge

module load GCC/12.3.0  make_lastz_chains/1.0.0  # load modules

conda activate mlc

./make_chains.py target query ${Path to Target Genome} ${Path to Query Genome} --pd test_out -f --chaining_memory 16 --executor slurm</code></pre>
<p>Then run this script using the command</p>
<pre class="r"><code>sbatch run-mlc.sh</code></pre>
</div>
<div id="running-in-a-conda-environment" class="section level4">
<h4>Running in a Conda Environment</h4>
<p>Follow the instructions on the Hiller Lab GitHub <a
href="https://github.com/hillerlab/make_lastz_chains">here</a> to
install.</p>
</div>
</div>
<div id="troubleshooting" class="section level3">
<h3>Troubleshooting</h3>
<div
id="fatal-could-not-read-from-remote-repository.-please-make-sure-you-have-the-correct-access-rights-and-the-repository-exists."
class="section level4">
<h4>fatal: Could not read from remote repository. Please make sure you
have the correct access rights and the repository exists.</h4>
<p>The full error is here:</p>
<pre class="r"><code>Warning: Permanently added the ECDSA host key for IP address &#39;140.82.113.3&#39; to the list of known hosts. Permission denied (publickey). fatal: Could not read from remote repository. Please make sure you have the correct access rights and the repository exists.</code></pre>
<p>I corrected this by going into my home &gt; .ssh directory in Grace
and adding the passkey from <em>id_rsa.pub</em> to my GitHub SSH keys.
This is the public key the Grace terminal uses to download files from
GitHub. This allowed me to download from the Hiller Lab repository.</p>
<p>If your <em>id_rsa.pub</em> file is empty, visit <a
href="https://www.theserverside.com/blog/Coffee-Talk-Java-News-Stories-and-Opinions/Fix-GitHubs-Permission-denied-publickey-SSH-error">this
link</a>. It has the full documentation that I followed to fix this
error. Here you will find instructions on how to get the necessary
information in your <em>id_rsa.pub</em> file.</p>
</div>
<div id="error-detected-space-or-tab-containing-sequence"
class="section level4">
<h4>Error: detected space-or-tab-containing sequence</h4>
<p>The whole error is below:</p>
<pre class="r"><code>Error! File: data/Nitens-and-Greg/GCF_021461395.2_iqSchAmer2.1_genomic.fna - detected space-or-tab-containing sequence:
NC_060119.1 Schistocerca americana isolate TAMUIC-IGC-003095 chromosome 1, iqSchAmer2.1, whole genome shotgun sequence
Please exclude or fix sequences with spaces and tabs.</code></pre>
<p>To correct this error, first run your .fna files through this
script:</p>
<pre class="r"><code>#!/bin/bash

# Define the input and output files
input_file=&quot;input-name.fna&quot;
output_file=&quot;input-name_only_contig_name.fna&quot;

# Use sed to process both headers and sequences in one pass
sed -e &#39;s/^\(&gt;\S*\).*/\1/&#39; &quot;$input_file&quot; &gt; &quot;$output_file&quot;

echo &quot;Processing complete. Output written to $output_file.&quot;</code></pre>
<p>This removes everything following the contig name in each header of
the fasta file.</p>
</div>
<div id="error-unrecognized-arguments--f" class="section level4">
<h4>Error: unrecognized arguments: -f</h4>
<p>The -f or -force flag is meant to overwrite an output folder if a
folder exists with the same name already. This error occurs when there
is no pre-existing folder to overwrite, so simply remove this flag from
your command and rerun.</p>
</div>
<div id="modulenotfounderror-no-module-named-dataclasses"
class="section level4">
<h4>ModuleNotFoundError: No module named ‘dataclasses’</h4>
<p>This error occurs for me when the incorrect GCC module is loaded on
Grace, verify that GCC 12.3.0 is available on the cluster and if not
contact IT.</p>
<p>If running locally, verify that the correct GCC is loaded on your
machine.</p>
</div>
<div id="error-the-nextflow-process-lastz-crashed"
class="section level4">
<h4>Error! The nextflow process lastz crashed!</h4>
<p>This error is addressed on the Hiller Lab GitHub page <a
href="https://github.com/hillerlab/make_lastz_chains/issues/16">here</a>,
but here is the comment from Dr. Hiller containing both the source of
the issue and the solution: “I think, based on the log file, you don’t
get to the chaining step, as some of the lastz jobs fail (increasing
chaining mem is downstream). This typically happens when lastz ‘sees’
too many seeds from repeats.</p>
<p>If you already repeatModeled (a denovo RepeatModeler run is
important) and masked your genome, you could try reducing chunksize
(e.g. set –seq1_chunk 50000000 –seq2_chunk 10000000 ). Gives you more
but smaller jobs.”</p>
<p>So by adding the –seq1_chunk 50000000 and –seq2_chunk 10000000 flags
to your <em>run-mlc.sh</em> file, this problem should be fixed.</p>
</div>
<div id="number-of-jobs-should-not-exceed-6000" class="section level4">
<h4>Number of jobs should not exceed 6000</h4>
<p>Run the command</p>
<pre class="r"><code>rm -rf /scratch/group/songlab/emily/comparative-genomics/make_lastz_chains/greg-and-serialis/TEMP_run.lastz /scratch/group/songlab/emily/comparative-genomics/make_lastz_chains/greg-and-serialis/TEMP_psl</code></pre>
<p>This command is provided by the error message in the
make_lastz_chains module. If this does not work, increase your node
memory in your slurm script and the chaining memory in your ./run-mlc.sh
command. For example, IT had me increase my node memory from 50G to 360G
and my chaining memory from 16 to 200.</p>
</div>
</div>
<div id="citations" class="section level3">
<h3>Citations</h3>
<p>Kirilenko BM, Munegowda C, Osipova E, Jebb D, Sharma V, Blumer M,
Morales A, Ahmed AW, Kontopoulos DG, Hilgers L, Lindblad-Toh K, Karlsson
EK, Zoonomia Consortium, Hiller M. <em>Integrating gene annotation with
orthology inference at scale.</em>, Science, 380, 2023</p>
<p>Osipova E, Hecker N, Hiller M. <em>RepeatFiller newly identifies
megabases of aligning repetitive sequences and improves annotations of
conserved non-exonic elements</em>. GigaScience, 8(11), giz132, 2019</p>
<p>Suarez H, Langer BE, Ladde P, Hiller M. <em>chainCleaner improves
genome alignment specificity and sensitivity</em>. Bioinformatics,
33(11), 1596-1603, 2017</p>
<p>Kent WJ, Baertsch R, Hinrichs A, Miller W, Haussler D.
<em>Evolution’s cauldron: Duplication, deletion, and rearrangement in
the mouse and human genomes</em>, PNAS, 100(20):11484-9, 2003</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 22631)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.7.1

loaded via a namespace (and not attached):
 [1] vctrs_0.6.5       httr_1.4.7        cli_3.6.2         knitr_1.47       
 [5] rlang_1.1.3       xfun_0.45         stringi_1.8.4     processx_3.8.4   
 [9] promises_1.3.0    jsonlite_1.8.8    glue_1.7.0        rprojroot_2.0.4  
[13] git2r_0.33.0      htmltools_0.5.8.1 httpuv_1.6.15     ps_1.7.6         
[17] sass_0.4.9        fansi_1.0.6       rmarkdown_2.27    jquerylib_0.1.4  
[21] tibble_3.2.1      evaluate_0.24.0   fastmap_1.2.0     yaml_2.3.8       
[25] lifecycle_1.0.4   whisker_0.4.1     stringr_1.5.1     compiler_4.4.0   
[29] fs_1.6.4          pkgconfig_2.0.3   Rcpp_1.0.12       rstudioapi_0.16.0
[33] later_1.3.2       digest_0.6.35     R6_2.5.1          utf8_1.2.4       
[37] pillar_1.9.0      callr_3.7.6       magrittr_2.0.3    bslib_0.7.0      
[41] tools_4.4.0       cachem_1.1.0      getPass_0.2-4    </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
