<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Pairwise Sequentially Markovian Coalescent (PSMC) Model</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="theme3.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Home</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-microscope"></span>
     
    Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Comparative Genomics of Schistocerca gregaria -- In prep</li>
    <li class="dropdown-header">Comparative Genomics of Mormon Cricket subgroups -- UNDER CONSTRUCTION</li>
    <li class="dropdown-header">Ensifera Ear Transcriptomics -- UNDER CONSTRUCTION</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-code"></span>
     
    Workflows
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">TAMU</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="barkdull.html">FormicidaeMolecularEvolution - OrthoFinder Workflow for Longest Isoform</a>
        </li>
        <li>
          <a href="rep-model-mask.html">RepeatModeler &amp; RepeatMasker</a>
        </li>
        <li>
          <a href="psmc.html">PSMC Model Analysis</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">ASU</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="barkdull_asu.html">FormicidaeMolecularEvolution - OrthoFinder Workflow for Longest Isoform</a>
        </li>
        <li>
          <a href="rep-model-mask_asu.html">RepeatModeler &amp; RepeatMasker</a>
        </li>
        <li>
          <a href="psmc_asu.html">PSMC Model Analysis</a>
        </li>
        <li class="dropdown-header">Transcriptomics</li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Proteomics</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="alphafold.html">AlphaFold</a>
        </li>
        <li>
          <a href="chimerax.html">Chimera X</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-palette"></span>
     
    Visualization
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="kimura.html">Kimura Plots (Repeat Content)</a>
    </li>
    <li>
      <a href="genespace.html">GENESPACE (Synteny)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-book-open-reader"></span>
     
    Reference
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="slurm-comm.html">Linux and Slurm Commands</a>
    </li>
    <li>
      <a href="definitions.html">Glossary of Terminology</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Pairwise Sequentially Markovian Coalescent
(PSMC) Model</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-09-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>song-lab/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240605code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240605)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240605code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240605)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomemilycbakersonglabtree5cedb3d95a698f69039c19ccdb2e1bf63737eca7targetblank5cedb3da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/emily-cbaker/song-lab/tree/5cedb3d95a698f69039c19ccdb2e1bf63737eca7" target="_blank">5cedb3d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomemilycbakersonglabtree5cedb3d95a698f69039c19ccdb2e1bf63737eca7targetblank5cedb3da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/emily-cbaker/song-lab/tree/5cedb3d95a698f69039c19ccdb2e1bf63737eca7" target="_blank">5cedb3d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store

Unstaged changes:
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/psmc_asu.Rmd</code>) and HTML
(<code>docs/psmc_asu.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/emily-cbaker/song-lab/blob/959b8210675c3b5a73bb951298a634887878f34c/analysis/psmc_asu.Rmd" target="_blank">959b821</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-09-09
</td>
<td>
Updates for ASU (some)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<!-- This code gets rid of the auto-indent -->
<style>
/* Reset margin/padding/indentation for text elements, but NOT code blocks */
body, p, h1, h2, h3, h4, h5, h6, ol, ul, li, blockquote {
  margin-left: 0 !important;
  padding-left: 0 !important;
  text-indent: 0 !important;
}
</style>
<div id="population-size-history-inference"
class="section level3 tabset">
<h3 class="tabset">Population Size History Inference</h3>
<div id="what-is-psmc" class="section level4">
<h4>What is PSMC?</h4>
<center>
<img src="assets/gregaria_bootstrapped_plot.png" style="width:70.0%" />
</center>
<p>The Pairwise Sequentially Markovian Coalescent Model (PSMC) is a
model that infers a species’ population history using HiC data from the
species in question. This process is motivated by “Coalescence Theory”,
which is a statistical theory that relates the size of a population to
the coalescence times in the genealogy of sampled individuals (Mather et
al.). This is done by mapping HiC reads back to the genome and searching
for loci splits via variant calling and bootstrapping results to create
a coalescence model.</p>
<p><a href="https://www.ncbi.nlm.nih.gov/sra">Download HiC Data from
NCBI</a></p>
</div>
<div id="installation-running" class="section level4">
<h4>Installation &amp; Running</h4>
<p>In the drop down below the full script to run the entire pipeline in
one slurm script is available. Below this, each step is broken down for
informational and troubleshooting purposes. You can run this procedure
step-by-step or all in one script, but for the first run I would
recommend doing a step-by-step run as it simplifies troubleshooting and
familiarizes you with the procedure.</p>
<details>
<summary>
<b>Full PSMC Script</b>
</summary>
<pre class="r"><code>#!/bin/bash

##NECESSARY JOB SPECIFICATIONS
#SBATCH --job-name=PSMC-SPECIES           # Set the job name to &quot;PSMC-SPECIES&quot; (change to your species name)
#SBATCH --time=7-00:00:00                   # Set the wall clock limit to 7 dyas
#SBATCH --ntasks=1                          # Request 1 task
#SBATCH --cpus-per-task=8                   # Request 8 cpus
#SBATCH --mem=30G                           # Request 30GB per node
#SBATCH --mail-type=ALL                           # Email updates for all steps
#SBATCH --mail-user=user@@mail.com        # Email to this address
#SBATCH --output=stdout.%x.%j                   # Name stdout file with run name and job Id
#SBATCH --error=stderr.%x.%j                    # Name stderr file with run name and job Id


########################### STEP 0: CHANGE THESE TO YOUR DATA #########################

SPECIES_NAME_ABBR = &quot;Sgregaria&quot;
SRR_NUM = &quot;SRR17149718&quot;
GENOME_FILE = &quot;GCF_023897955.1_iqSchGreg1.2_genomic.fa&quot;

#######################################################################################



################################ STEP 1: Align with BWA ###############################
module purge 
ml gcc-13.2.0-gcc-12.1.0 BWA/0.7.18 SAMtools/1.19.2 

bwa index -p &quot;${SPECIES_NAME_ABBR}&quot; &quot;${GENOME_FILE}&quot;

# assuming you have 8 threads, not that this .sam file will be huge, and we&#39;ll remove it at the end
bwa mem -t 8 {SPECIES_NAME_ABBR.} &quot;${SRR_NUM}_1.fastq.gz&quot; &quot;${SRR_NUM}_2.fastq.gz&quot; &gt; ${base}.sam

# convert all .sam to .bam
samtools view -S -b &quot;${SRR_NUM}.sam&quot; &gt; unsorted_alignment.bam

# Merge all .bam files into one
samtools merge unsorted_alignment.bam *.bam

#######################################################################################



################################ STEP 2: Sort .bam file ###############################
module purge
module load picard/2.25.1-Java-11 

java -jar $EBROOTPICARD/picard.jar SortSam \
      INPUT=unsorted_alignment.bam \
      OUTPUT=sorted.bam \
      SORT_ORDER=coordinate

#######################################################################################



################### STEP 3: Generate variant consensus fastq for PSMC #################
module purge
ml gcc-13.2.0-gcc-12.1.0 BCFtools/1.19

bcftools mpileup -Q 20 -q 20 -f GCF_023897955.1_iqSchGreg1.2_genomic.fa \
    sorted.bam --threads 8 | \
    bcftools call -c --threads 8 | \
    vcfutils.pl vcf2fq -d 1 -D 100 -Q 20 &gt; variant_consensus.fq

module purge
ml GCC/13.2.0 psmc/0.6.5_20221121 parallel/20240322 gnuplot/6.0.1 texlive/20230313

#######################################################################################



######################### STEP 4: Generate PSMC analysis &amp; plot #######################
# Convert .fq to .psmcfa
fq2psmcfa variant_consensus.fq &gt; psmc_input.psmcfa

# Generate PSMC results
psmc -N25 -t15 -r5 -p &quot;4+25*2+4+6&quot; -o sample.psmc psmc_input.psmcfa

mkdir boot

# Bootstrap results and plot 
parallel &#39;psmc -N25 -t10 -r5 -b -p &quot;8*1+30*2+4+6&quot; \
    -o $SLURM_SUBMIT_DIR/boot/sample_r{}.psmc $SLURM_SUBMIT_DIR/psmc_input.psmcfa&#39; \
    ::: $(seq 50)

cat sample.psmc boot/sample_r*.psmc &gt; sample_combined.psmc

psmc_plot.pl -p -g 2.2 -u 0.00000000506 bootstrapped_plot  sample_combined.psmc


#######################################################################################</code></pre>
</details>
<p><strong>1.</strong> First, you must map the HiC reads back to the
original genome using BWA. Doing this step on the (enormous) <em>S.
gregaria</em> genome (8.3 Gb) took around 1-2 days. We do this using the
following procedure:</p>
<pre class="r"><code>#!/bin/bash

##NECESSARY JOB SPECIFICATIONS
#SBATCH --job-name=create-unsorted-alignment        #Set the job name to &quot;psmc-referencegenome&quot;
#SBATCH --time=3-00:00:00         #Set the wall clock limit to 2 hours
#SBATCH --ntasks=1             #Request 1 task
#SBATCH --cpus-per-task=8       #Request 8 cpus
#SBATCH --mem=30G              #Request 30GB per node
#SBATCH --output=stdout.%x.%j
#SBATCH --error=stderr.%x.%j

# Align with BWA
module purge 
ml GCC/13.2.0 BWA/0.7.18 SAMtools/1.19.2 

# bwa index -p Sgregaria GCF_023897955.1_iqSchGreg1.2_genomic.fa
# assuming you have 8 threads, not that this .sam file will be huge, and we&#39;ll remove it at the end

bwa mem -t 8 Sgregaria SRR17149718_1.fastq SRR17149718_2.fastq &gt; SRR17149718.sam

# convert .sam to .bam
samtools view -S -b SRR17149718.sam &gt; unsorted_alignment.bam</code></pre>
<p><strong>2.</strong> Sort the unsorted .bam file</p>
<pre class="r"><code>##NECESSARY JOB SPECIFICATIONS
#SBATCH --job-name=sort-alignment        #Set the job name to &quot;psmc-referencegenome&quot;
#SBATCH --time=1-00:00:00         #Set the wall clock limit to 2 hours
#SBATCH --ntasks=1             #Request 1 task
#SBATCH --cpus-per-task=8       #Request 8 cpus
#SBATCH --mem=30G              #Request 30GB per node
#SBATCH --output=stdout.%x.%j
#SBATCH --error=stderr.%x.%j


# sort the .bam file
module purge
module load picard/2.25.1-Java-11 

java -jar $EBROOTPICARD/picard.jar SortSam \
      INPUT=unsorted_alignment.bam \
      OUTPUT=sorted.bam \
      SORT_ORDER=coordinate</code></pre>
<p><strong>3.</strong> Generate a variant consensus file from the sorted
.bam file</p>
<pre class="r"><code>#!/bin/bash

##NECESSARY JOB SPECIFICATIONS
#SBATCH --job-name=create-var-con                   # Set the job name to &quot;create-var-con&quot;
#SBATCH --time=2-00:00:00                           # Set the wall clock limit to 2 days
#SBATCH --ntasks=1                                  # Request 1 task
#SBATCH --cpus-per-task=8                           # Request 8 cpus
#SBATCH --mem=30G                                   # Request 30GB per node
#SBATCH --mail-type=ALL                                       # Email updates for all steps
#SBATCH --mail-user=user@mail.edu                       # Email to this address
#SBATCH --output=stdout.%x.%j                       # Name stdout file after job name and ID
#SBATCH --error=stderr.%x.%j                        # Name stderr file after job name and ID


module purge
ml GCC/13.2.0 BCFtools/1.19

# Generate consensus fastq for PSMC
bcftools mpileup -Q 20 -q 20 -f GCF_023897955.1_iqSchGreg1.2_genomic.fa \
    sorted.bam --threads 8 | \
    bcftools call -c --threads 8 | \
    vcfutils.pl vcf2fq -d 1 -D 100 -Q 20 &gt; variant_consensus.fq</code></pre>
<p><strong>4.</strong> Generate PSMC results</p>
<pre class="r"><code>#!/bin/bash

##NECESSARY JOB SPECIFICATIONS
#SBATCH --job-name=psmc-input-and-plot
#SBATCH --time=4-00:00:00
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=8
#SBATCH --mem=120G
#SBATCH --mail-type=ALL
#SBATCH --mail-user=ecbaker7@tamu.edu
#SBATCH --output=stdout.%x.%j
#SBATCH --error=stderr.%x.%j

module purge
ml GCC/13.2.0 psmc/0.6.5_20221121 parallel/20240322 gnuplot/6.0.1 texlive/20230313

# Convert .fq to .psmcfa
fq2psmcfa variant_consensus.fq &gt; psmc_input.psmcfa

# Generate PSMC results
psmc -N25 -t15 -r5 -p &quot;4+25*2+4+6&quot; -o sample.psmc psmc_input.psmcfa

mkdir boot

# Bootstrap results and plot 
parallel &#39;psmc -N25 -t10 -r5 -b -p &quot;8*1+30*2+4+6&quot; \
    -o $SLURM_SUBMIT_DIR/boot/sample_r{}.psmc $SLURM_SUBMIT_DIR/psmc_input.psmcfa&#39; \
    ::: $(seq 50)

cat sample.psmc boot/sample_r*.psmc &gt; sample_combined.psmc

psmc_plot.pl -p -g 2.2 -u 0.00000000506 bootstrapped_plot  sample_combined.psmc</code></pre>
<div id="optional-adding-a-title-to-your-plot" class="section level5">
<h5>OPTIONAL: Adding a title to your plot</h5>
<center>
<b><a href="orthop-genomes-project.html#psmc-analysis">Example of a
Titled Plot</a></b>
</center>
<p>%&gt;% There is no option to add a plot title in the psmc module, so
if you want your plot to be named, you’ll have to directly edit the
gnuplot file. Search for a file in your PSMC dirctory ending in the
extension <em>.gnuplot</em> or <em>.gp</em>. Then type</p>
<pre class="r"><code>nano YOURFILE.gp</code></pre>
<p>into your terminal to edit the gnuplot file. Then, find where the x
and y labels are set. The file layout should look something like
this:</p>
<pre class="r"><code>  set size 1, 0.8;
  set xran [10000:*];
  set log x;
  set format x &quot;10^{%L}&quot;;
  set mxtics 10;
  set mytics 10;
  unset grid;
  set key off;
  set xtics font &quot;Helvetica,16&quot;;
  set ytics nomirror font &quot;Helvetica,16&quot;;
  set xlab &quot;Years (g=2.2, {/Symbol m}=0.5x10^{-8})&quot; font &quot;Helvetica,16&quot;;
  set t po eps enhance so co &quot;Helvetica,16&quot;;

  set yran [0:*];
  set ylab &quot;Effective population size (x10^4)&quot; font &quot;Helvetica,16&quot;;
  set out &quot;Schistocerca piceifrons.eps&quot;;
  set style line 1 lt 1 lc rgb &quot;#FF0000&quot; lw 4;</code></pre>
<p>and add the line set title “Your Title” below the set x and y label
commands. For example, my file looks like this for <em>Schistocerca
piceifrons</em>:</p>
<pre class="r"><code> set size 1, 0.8;
  set xran [10000:*];
  set log x;
  set format x &quot;10^{%L}&quot;;
  set mxtics 10;
  set mytics 10;
  unset grid;
  set key off;
  set xtics font &quot;Helvetica,16&quot;;
  set ytics nomirror font &quot;Helvetica,16&quot;;
  set xlab &quot;Years (g=2.2, {/Symbol m}=0.5x10^{-8})&quot; font &quot;Helvetica,16&quot;;
  set t po eps enhance so co &quot;Helvetica,16&quot;;

  set yran [0:*];
  set ylab &quot;Effective population size (x10^4)&quot; font &quot;Helvetica,16&quot;;

  set title &quot;Schistocerca piceifrons&quot; font &quot;Helvetica-Italic,24&quot;;

  set out &quot;Schistocerca piceifrons.eps&quot;;
  set style line 1 lt 1 lc rgb &quot;#FF0000&quot; lw 4;</code></pre>
</div>
</div>
<div id="citations" class="section level4">
<h4>Citations</h4>
<p>Mather, Niklas, et al. “A Practical Introduction to Sequentially
Markovian Coalescent Methods for Estimating Demographic History From
Genomic Data.” Ecology and Evolution, vol. 10, no. 1, Dec. 2019,
pp. 579–89, <a href="doi:10.1002/ece3.5888"
class="uri">doi:10.1002/ece3.5888</a>.</p>
<p>Unm-Carc. “QuickBytes/psmc_quickbyte.md at Master ·
UNM-CARC/QuickBytes.” GitHub,
github.com/UNM-CARC/QuickBytes/blob/master/psmc_quickbyte.md.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Phoenix
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.7.1

loaded via a namespace (and not attached):
 [1] vctrs_0.6.5       httr_1.4.7        cli_3.6.4         knitr_1.50       
 [5] rlang_1.1.6       xfun_0.52         stringi_1.8.7     processx_3.8.6   
 [9] promises_1.3.2    jsonlite_2.0.0    glue_1.8.0        rprojroot_2.0.4  
[13] git2r_0.36.2      htmltools_0.5.8.1 httpuv_1.6.16     ps_1.9.1         
[17] sass_0.4.10       rmarkdown_2.29    jquerylib_0.1.4   tibble_3.2.1     
[21] evaluate_1.0.3    fastmap_1.2.0     yaml_2.3.10       lifecycle_1.0.4  
[25] whisker_0.4.1     stringr_1.5.1     compiler_4.4.2    fs_1.6.6         
[29] pkgconfig_2.0.3   Rcpp_1.0.14       rstudioapi_0.17.1 later_1.4.2      
[33] digest_0.6.37     R6_2.6.1          pillar_1.10.2     callr_3.7.6      
[37] magrittr_2.0.3    bslib_0.9.0       tools_4.4.2       cachem_1.1.0     
[41] getPass_0.2-4    </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
