<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Repeat Modeler and Masker</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="theme3.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Home</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-microscope"></span>
     
    Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="orthop-genomes-project.html">Comparative Genomics of Schistocerca gregaria</a>
    </li>
    <li class="dropdown-header">Comparative Genomics of Mormon Cricket subgroups -- UNDER CONSTRUCTION</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-code"></span>
     
    Program Information
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">TOGA &amp; Related Pipelines</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="toga.html">TOGA</a>
        </li>
        <li>
          <a href="mlc.html">make_lastz_chains</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="genespace.html">GENESPACE</a>
    </li>
    <li>
      <a href="barkdull.html">FormicidaeMolecularEvolution (OrthoFinder)</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Annotation Tools</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="Apollo.html">Apollo</a>
        </li>
        <li>
          <a href="augustus.html">Augustus</a>
        </li>
        <li>
          <a href="braker.html">Braker</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="rep-model-mask.html">RepeatModeler &amp; RepeatMasker</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Proteomics</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="alphafold.html">AlphaFold</a>
        </li>
        <li>
          <a href="chimerax.html">Chimera X</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-book-open-reader"></span>
     
    Reference
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="slurm-comm.html">Linux and Slurm Commands</a>
    </li>
    <li>
      <a href="definitions.html">Glossary of Terminology</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Repeat Modeler and Masker</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-07-03
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>song-lab/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240605code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240605)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240605code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240605)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomecbaker7tamusonglabtreeeb41d80e797e0c21e97cf8322df06d5a6cb56aedtargetblankeb41d80a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/ecbaker7-tamu/song-lab/tree/eb41d80e797e0c21e97cf8322df06d5a6cb56aed" target="_blank">eb41d80</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomecbaker7tamusonglabtreeeb41d80e797e0c21e97cf8322df06d5a6cb56aedtargetblankeb41d80a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/ecbaker7-tamu/song-lab/tree/eb41d80e797e0c21e97cf8322df06d5a6cb56aed" target="_blank">eb41d80</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store

Unstaged changes:
    Modified:   analysis/rep-model-mask.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/rep-model-mask.Rmd</code>) and
HTML (<code>docs/rep-model-mask.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/4e146c4576f9c498f75eb81aa9193b039c9d4fcc/docs/rep-model-mask.html" target="_blank">4e146c4</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-07-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/bf7765b893698fba805f5c3263c00881f04c8413/analysis/rep-model-mask.Rmd" target="_blank">bf7765b</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-07-02
</td>
<td>
Updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/bf7765b893698fba805f5c3263c00881f04c8413/docs/rep-model-mask.html" target="_blank">bf7765b</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-07-02
</td>
<td>
Updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/7a5615778c5c2cf842b2ae1b8c5969112f4bef73/docs/rep-model-mask.html" target="_blank">7a56157</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-05-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/fecf69695ef32ed4f732693ce7fb0749bedc3a3e/docs/rep-model-mask.html" target="_blank">fecf696</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-05-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/9ed28350316f5760a28dc7c92480d718b24a7c5b/docs/rep-model-mask.html" target="_blank">9ed2835</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-05-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/05539da9915300f741c6fda620038213ded63e5b/docs/rep-model-mask.html" target="_blank">05539da</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-05-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/ad2ac0606d6ada9812f82d214d88171a2d69673a/docs/rep-model-mask.html" target="_blank">ad2ac06</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-04-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/729de9688ab12a7636a75643ec35e40f91cadddd/docs/rep-model-mask.html" target="_blank">729de96</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-04-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/63d1ad06520e7f288ebaa6562d63fe2865527bf7/docs/rep-model-mask.html" target="_blank">63d1ad0</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/aaa78c46da9d439cbb501e9591ecc971e6283817/docs/rep-model-mask.html" target="_blank">aaa78c4</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-01-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/700616cffba85a3aa3e292f5a0d80a006e4a5aba/docs/rep-model-mask.html" target="_blank">700616c</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/540436d314b73393e099b0ed4024078893e26457/analysis/rep-model-mask.Rmd" target="_blank">540436d</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-20
</td>
<td>
wflow_publish("analysis/rep-model-mask.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/1a2ce204ff7f170edf2a69889c66d330dce184fa/docs/rep-model-mask.html" target="_blank">1a2ce20</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/155bc26e3225dad6f2286464187f059ca5fe5421/analysis/rep-model-mask.Rmd" target="_blank">155bc26</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-20
</td>
<td>
Update rep-model-mask.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/3e09acdaaf90fe1a4f256f990c96c0703bcf8947/analysis/rep-model-mask.Rmd" target="_blank">3e09acd</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-20
</td>
<td>
Repeat Modeler &amp; Masker page update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/3e09acdaaf90fe1a4f256f990c96c0703bcf8947/docs/rep-model-mask.html" target="_blank">3e09acd</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-20
</td>
<td>
Repeat Modeler &amp; Masker page update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/840f400efc8a7a06882560c830e0220c8bbff86e/analysis/rep-model-mask.Rmd" target="_blank">840f400</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-17
</td>
<td>
Forgot a slash
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/70e69bb9cff81733b4f7c6ccb473bd09f1afdbd5/analysis/rep-model-mask.Rmd" target="_blank">70e69bb</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-17
</td>
<td>
Fix Hyperlink
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/8be8cf9a1b07455989a9e425c7cf07d73cc78ef0/docs/rep-model-mask.html" target="_blank">8be8cf9</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/4f8a5c04aa5abc4f145f3016c7c973651361f755/analysis/rep-model-mask.Rmd" target="_blank">4f8a5c0</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-17
</td>
<td>
wflow_publish("analysis/rep-model-mask.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/77c736c79cc4d22a792a2e46f75ee9ab4068ec30/docs/rep-model-mask.html" target="_blank">77c736c</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/0ad9cd4c1a156afc748344653cfd633406b66204/analysis/rep-model-mask.Rmd" target="_blank">0ad9cd4</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-17
</td>
<td>
Trying to fix file download
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/0ad9cd4c1a156afc748344653cfd633406b66204/docs/rep-model-mask.html" target="_blank">0ad9cd4</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-17
</td>
<td>
Trying to fix file download
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/a9233f82b9e3a659c6612c7f6f8279fb2e79e21d/docs/rep-model-mask.html" target="_blank">a9233f8</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/e05dc1e7a4db86c6623bffe2a768bd6066d68ecb/analysis/rep-model-mask.Rmd" target="_blank">e05dc1e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-16
</td>
<td>
wflow_publish(c("analysis/augustus.Rmd", "analysis/barkdull.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="identifying-and-processing-repeat-content"
class="section level2">
<h2>Identifying and Processing Repeat Content</h2>
<!-- ## What is RepeatModeler & RepeatMasker? -->
</div>
<div id="script" class="section level2">
<h2>Script</h2>
<p>To run both Repeat Modeler and Masker, we used the following script
created by Amanda Stahlke at Colorado Mesa University, modified to run
on the TAMU Grace cluster.</p>
<pre class="r"><code>#!/bin/bash
#SBATCH --job-name=Repeat-SPECIES
#SBATCH --time=7-00:00:00
#SBATCH --nodes=1          # max 64 nodes for partition long
#SBATCH --ntasks-per-node=1
#SBATCH --cpus-per-task=48
#SBATCH --mem=360G
#SBATCH --mail-type=ALL
#SBATCH --mail-user=user@mail.com
#SBATCH --output=stdout.%x.%j
#SBATCH --error=stderr.%x.%j

ASM={Genome File Suffix}_genomic.fa
NM={Genome Name}

module purge
module load GCC/12.3.0 OpenMPI/4.1.5 RepeatModeler/2.0.6

printf &quot;Built repeatmodler2 database on $ASM to build de novo custom library called $NM \n&quot;
BuildDatabase -name $NM $ASM 
printf &quot;Ran repeatmodler2 with LTRStruct on $ASM for  de novo custom library called $NM \n&quot;
RepeatModeler -threads 40 -LTRStruct -database $NM 

module purge
module load GCC/11.3.0 OpenMPI/4.1.4 RepeatMasker/4.1.4

printf &quot;Running RepeatMasker on $ASM&quot;
RepeatMasker -pa 40 -a -gff -lib ${NM}-families.fa -dir denovo2 $ASM</code></pre>
<p>Please note the use of the <em>-a</em> flag, which is used to
generate an alignment file during the masking process. This is important
for generating the Kimura plots mentioned later on this page</p>
</div>
<div id="downstream-analysis" class="section level2">
<h2>Downstream Analysis</h2>
<div id="visualizing-repeat-content" class="section level3">
<h3>Visualizing Repeat Content</h3>
<div id="for-a-single-genome" class="section level4">
<h4>For a single genome</h4>
<p><img src="assets/SchCanc_Kimura.png" /></p>
<p>To generate a kimura repeat content plot like the one above, first
you must run the script included in the RepeatMasker module called
<em>calcDivergenceFromAlign.pl</em>. I did this using the script below
on the TAMU Grace Cluster</p>
<pre class="r"><code>#!/bin/bash
#SBATCH --job-name=Generate-Dist
#SBATCH --time=03:00:00
#SBATCH --nodes=1          # max 64 nodes for partition long
#SBATCH --ntasks-per-node=1
#SBATCH --cpus-per-task=48
#SBATCH --mem=360G
#SBATCH --mail-type=ALL
#SBATCH --mail-user=user@mail.com
#SBATCH --output=stdout.%x.%j
#SBATCH --error=stderr.%x.%j

module purge
module load GCC/11.3.0 OpenMPI/4.1.4 RepeatMasker/4.1.4

/sw/eb/sw/RepeatMasker/4.1.4-foss-2022a/util/calcDivergenceFromAlign.pl -s {SpeciesName}.distance {SpeciesName}_genomic.fa.align</code></pre>
<p>This will generate a file with a <em>.distance</em> suffix. I find it
easier to just download this file and generate the figure locally, as we
will need to edit this file as well. When you open your
<em>species.distance</em> file you should see something like in the
block below:</p>
<pre class="r numberLines"><code>Jukes/Cantor and Kimura subsitution levels adjusted for CpG sites
=================================================================
File: GCF_023864275.1_iqSchCanc2.1_genomic.fa.align
Weighted average Kimura divergence for each repeat family
Class   Repeat  absLen  wellCharLen Kimura%
-----   ------  ------  ----------- -------
DNA rnd-4_family-3901   2850838 2828840 9.37
DNA rnd-5_family-7626   384874  375022  18.86
DNA/Academ-1    rnd-5_family-132    136309  134796  21.75
⋮
LINE    rnd-4_family-267    2208155 2149395 32.22
LINE    rnd-4_family-268    243936  240416  20.67
LINE    rnd-5_family-1066   674334  658113  22.96
⋮
LTR/Copia   rnd-5_family-1055   766978  760738  15.83
LTR/Copia   rnd-5_family-1977   149895  146399  25.27
⋮
RC/Helitron rnd-4_family-1433   1591130 1508796 17.81
RC/Helitron rnd-4_family-1977   1352003 1334534 17.66
⋮
Simple_repeat   combined    45799202    45402746    ----
Unknown rnd-1_family-1  141049217   140819533   5.19
⋮
tRNA    rnd-5_family-26180  476868  476755  3.98
tRNA    rnd-5_family-5841   2251189 2236163 10.50


Coverage for each repeat class and divergence (Kimura)
Div DNA DNA/Academ-1 ... LINE LINE/CR1 ... LTR/Copia ... RC/Helitron Simple_repeat Unknown tRNA
0 290 2240 223288 ... 214984 707342 328 ... 6524  711 0 56168370 340641 
⋮</code></pre>
<p>But much, much larger. Copy and paste from where it starts with
“Div”, below the header for the Kimura scores. Create a new file named
{SpeciesSuffix}.distance and copy and paste the text to this file. It
should look like this:</p>
<pre class="r"><code>Div DNA DNA/Academ-1 ... LINE LINE/CR1 ... LTR/Copia ... RC/Helitron Simple_repeat Unknown tRNA
0 290 2240 223288 ... 214984 707342 328 ... 6524  711 0 56168370 340641 
⋮</code></pre>
<p>Now we’re ready to create our Kimura plot. Create a new R file called
<em>kimura-plot-single-genome.r</em> and copy and paste the following
code into it:</p>
<pre class="r"><code># ------------------------------------------------------------------------------------------------ #
# Repeat annotation summary
#
# Simple figure to summarise repeat families within each genome assembly

# ------------------------------------------------------------------------------------------------ #
# Libraries
suppressPackageStartupMessages({
  library(tidyverse)
  library(magrittr)
  library(here)
  library(scales)
  library(ragg)
  library(readr)
})

# ------------------------------------------------------------------------------------------------ #

# ------------------- Edit for your species ------------------------------------------------- #

size = 8500000000 # Total genome size
sample_name = &quot;Schistocerca cancellata&quot;
file &lt;- &quot;song_assembly/SchCanc.distance&quot;

# ------------------------------------------------------------------------------------------------ #

genome_sizes &lt;- tibble(sample = sample_name, size = size)
pal &lt;- colorRampPalette(RColorBrewer::brewer.pal(n = 7, &#39;Set3&#39;)) # Set color pallette
header &lt;- read_lines(file=file, n_max=1)

df.kimura &lt;- read_table(file, col_types = cols(.default = col_double())) # Read in data (1 species)
df.kimura &lt;- df.kimura[, -ncol(df.kimura)] %&gt;% # Remove last column created from whitespace on EOL
  mutate(sample = sample_name) %&gt;% # Add in col for sample name, unnecessary?
  left_join(genome_sizes, by = &quot;sample&quot;) %&gt;% # Join df.kimura with genome.sizes by sample
   pivot_longer(cols = -c(sample, size, Div), names_to = &quot;repeat_type&quot;, values_to = &quot;value&quot;)  %&gt;%# Make the width of the column the height
   separate(col = repeat_type, into = c(&quot;Family&quot;, &quot;Sub-family&quot;), sep = &quot;/&quot;)  %&gt;% # Seperate Family and Sub-family by &quot;/&quot;
   mutate(`Sub-family` = ifelse(is.na(`Sub-family`), Family, `Sub-family`)) %&gt;% # If sub-family empty, put Family name
  mutate(
    percentage = (value/size) * 100 # calculate percentage composition of genome
    ) %&gt;%
    mutate(`Sub-family` = ifelse(is.na(`Sub-family`), Family, `Sub-family`))

tmp &lt;- df.kimura |&gt;
  filter(!is.na(value), !str_detect(Family, &#39;\\?&#39;)) |&gt;
  group_by(sample, Div, Family) |&gt;
  mutate(total_family = sum(value), percentage_family = (total_family/size) * 100) |&gt;
  select(sample, Div, Family, percentage_family) |&gt;
  ungroup() |&gt;
  distinct()

tmp_filtered &lt;- tmp |&gt; filter(percentage_family &gt; 0.01)
 # Plotting
kimura.gg &lt;- ggplot(tmp_filtered, aes(x = Div, y = percentage_family, fill = Family)) +
  geom_bar(position = &quot;stack&quot;, stat = &quot;identity&quot;, alpha = 0.6) +
  scale_fill_manual( # associate colors with family names
    values = c(
      DNA = &#39;#4eaf49&#39;,
      LINE = &#39;#984f9f&#39;,
      LTR = &#39;#397fb9&#39;,
      MITE = &#39;#f7941d&#39;,
      Unknown = &#39;#737e76&#39;
    ),
    labels = c(
      DNA = &quot;DNA&quot;,
      LINE = &quot;LINEs&quot;,
      LTR = &quot;LTRs&quot;,
      MITE = &quot;MITEs&quot;,
      Unknown = &quot;Unknown Repeats&quot;
    )  ) +
  labs(
    x = &#39;\nKimura substitution level&#39;,
    y = &#39;Genome percentage\n&#39;,
    fill = &#39;\n&#39;
  ) +
  scale_y_continuous(labels = scales::label_number(suffix = &#39;%&#39;)) +
  scale_x_continuous(
    breaks = seq(0, 50, 5),
    limits = c(0, 55),
    expand = c(0, 0)
  ) +
  coord_cartesian(xlim = c(0, 51)) +
  theme_minimal() +
  theme(
    axis.title = element_text(size = 20, face = &#39;bold&#39;),
    axis.text = element_text(size = 18),
    legend.position = &#39;right&#39;,
    legend.title = element_text(size = 18),
    legend.text = element_text(size = 18),
    strip.text = element_text(size = 20, face = &#39;bold.italic&#39;)
  ) +
  facet_wrap(~sample, nrow = 4)

# Save plot
pdf(
  file = paste0(sample_name, &quot;-kimura.pdf&quot;),
  width = 12,
  height = 9
)
print(kimura.gg)
invisible(dev.off())</code></pre>
</div>
<div id="for-multiple-genomes-stacked" class="section level4">
<h4>For multiple genomes (stacked)</h4>
<p><img src="assets/stacked-kimura.png" /></p>
<p>Follow the same procedure for each <em>.distance</em> file as
described in the single-genome process. The only difference in this
procedure is the script we will run to create the Kimura plots. Create a
new R script called <em>kimura-plot-mult-genomes.R</em> and copy and
paste the following script into it:</p>
<pre class="r"><code># ------------------------------------------------------------------------------------------------ #
# Repeat annotation summary
#
# Simple figure to summarise repeat families within each genome assembly

# ------------------------------------------------------------------------------------------------ #
# Librariesx
suppressPackageStartupMessages({
  library(tidyverse)
  library(magrittr)
  library(here)
  library(scales)
  library(ragg)
})

# ------------------------------------------------ EDIT HERE ---------------------------------------------------- #
# genome sizes
genome.sizes &lt;- tibble(
  sample = c(&#39;Schistocerca piceifrons&#39;, &#39;Schistocerca cancellata&#39;, &#39;Schistocerca nitens&#39;, &#39;Schistocerca serialis cubense&#39;, &#39;Anabrus simplex - Idaho&#39;, &#39;Anabrus simplex - Colorado&#39;),
  size = c(8700000000, 8500000000, 8800000000, 9100000000, 6400000000, 6900000000)
)

pal &lt;- colorRampPalette(RColorBrewer::brewer.pal(n = 7, &#39;Set3&#39;))

# ---------------------------------------------------------------------------------------------------------------- #
# Kimura divergence tables
df.kimura &lt;- fs::dir_ls(
  path = &#39;song_assembly&#39;, # &lt;----------------------------- EDIT HERE
  glob = &#39;*distance&#39;,
  recurse = TRUE
) |&gt;
  map(read_delim, delim = &#39; &#39;, col_names = TRUE, col_types = cols()) |&gt;
  list_rbind(names_to = &#39;sample&#39;) |&gt;
  mutate( # ------------------------------------------------ EDIT HERE -------------------------------------------- #
    sample = case_when(
      sample == &#39;song_assembly/SchPice.distance&#39; ~ &#39;Schistocerca piceifrons&#39;,
      sample == &#39;song_assembly/SchCanc.distance&#39; ~ &#39;Schistocera cancellata&#39;,
      sample == &#39;song_assembly/SchNite.distance&#39; ~ &#39;Schistocerca nitens&#39;,
      sample == &#39;song_assembly/SchSeri.distance&#39; ~ &#39;Schistocerca serialis cubense&#39;,
      sample == &#39;song_assembly/AnsimpID.distance&#39; ~ &#39;Anabrus simplex - Idaho&#39;,
      sample == &#39;song_assembly/AnsimpCO.distance&#39; ~ &#39;Anabrus simplex - Colorado&#39;,
      .default = sample
    )
  ) |&gt;
  select(-starts_with(&#39;...&#39;)) |&gt;
  (\(x) pivot_longer(data = x, names_to = &#39;repeat_type&#39;, values_to = &#39;value&#39;, cols = 3:ncol(x)))() |&gt;
  left_join(genome.sizes, by = &quot;sample&quot;)|&gt;
  mutate(
    percentage = (value/size) * 100,
    sample = factor(sample, levels = c(&#39;Schistocerca piceifrons&#39;, &#39;Schistocerca cancellata&#39;, &#39;Schistocerca nitens&#39;, &#39;Schistocerca serialis cubense&#39;, &#39;Anabrus simplex - Idaho&#39;, &#39;Anabrus simplex - Colorado&#39;))
    ) |&gt; # -------------------------------------------------------------------------------------------------------- #
  separate(col = repeat_type, into = c(&#39;Family&#39;, &#39;Sub-family&#39;), sep = &#39;/&#39;) |&gt;
  mutate(`Sub-family` = ifelse(is.na(`Sub-family`), Family, `Sub-family`))

# Temporary while I re-run repeat masker with repbase
tmp &lt;- df.kimura |&gt;
  filter(!is.na(value), !str_detect(Family, &#39;\\?&#39;)) |&gt;
  group_by(sample, Div, Family) |&gt;
  mutate(total_family = sum(value), percentage_family = (total_family/size) * 100) |&gt;
  select(sample, Div, Family, percentage_family) |&gt;
  ungroup() |&gt;
  distinct()

# ------------------------------------------------------------------------------------------------ #
# Plot divergence plots
kimura.gg &lt;- tmp |&gt;
  filter(percentage_family &gt; 0.01) |&gt;
  ggplot(aes(x = Div, y = percentage_family, fill = Family)) +
  geom_bar(position = &quot;stack&quot;, stat = &quot;identity&quot;, alpha = 0.6) +
  scale_fill_manual( # associate colors with family names
    values = c(
      DNA = &#39;#4eaf49&#39;,
      LINE = &#39;#984f9f&#39;,
      LTR = &#39;#397fb9&#39;,
      MITE = &#39;#f7941d&#39;,
      Unknown = &#39;#737e76&#39;
    ),
    labels = c(
      DNA = &quot;DNA&quot;,
      LINE = &quot;LINEs&quot;,
      LTR = &quot;LTRs&quot;,
      MITE = &quot;MITEs&quot;,
      Unknown = &quot;Unknown Repeats&quot;
    )  ) +
  labs(
    x = &#39;\nKimura substitution level&#39;,
    y = &#39;Genome percentage\n&#39;,
    fill = &#39;\n&#39;
  ) +
  scale_y_continuous(labels = label_number(suffix = &#39;%&#39;)) +
  scale_x_continuous(
    breaks = seq(0, 50, 5),
    limits = c(0, 55),
    expand = c(0,0)
  ) +
  coord_cartesian(xlim = c(0, 51)) +
  theme_minimal() +
  theme(
    axis.title = element_text(size = 20, face = &#39;bold&#39;),
    axis.text = element_text(size = 18),
    legend.position = &#39;bottom&#39;,
    legend.title = element_text(size = 18),
    legend.text = element_text(size = 18),
    strip.text = element_text(size = 20, face = &#39;bold.italic&#39;)
  ) +
  facet_wrap(~sample, nrow = 4)

# Save PNG
pdf(
  file = here(&#39;multigenome-kimura.pdf&#39;),
  width = 20,
  height = 10
)
print(kimura.gg)
invisible(dev.off())</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] vctrs_0.6.5       cli_3.6.4         knitr_1.50        rlang_1.1.6      
 [5] xfun_0.52         stringi_1.8.7     promises_1.3.2    jsonlite_2.0.0   
 [9] workflowr_1.7.1   glue_1.8.0        rprojroot_2.0.4   git2r_0.36.2     
[13] htmltools_0.5.8.1 httpuv_1.6.16     sass_0.4.10       rmarkdown_2.29   
[17] evaluate_1.0.3    jquerylib_0.1.4   tibble_3.2.1      fastmap_1.2.0    
[21] yaml_2.3.10       lifecycle_1.0.4   whisker_0.4.1     stringr_1.5.1    
[25] compiler_4.4.2    fs_1.6.6          Rcpp_1.0.14       pkgconfig_2.0.3  
[29] rstudioapi_0.17.1 later_1.4.2       digest_0.6.37     R6_2.6.1         
[33] pillar_1.10.2     magrittr_2.0.3    bslib_0.9.0       tools_4.4.2      
[37] cachem_1.1.0     </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
