<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Repeat Modeler and Masker</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="theme3.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Home</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-note-sticky"></span>
     
    Notes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">June</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="week1-notes.html">6/4 - 6/7</a>
        </li>
        <li>
          <a href="week2-notes.html">6/10 - 6/14</a>
        </li>
        <li>
          <a href="week3-notes.html">6/17 - 6/21</a>
        </li>
        <li>
          <a href="week4-notes.html">6/24 - 6/28</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">July</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="week5-notes.html">6/31 - 7/5</a>
        </li>
        <li>
          <a href="week6-notes.html">7/8 - 7/12</a>
        </li>
        <li>
          <a href="week7-notes.html">7/15 - 7/19</a>
        </li>
        <li>
          <a href="week9-notes.html">7/29 - 8/2</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">August</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="week12-notes.html">8/19 - 8/30</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">September</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="week14-notes.html">9/9 - 9/20</a>
        </li>
        <li>
          <a href="week16-notes.html">9/30 - 10/4</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-microscope"></span>
     
    Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="orthop-genomes-project.html">Comparative Genomics of Schistocerca gregaria</a>
    </li>
    <li class="dropdown-header">Comparative Genomics of Mormon Cricket subgroups -- UNDER CONSTRUCTION</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-code"></span>
     
    Program Information
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">TOGA &amp; Related Pipelines</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="toga.html">TOGA</a>
        </li>
        <li>
          <a href="mlc.html">make_lastz_chains</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="genespace.html">GENESPACE</a>
    </li>
    <li>
      <a href="compare-genomes.html">Compare_Genomes</a>
    </li>
    <li>
      <a href="barkdull.html">FormicidaeMolecularEvolution (OrthoFinder)</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Annotation Tools</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="Apollo.html">Apollo</a>
        </li>
        <li>
          <a href="augustus.html">Augustus</a>
        </li>
        <li>
          <a href="braker.html">Braker</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="rep-model-mask.html">RepeatModeler &amp; RepeatMasker</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-book-open-reader"></span>
     
    Reference
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="definitions.html">Glossary of Terminology</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Repeat Modeler and Masker</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-04-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>song-lab/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240605code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240605)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240605code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240605)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomecbaker7tamusonglabtreea5e996b567a8526569a17d2dfdb2a6c20e6ae56atargetblanka5e996ba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/ecbaker7-tamu/song-lab/tree/a5e996b567a8526569a17d2dfdb2a6c20e6ae56a" target="_blank">a5e996b</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomecbaker7tamusonglabtreea5e996b567a8526569a17d2dfdb2a6c20e6ae56atargetblanka5e996ba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/ecbaker7-tamu/song-lab/tree/a5e996b567a8526569a17d2dfdb2a6c20e6ae56a" target="_blank">a5e996b</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/rep-model-mask.Rmd</code>) and
HTML (<code>docs/rep-model-mask.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/aaa78c46da9d439cbb501e9591ecc971e6283817/docs/rep-model-mask.html" target="_blank">aaa78c4</a>
</td>
<td>
ecbaker7-tamu
</td>
<td>
2025-01-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/700616cffba85a3aa3e292f5a0d80a006e4a5aba/docs/rep-model-mask.html" target="_blank">700616c</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/540436d314b73393e099b0ed4024078893e26457/analysis/rep-model-mask.Rmd" target="_blank">540436d</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-20
</td>
<td>
wflow_publish("analysis/rep-model-mask.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/1a2ce204ff7f170edf2a69889c66d330dce184fa/docs/rep-model-mask.html" target="_blank">1a2ce20</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/155bc26e3225dad6f2286464187f059ca5fe5421/analysis/rep-model-mask.Rmd" target="_blank">155bc26</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-20
</td>
<td>
Update rep-model-mask.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/3e09acdaaf90fe1a4f256f990c96c0703bcf8947/analysis/rep-model-mask.Rmd" target="_blank">3e09acd</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-20
</td>
<td>
Repeat Modeler &amp; Masker page update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/3e09acdaaf90fe1a4f256f990c96c0703bcf8947/docs/rep-model-mask.html" target="_blank">3e09acd</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-20
</td>
<td>
Repeat Modeler &amp; Masker page update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/840f400efc8a7a06882560c830e0220c8bbff86e/analysis/rep-model-mask.Rmd" target="_blank">840f400</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-17
</td>
<td>
Forgot a slash
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/70e69bb9cff81733b4f7c6ccb473bd09f1afdbd5/analysis/rep-model-mask.Rmd" target="_blank">70e69bb</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-17
</td>
<td>
Fix Hyperlink
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/8be8cf9a1b07455989a9e425c7cf07d73cc78ef0/docs/rep-model-mask.html" target="_blank">8be8cf9</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/4f8a5c04aa5abc4f145f3016c7c973651361f755/analysis/rep-model-mask.Rmd" target="_blank">4f8a5c0</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-17
</td>
<td>
wflow_publish("analysis/rep-model-mask.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/77c736c79cc4d22a792a2e46f75ee9ab4068ec30/docs/rep-model-mask.html" target="_blank">77c736c</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/0ad9cd4c1a156afc748344653cfd633406b66204/analysis/rep-model-mask.Rmd" target="_blank">0ad9cd4</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-17
</td>
<td>
Trying to fix file download
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/0ad9cd4c1a156afc748344653cfd633406b66204/docs/rep-model-mask.html" target="_blank">0ad9cd4</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-17
</td>
<td>
Trying to fix file download
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/a9233f82b9e3a659c6612c7f6f8279fb2e79e21d/docs/rep-model-mask.html" target="_blank">a9233f8</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/e05dc1e7a4db86c6623bffe2a768bd6066d68ecb/analysis/rep-model-mask.Rmd" target="_blank">e05dc1e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-12-16
</td>
<td>
wflow_publish(c("analysis/augustus.Rmd", "analysis/barkdull.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<!-- ## What is RepeatModeler & RepeatMasker? -->
<div id="step-0-file-directory-creation" class="section level2">
<h2>Step 0: File &amp; Directory Creation</h2>
<div id="directories" class="section level3">
<h3>Directories</h3>
<p>Now we must create the necessary directories for this pipeline to
function.</p>
<div id="working-directory" class="section level4">
<h4>Working Directory</h4>
<p>Create a new directory in your scratch folder (for example, you could
put it in the following file path:
scratch/user/YOUR_USERNAME/PROJECT_NAME). Call this new directory
<em>rep-model-mask</em>. This is where you will add in the files we will
create later.</p>
</div>
<div id="genome-directory" class="section level4">
<h4>Genome Directory</h4>
<p>Create a directory in your scratch directory called <em>genomes</em>.
This is where we will store our data for the repeat modeler and repeat
masker runs.</p>
</div>
</div>
<div id="files" class="section level3">
<h3>Files</h3>
<p>The main problem with the currently available RepeatModeler and
RepeatMasker software is that they create thousands of large
“intermediary” files that do not get deleted once the pipeline is
finished running. Devon J. Boland from the Texas A&amp;M Institute of
Genome Sciences and Society (TIGSS) developed this SnakeMake pipeline
that automatically deletes these files, freeing up storage after the
process has completed its run. To run this pipeline you will need to
create the following 3 files:</p>
<div id="i-snakefile" class="section level4">
<h4>i) Snakefile</h4>
<p>Create the file “Snakefile” in your rep-model-mask directory and copy
and paste the below code into it.<mark> You will need to change lines 8
and 9 to the file paths of the work and genome folders you created in
your scratch directory (see Step 0: Directories).</mark></p>
<pre><code>### =================================================================
### GENOME REPEAT MASKING PIPELINE (v. DECEMBER 2024)
### Project: Comparative Genomics Of Locusts
### Devon J. Boland, &lt;devonjboland@tamu.edu&gt;
### =================================================================

### SET DIRECTORY PATHS FOR REFERENCE AND OUTPUT DATA
WORKDir = &quot;/scratch/user/devonjboland/BPRI/Genomes/RepeatMasking&quot;
GENOMEDir = &quot;/scratch/user/devonjboland/BPRI/Genomes&quot;

# For local testing
#WORKDir = &quot;/Users/devonjboland/Documents/TIGSS/Maeva_TECHER_SongLab/Repeat_Masking/SnakeMake&quot;
#GENOMEDir = &quot;/Users/devonjboland/Documents/TIGSS/Maeva_TECHER_SongLab/Repeat_Masking/SnakeMake/Genomes&quot;

### SET SAMPLES LIST
INSECTS, = glob_wildcards(GENOMEDir + &quot;/{insect}.fa&quot;)

### SET PATHS FOR STATIC FILES
TRFSif = &quot;/scratch/user/devonjboland/SIFs/trf_4.09.1--h031d066_4.sif&quot;

# SET PSEUDO-RULE

rule all:
        input:
                ###   GENERATING TRANSCRIPT MAPPING AND READS COUNTS
                expand(WORKDir + &quot;/{insect}/{insect}.fa.2.7.7.80.10.50.500.mask.masked&quot;, insect = INSECTS)

### =================================================================
###  Build TE Family Database Using RepeatModeler
### =================================================================

rule repeat_modeler:
        input:
                genome = GENOMEDir + &quot;/{insect}.fa&quot;
        threads: 48
        params:
                genome_loc = GENOMEDir,
                work_dir = WORKDir + &quot;/{insect}&quot;,
                short_name = &quot;{insect}&quot;
        output:
                rm_lib = WORKDir + &quot;/{insect}/{insect}-families.fa&quot;
        shell:
                &quot;&quot;&quot;
                set -euxo pipefail
                ml purge
                ml GCC/12.3.0  OpenMPI/4.1.5 RepeatModeler/2.0.6

                cp {input.genome} $TMPDIR
                cd $TMPDIR

                # Create Blastn database of genome
                BuildDatabase -name {params.short_name} {params.short_name}.fa

                # Execute Repeat Modeler
                RepeatModeler -database {params.short_name} -threads {threads} -LTRStruct
                mv {params.short_name}* {params.work_dir}
                mv {params.short_name}.fa {params.work_dir}
                cd ${params.work_dir}
                ml purge

                &quot;&quot;&quot;


### =================================================================
###  Identify Tandem Repeats Using TRF
### =================================================================

rule run_trf:
        params:
                genome = WORKDir + &quot;/{insect}/{insect}.fa&quot;,
                work_dir = WORKDir + &quot;/{insect}&quot;,
                sif = TRFSif
        output:
                masked_trf = WORKDir + &quot;/{insect}/{insect}.fa.2.7.7.80.10.50.500.mask&quot;
        shell:
                &quot;&quot;&quot;
                ml purge
                cd {params.work_dir}
                singularity exec {params.sif} trf {params.genome} 2 7 7 80 10 50 500 -f -d -m -h

                &quot;&quot;&quot;

### =================================================================
###  Mask TEs Using RepeatMasker on TRF Output with RM Input
### =================================================================

rule repeat_masker:
        input:
                genome_trf = WORKDir + &quot;/{insect}/{insect}.fa.2.7.7.80.10.50.500.mask&quot;,
                rm_lib = WORKDir + &quot;/{insect}/{insect}-families.fa&quot;
        threads: 48
        output:
                final_mask = WORKDir + &quot;/{insect}/{insect}.fa.2.7.7.80.10.50.500.mask.masked&quot;
        shell:
                &quot;&quot;&quot;
                set -euxo pipefail
                ml purge
                ml GCC/11.3.0  OpenMPI/4.1.4 RepeatMasker/4.1.4

                RepeatMasker -lib {input.rm_lib} --pa ${threads} {input.genome_trf}

                ml purge
                &quot;&quot;&quot;</code></pre>
</div>
<div id="ii-snakefile.slurm" class="section level4">
<h4>ii) snakefile.slurm</h4>
<p>Create the file “snakefile.slurm” in your rep-model-mask directory
and copy and paste the following code into it. This file specifies that
the cluster should send a notification when any updates occurring the
job process (job begin or end, next stage, job canceled, etc.) to the
specified email. <mark> You will need to edit line 10 to email
notifications to the appropriate email address. If you do not wish to
receive the email updates, you can delete lines 9 and 10.</mark></p>
<pre><code>#!/bin/bash
#SBATCH --job-name=snakemake-master
#SBATCH --time=&quot;7-00:00:00&quot;
#SBATCH --mem=100GB
#SBATCH --cpus-per-task=1
#SBATCH --ntasks=1

## OPTIONAL ##
#SBATCH --mail-type=ALL              #Send email on all job events
#SBATCH --mail-user=USEREMAIL@tamu.edu    #Send all emails

ml GCC/12.2.0  OpenMPI/4.1.4 snakemake/7.32.3

snakemake --latency-wait 1000 \
--restart-times 1 \
-j 30 -p \
--max-jobs-per-second 1 \
--cluster-config cluster.json \
--cluster &quot;sbatch -p {cluster.partition} --ntasks {cluster.ntasks} --cpus-per-task {cluster.cpus-per-task} -t {cluster.time} --mem {cluster.mem} --output {cluster.output} --error {cluster.error}&quot; \
--rerun-incomplete \
--notemp \
--nolock</code></pre>
</div>
<div id="iii-cluster.json" class="section level4">
<h4>iii) cluster.json</h4>
<p>Create a file called “cluster.json” in your rep-model-mask directory
and copy and paste the following code into it. This file provides
specifications for how the cluster should allocate memory and name the
output files based on what task is being run. <mark> You do not need to
edit any lines of code in this file prior to submitting the job.
</mark></p>
<pre><code>{
&quot;__default__&quot; :
{
    &quot;cpus-per-task&quot; : 12,
    &quot;partition&quot; : &quot;medium&quot;,
    &quot;ntasks&quot;: 1,
    &quot;mem&quot; : &quot;40G&quot;,
    &quot;time&quot;: &quot;1-00:00:00&quot;,
    &quot;out&quot;: &quot;%j.out.txt&quot;
},
&quot;repeat_modeler&quot;:
{
    &quot;cpus-per-task&quot; : 48,
    &quot;partition&quot; : &quot;long&quot;,
    &quot;ntasks&quot;: 1,
    &quot;mem&quot; : &quot;320G&quot;,
    &quot;time&quot;: &quot;7-00:00:00&quot;,
    &quot;output&quot;: &quot;repeat_modeler.%j.stdout.txt&quot;,
    &quot;error&quot;: &quot;repeat_modeler.%j.stderr.txt&quot;
},
&quot;run_trf&quot;:
{
    &quot;cpus-per-task&quot; : 48,
    &quot;partition&quot; : &quot;long&quot;,
    &quot;ntasks&quot;: 1,
    &quot;mem&quot; : &quot;320G&quot;,
    &quot;time&quot;: &quot;3-00:00:00&quot;,
    &quot;output&quot;: &quot;run_trf.%j.stdout.txt&quot;,
    &quot;error&quot;: &quot;run_trf.%j.stderr.txt&quot;
},
&quot;repeat_masker&quot;:
{
    &quot;cpus-per-task&quot; : 48,
    &quot;partition&quot; : &quot;long&quot;,
    &quot;ntasks&quot;: 1,
    &quot;mem&quot; : &quot;320G&quot;,
    &quot;time&quot;: &quot;7-00:00:00&quot;,
    &quot;output&quot;: &quot;repeat_masker.%j.stdout.txt&quot;,
    &quot;error&quot;: &quot;repeat_masker.%j.stderr.txt&quot;
}
}</code></pre>
</div>
</div>
</div>
<div id="step-1-repeat-modeler" class="section level2">
<h2>Step 1: Repeat Modeler</h2>
<p>First we will run RepeatModeler on our genome of choice using the
following code to submit the job to the cluster</p>
<pre class="r"><code>sbatch snakemake.slurm repeat_modeler</code></pre>
<p>This job will launch jobs to run in parallel, so don’t be alarmed if
you see more jobs running than you’ve submitted.</p>
</div>
<div id="step-2-repeat-masker" class="section level2">
<h2>Step 2: Repeat Masker</h2>
<p>Run RepeatMasker using the command</p>
<pre class="r"><code>sbatch snakemake.slurm repeat_masker</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.7.1

loaded via a namespace (and not attached):
 [1] vctrs_0.6.5       httr_1.4.7        cli_3.6.4         knitr_1.49       
 [5] rlang_1.1.5       xfun_0.51         stringi_1.8.4     processx_3.8.5   
 [9] promises_1.3.2    jsonlite_1.8.9    glue_1.8.0        rprojroot_2.0.4  
[13] git2r_0.35.0      htmltools_0.5.8.1 httpuv_1.6.15     ps_1.8.1         
[17] sass_0.4.9        rmarkdown_2.29    jquerylib_0.1.4   tibble_3.2.1     
[21] evaluate_1.0.3    fastmap_1.2.0     yaml_2.3.10       lifecycle_1.0.4  
[25] whisker_0.4.1     stringr_1.5.1     compiler_4.4.2    fs_1.6.5         
[29] pkgconfig_2.0.3   Rcpp_1.0.14       rstudioapi_0.17.1 later_1.4.1      
[33] digest_0.6.37     R6_2.6.1          pillar_1.10.1     callr_3.7.6      
[37] magrittr_2.0.3    bslib_0.9.0       tools_4.4.2       cachem_1.1.0     
[41] getPass_0.2-4    </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
