<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Mapping Transcription Factors to Trancription Factor Families</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="theme3.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Home</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-microscope"></span>
     
    Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Comparative Genomics of Schistocerca gregaria -- In prep</li>
    <li class="dropdown-header">Comparative Genomics of Mormon Cricket subgroups -- UNDER CONSTRUCTION</li>
    <li class="dropdown-header">Ensifera Ear Transcriptomics -- UNDER CONSTRUCTION</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-code-compare"></span>
     
    Comparative Genomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">FormicidaeMolecularEvolution - OrthoFinder Workflow for Longest Isoform</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="barkdull.html">TAMU Workflow</a>
        </li>
        <li>
          <a href="barkdull_asu.html">ASU Workflow</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">RepeatModeler &amp; RepeatMasker</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="rep-model-mask.html">TAMU Workflow</a>
        </li>
        <li>
          <a href="rep-model-mask_asu.html">ASU Workflow</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">PSMC Model Analysis</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="psmc.html">TAMU Workflow</a>
        </li>
        <li>
          <a href="psmc_asu.html">ASU Workflow</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-pen-to-square"></span>
     
    Transcriptomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="tf_mapping.html">Transcription Factor Mapping (ASU Only)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-flask"></span>
     
    Proteomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="alphafold.html">AlphaFold</a>
    </li>
    <li>
      <a href="chimerax.html">Chimera X</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-palette"></span>
     
    Visualization
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="kimura.html">Kimura Plots (Repeat Content)</a>
    </li>
    <li>
      <a href="genespace.html">GENESPACE (Synteny)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-book-open-reader"></span>
     
    Reference
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="slurm-comm.html">Linux and Slurm Commands</a>
    </li>
    <li>
      <a href="definitions.html">Glossary of Terminology</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Mapping Transcription Factors to
Trancription Factor Families</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-09-17
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>song-lab/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240605code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240605)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240605code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240605)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomemilycbakersonglabtree2994ca65d04ff8250d18347958000c84656c99bftargetblank2994ca6a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/emily-cbaker/song-lab/tree/2994ca65d04ff8250d18347958000c84656c99bf" target="_blank">2994ca6</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomemilycbakersonglabtree2994ca65d04ff8250d18347958000c84656c99bftargetblank2994ca6a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/emily-cbaker/song-lab/tree/2994ca65d04ff8250d18347958000c84656c99bf" target="_blank">2994ca6</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/tf_mapping.Rmd</code>) and HTML
(<code>docs/tf_mapping.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/emily-cbaker/song-lab/blob/2994ca65d04ff8250d18347958000c84656c99bf/analysis/tf_mapping.Rmd" target="_blank">2994ca6</a>
</td>
<td>
emilyc-baker
</td>
<td>
2025-09-17
</td>
<td>
wflow_publish("analysis/tf_mapping.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/d3e94a612432a603cd0b8afb5555c0d486e1d7fd/docs/tf_mapping.html" target="_blank">d3e94a6</a>
</td>
<td>
emilyc-baker
</td>
<td>
2025-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/emily-cbaker/song-lab/blob/e33c24f2b290d9cb40ea5df3c44fb228dff354a8/analysis/tf_mapping.Rmd" target="_blank">e33c24f</a>
</td>
<td>
emilyc-baker
</td>
<td>
2025-09-17
</td>
<td>
wflow_publish("analysis/tf_mapping.Rmd")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/emily-cbaker/song-lab/blob/c443d10c314d3d9b032358ec31bd7ee21baf7fdb/analysis/tf_mapping.Rmd" target="_blank">c443d10</a>
</td>
<td>
emilyc-baker
</td>
<td>
2025-09-17
</td>
<td>
Update tf_mapping.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/emily-cbaker/song-lab/blob/fbcf41ef2f4462475bfe47acbfaad1613b053017/analysis/tf_mapping.Rmd" target="_blank">fbcf41e</a>
</td>
<td>
emilyc-baker
</td>
<td>
2025-09-17
</td>
<td>
Update tf_mapping.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/692774db9a50424fd7e14aaf6cf69cfeb29bf610/docs/tf_mapping.html" target="_blank">692774d</a>
</td>
<td>
emilyc-baker
</td>
<td>
2025-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/34ebc36e2ab1d999ef282828831b55bf62e1a904/docs/tf_mapping.html" target="_blank">34ebc36</a>
</td>
<td>
emilyc-baker
</td>
<td>
2025-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/9d1da312b33d8122a4ca690faf6cfe56b42452d2/docs/tf_mapping.html" target="_blank">9d1da31</a>
</td>
<td>
emilyc-baker
</td>
<td>
2025-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/5f7fdae208ce998d6a01e42eb395a331295a8a63/docs/tf_mapping.html" target="_blank">5f7fdae</a>
</td>
<td>
emilyc-baker
</td>
<td>
2025-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/emily-cbaker/song-lab/blob/36abff94147c225664d99cf5ce238f07b4dc794a/analysis/tf_mapping.Rmd" target="_blank">36abff9</a>
</td>
<td>
emilyc-baker
</td>
<td>
2025-09-17
</td>
<td>
wflow_publish(c("analysis/alphafold.Rmd", "analysis/Apollo.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="mapping-transcription-factors" class="section level2 tabset">
<h2 class="tabset">Mapping Transcription Factors</h2>
<div id="running-on-a-cluster" class="section level3">
<h3>Running on a Cluster</h3>
<center>
<mark> This process is configured to run on the ASU Sols cluster.
</mark>
</center>
<p>To conduct this process we will be using HMMER to map our
transcription factors against the Pfam database. Additionally, this
tutorial will be done the proteins of <em>Schistocerca piceifrons</em>.
The protein data is available from NCBI <a
href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/021/461/385/GCF_021461385.2_iqSchPice1.1/">here</a></p>
<div id="step-1-download-and-prepare-pfam" class="section level4">
<h4>Step 1: Download and Prepare Pfam</h4>
<pre class="r"><code># i) Get the latest Pfam-A HMMs
wget ftp://ftp.ebi.ac.uk/pub/databases/Pfam/current_release/Pfam-A.hmm.gz
gunzip Pfam-A.hmm.gz

module load hmmer-3.1b2-gcc-12.1.0 

# ii) Index for hmmscan
hmmpress Pfam-A.hmm

# iii) Download protein data from NCBI, decompress file, and rename
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/021/461/385/GCF_021461385.2_iqSchPice1.1/GCF_021461385.2_iqSchPice1.1_protein.faa.gz

gunzip GCF_021461385.2_iqSchPice1.1_protein.faa.gz

mv GCF_021461385.2_iqSchPice1.1_protein.faa GCF_021461385.2_iqSchPice1.1_protein.fasta</code></pre>
</div>
<div id="step-2-run-hmmscan-on-proteins" class="section level4">
<h4>Step 2: Run hmmscan on proteins</h4>
<p>Create a file called <em>scan-prot.sh</em> and copy and paste the
following code into it:</p>
<pre class="r"><code>#!/bin/bash

##NECESSARY JOB SPECIFICATIONS
#SBATCH --job-name=scan-proteins-serialis       #Set the job name to &quot;psmc-referencegenome&quot;
#SBATCH --time=0-08:00:00         #Set the wall clock limit to 2 hours
#SBATCH --ntasks=1             #Request 1 task
#SBATCH --cpus-per-task=8       #Request 8 cpus
#SBATCH --mem=30G              #Request 30GB per node
#SBATCH --output=stdout.%x.%j
#SBATCH --error=stderr.%x.%j

module load hmmer-3.1b2-gcc-12.1.0 

hmmscan --cpu 4 \
  --domtblout pfam.domtblout \
  ../Pfam-A.hmm \
  GCF_021461385.2_iqSchPice1.1_protein.fasta</code></pre>
<p>This step will create a file called <em>pfam.domtblout</em>. Use the
command ls -lh to check the file is not empty and the step has
successfully run.</p>
</div>
<div id="step-3-parse-and-map-to-transcription-factor-families-in-r"
class="section level4">
<h4>Step 3: Parse and map to transcription factor families in R</h4>
<p>Create a file named parse-map-tf.R and copy and paste the following
code into it:</p>
<pre class="r"><code>library(dplyr)

# 1) Read the domtblout (skip &quot;#&quot; comments)
dom &lt;- read.delim(&quot;pfam.domtblout&quot;,
                  header=FALSE,
                  sep=&quot;&quot;)

# 2) Name the first few columns per HMMER spec
colnames(dom)[1:19] &lt;- c(
  &quot;GeneID&quot;,&quot;Accession&quot;,&quot;Domain&quot;,&quot;EValue&quot;,&quot;Score&quot;,&quot;Bias&quot;,
  paste0(&quot;i&quot;,1:10),
  &quot;EnvStart&quot;,&quot;EnvEnd&quot;,&quot;DomainLength&quot;
)

# 3) Define which Pfam accessions → TF Family
#    (extend this list with all families you care about)
pfam2fam &lt;- tibble(
  Accession = c(&quot;PF22861.1&quot;,&quot;PF00046&quot;,&quot;PF02365&quot;,&quot;PF00249&quot;,&quot;PF04434&quot;),
  Family    = c(&quot;GEO12453p1-like&quot;,       &quot;GATA&quot;,    &quot;NAC&quot;,    &quot;MYB&quot;,    &quot;TCP&quot;)
)

# 4) Filter only hits to those TF domains
tf_hits &lt;- dom %&gt;%
  filter(Accession %in% pfam2fam$Accession) %&gt;%
  distinct(GeneID, Accession)

# 5) Join to get Family names
reg_fam &lt;- tf_hits %&gt;%
  left_join(pfam2fam, by = &quot;Accession&quot;) %&gt;%
  select(GeneID, Family) %&gt;%
  distinct()

# 6) Your regulator vector
regulators &lt;- reg_fam$GeneID

# 7) Quick sanity check
cat(&quot;Found&quot;, nrow(reg_fam), &quot;putative TFs across&quot;, length(unique(reg_fam$Family)), &quot;families\n&quot;)
table(reg_fam$Family)</code></pre>
<p>Then create a file called run-parse-map.sh and copy and paste the
following code into it:</p>
<pre class="r"><code>#!/bin/bash

##NECESSARY JOB SPECIFICATIONS
#SBATCH --job-name=parse-and-map-piceifrons       #Set the job name to &quot;psmc-referencegenome&quot;
#SBATCH --time=02:00:00         #Set the wall clock limit to 2 hours
#SBATCH --ntasks=1             #Request 1 task
#SBATCH --cpus-per-task=8       #Request 8 cpus
#SBATCH --mem=30G              #Request 30GB per node
#SBATCH --output=stdout.%x.%j
#SBATCH --error=stderr.%x.%j

module load r-4.3.2-gcc-11.2.0

Rscript parse-map-tf.R</code></pre>
<p>After this has completed running, use the command <em>cat
stdout.{YOUR_JOB_NUMBER}</em> to check the results. If you followed this
tutorial, the output should read:</p>
<pre class="r"><code>Found 1 putative TFs across 1 families

GEO12453p1-like 
              1 </code></pre>
<p>Don’t worry if you know there’s multiple proteins in this family, it
is only checking for unique protein families not individual
proteins.</p>
</div>
</div>
<div id="troubleshooting" class="section level3">
<h3>Troubleshooting</h3>
<div id="found-0-putative-tfs-across-0-families" class="section level4">
<h4>Found 0 putative TFs across 0 families</h4>
<p>For me, this error occured when no proteins in my pfam2fam matched
proteins in the pfam.domtblout file. To verify this is the issue, use
the command <em>head pfam.domtblout</em> and choose the first protein
listed. Rerun the program and it should say “Found 1 putative TFs across
1 families” along with the protein family listed below this message.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Phoenix
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.7.2

loaded via a namespace (and not attached):
 [1] vctrs_0.6.5       httr_1.4.7        cli_3.6.5         knitr_1.50       
 [5] rlang_1.1.6       xfun_0.53         stringi_1.8.7     processx_3.8.6   
 [9] promises_1.3.3    jsonlite_2.0.0    glue_1.8.0        rprojroot_2.1.1  
[13] git2r_0.36.2      htmltools_0.5.8.1 httpuv_1.6.16     ps_1.9.1         
[17] sass_0.4.10       rmarkdown_2.29    jquerylib_0.1.4   tibble_3.3.0     
[21] evaluate_1.0.5    fastmap_1.2.0     yaml_2.3.10       lifecycle_1.0.4  
[25] whisker_0.4.1     stringr_1.5.2     compiler_4.4.2    fs_1.6.6         
[29] pkgconfig_2.0.3   Rcpp_1.1.0        rstudioapi_0.17.1 later_1.4.4      
[33] digest_0.6.37     R6_2.6.1          pillar_1.11.0     callr_3.7.6      
[37] magrittr_2.0.4    bslib_0.9.0       tools_4.4.2       cachem_1.1.0     
[41] getPass_0.2-4    </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
