<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>TOGA: Tool to Infer Orthologs from Genome Alignments</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="theme3.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Home</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-note-sticky"></span>
     
    Notes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="week1-notes.html">6/4 - 6/7</a>
    </li>
    <li>
      <a href="week2-notes.html">6/10 - 6/14</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-book-open"></span>
     
    Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="orthop-genomes-project.html">Comparative Genomics of Large Orthoptera Genomes</a>
    </li>
    <li>
      <a href="pop-genomics-morm-crick.html">Population Genomics of Mormon Crickets, Central American Locusts, Eremidium Grasshoppers</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-code"></span>
     
    Program Information
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="toga.html">TOGA</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">TOGA: Tool to Infer Orthologs from Genome
Alignments</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-06-13
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>bioinformatics-notes/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240605code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240605)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240605code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240605)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomecbaker7tamusonglabtree50e7daed6663df9172061ada744edf1e4b7e693ctargetblank50e7daea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/ecbaker7-tamu/song-lab/tree/50e7daed6663df9172061ada744edf1e4b7e693c" target="_blank">50e7dae</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomecbaker7tamusonglabtree50e7daed6663df9172061ada744edf1e4b7e693ctargetblank50e7daea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/ecbaker7-tamu/song-lab/tree/50e7daed6663df9172061ada744edf1e4b7e693c" target="_blank">50e7dae</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    bootswatch/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/toga.Rmd</code>) and HTML
(<code>docs/toga.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/50e7daed6663df9172061ada744edf1e4b7e693c/analysis/toga.Rmd" target="_blank">50e7dae</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-13
</td>
<td>
wflow_publish("analysis/toga.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/049c849487794b6193680b5695ba5bbcd1ed3c9f/docs/toga.html" target="_blank">049c849</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/c23e114a28012db85c96c68aa2546a81b0e6e227/analysis/toga.Rmd" target="_blank">c23e114</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-13
</td>
<td>
wflow_publish("analysis/toga.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/b35fb5e98ba20827cb21322d78de2d44b88e27a1/docs/toga.html" target="_blank">b35fb5e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/af7a6105d0b64194f2e5e3bc1245eb02f6ac791e/analysis/toga.Rmd" target="_blank">af7a610</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-12
</td>
<td>
wflow_publish("analysis/toga.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/340b8a4591c55acafd18d4b41764fe04979242d0/docs/toga.html" target="_blank">340b8a4</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/33cb0c89985d5be99facb76adfda4a0d1cd72fae/docs/toga.html" target="_blank">33cb0c8</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/699516f7c7eed09ab1b51b7ab84287a6fe482d29/analysis/toga.Rmd" target="_blank">699516f</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-12
</td>
<td>
wflow_publish("analysis/toga.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/ad4fd013364928ee63e1a97d0f72bec7347b9804/docs/toga.html" target="_blank">ad4fd01</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/e86ae7a1ade559c5d914fe5e31ff1459a4083b30/analysis/toga.Rmd" target="_blank">e86ae7a</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-12
</td>
<td>
wflow_publish("analysis/toga.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/757d1db6beac5cc327588ad25e91b81ec5cfd21b/docs/toga.html" target="_blank">757d1db</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/38d8b0f47e9df9424741cef76dfd0342aab2b523/docs/toga.html" target="_blank">38d8b0f</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/47de84c7a6c5eec8fd2003d72401ab67689e155e/analysis/toga.Rmd" target="_blank">47de84c</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-11
</td>
<td>
wflow_publish("analysis/toga.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/09b48c44a2d22cd118a02154df579a56879439a7/docs/toga.html" target="_blank">09b48c4</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/428b264ad87cc068be4acf365fafe5f7fd85bfe2/analysis/toga.Rmd" target="_blank">428b264</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-11
</td>
<td>
Whole lot of changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/428b264ad87cc068be4acf365fafe5f7fd85bfe2/docs/toga.html" target="_blank">428b264</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-11
</td>
<td>
Whole lot of changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/beb300837454a169aae997c997703b1774891f2b/docs/toga.html" target="_blank">beb3008</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-11
</td>
<td>
Try commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/41bcf66710b8066e96cb467da9b55291ab474a86/docs/toga.html" target="_blank">41bcf66</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ecbaker7-tamu/song-lab/98dac6edba6a58ec37e7da42519b279f5bcfdc64/docs/toga.html" target="_blank">98dac6e</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ecbaker7-tamu/song-lab/blob/e5dd452f742fbc4bcd123c7093bea2afc0b54eaa/analysis/toga.Rmd" target="_blank">e5dd452</a>
</td>
<td>
Emily Baker
</td>
<td>
2024-06-11
</td>
<td>
wflow_publish("analysis/toga.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="toga" class="section level1 tabset">
<h1 class="tabset">TOGA</h1>
<div id="what-is-toga" class="section level2">
<h2>What is TOGA?</h2>
<blockquote>
<p>“Annotating coding genes and inferring orthologs are two classical
&gt; challenges in genomics and evolutionary biology that have
traditionally been approached separately, limiting scalability. We
present TOGA (Tool to infer Orthologs from Genome Alignments), a method
that integrates structural gene annotation and orthology inference. TOGA
implements a different paradigm to infer orthologous loci, improves
ortholog detection and annotation of conserved genes compared with
state-of-the-art methods, and handles even highly fragmented assemblies.
TOGA scales to hundreds of genomes, which we demonstrate by applying it
to 488 placental mammal and 501 bird assemblies, creating the largest
comparative gene resources so far. Additionally, TOGA detects gene
losses, enables selection screens, and automatically provides a superior
measure of mammalian genome quality. TOGA is a powerful and scalable
method to annotate and compare genes in the genomic era.”</p>
</blockquote>
<p>— Bogdan M. Kirilenko <em>et al.</em>, <em>Integrating gene
annotation with orthology inference at scale</em></p>
<div id="steps-of-toga" class="section level4">
<h4>Steps of TOGA</h4>
<ol style="list-style-type: decimal">
<li>First, TOGA uses pairwise genome alignment between the reference
&amp; query genome represented by chains of co-linear local alignments.
Through this process, it captures orthologous gene loci and loci
containing paralogs or processed pseudo-genes. To distinguish between
these 2 groups, TOGA computes characteristic features that capture the
amount of intronic and intergenic alignments, considering each gene
&amp; overlapping chain. Additionally, synteny (conserved gene order)
can be used to help further distinguish orthologs from paralogs. TOGA
also implements machine learning to compute the probability that a chain
is an orthologous locus for the gene in question.</li>
</ol>
<ul>
<li><p>The most important feature for classification is the capture of
intronic and intergenic alignments, leaving synteny as the least
important. Hence, synteny is used as an auxiliary feature that allows
TOGA to detect orthologs that underwent translocation or
inversion.</p></li>
<li><details>
<summary>
<p>More on the machine learning feature</p>
</summary>
<p>The machine learning classifier was trained on known orthologous
genes between humans (reference) and mice (query) from Ensembl Compara.
This classifier was then tested on independent query species (rat, dog,
and armadillo) that represented different placental mammal orders and
produced a nearly perfect classification of orthologous chains.</p>
<p>When manually examining data it was found that misclassifications
that produced false positives are mostly due to partial or full gene
duplications and half of false negative may be due to faster X
chromosome evolution.</p></li>
</ul>
</details>
<ol start="2" style="list-style-type: decimal">
<li>CESAR v. 2.0 is used to determine the position of coding exons of
the focal gene in each (co-)orthologous query locus. TOGA then assesses
the reading frame intactness for each transcript and applies a gene loss
detection approach to identify gene-inactivating mutations while taking
assembly incompleteness into account. A gene is classified as
<em>lost</em> if all transcripts at all (co-)orthologous loci are
classified as lost.</li>
</ol>
<ul>
<li><details>
<summary>
<p>More on CESAR v. 2.0</p>
</summary>
<p>“CESAR 2.0 (Coding Exon Structure Aware Realigner 2.0) is a method to
realign coding exons or genes to DNA sequences using a Hidden Markov
Model.</p>
<p>Compared to its predecessor, CESAR 2.0 is 77X times faster on average
(132X times faster for large exons) and requires 30-times less memory.
In addition, CESAR 2.0 improves the accuracy of the comparative gene
annotation by two new features. First, CESAR 2.0 substantially improves
the identification of splice sites that have shifted over a larger
distance, which improves the accuracy of detecting the correct exon
boundaries. Second, CESAR 2.0 provides a new gene mode that re-aligns
entire genes at once. This mode is able to recognize complete intron
deletions and will annotate larger joined exons that arose by intron
deletion events.”</p>
<p>– From the <a href="https://github.com/hillerlab/CESAR2.0">Hiller Lab
GitHub page on CESAR 2.0</a></p>
</details></li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>In this step, TOGA determines the orthology type by considering all
reference genes and orthologous query loci that encode within an intact
reading frame. TOGA then uses an orthology graph to resolve any weakly
supported orthology relationship among many:many orthologs.</li>
</ol>
</div>
</div>
<div id="installation-running" class="section level2">
<h2>Installation &amp; Running</h2>
<div id="installation" class="section level3">
<h3>Installation</h3>
<div id="locally" class="section level4">
<h4>Locally</h4>
<p>Dependencies:<br />
* Python (Version 3.9 or later)<br />
* Java (Version 8 or later)</p>
<p>To setup TOGA, you must first install Nextflow using the command</p>
<pre class="r"><code>curl -fsSL https://get.nextflow.io | bash
# OR
conda install -c bioconda nextflow</code></pre>
<p>If you download using curl, move the nextflow.exe into a directory
that is accessible by your $PATH variable.</p>
<p>You must then clone the Hiller Lab GitHub repository onto your
machine and then change into the TOGA directory using the commands</p>
<pre class="r"><code>git clone https://github.com/hillerlab/TOGA.git
cd TOGA</code></pre>
<p>We then must install any necessary Python packages using</p>
<pre class="r"><code>python3 -m pip install -r requirements.txt --user</code></pre>
</div>
<div id="on-a-cluster" class="section level4">
<h4>On a Cluster</h4>
<p><mark> If running on a Slurm cluster (like GRACE), there are
configuration files already available in the Hiller Lab GitHub under the
nextflow_config_files directory. Therefore you may download those files
and skip this step! </mark></p>
<p>Here are the steps to prepare to run TOGA:</p>
<ol style="list-style-type: decimal">
<li><p>Create a separate directory for configuration files.</p></li>
<li><p>Create a file called “extract_chain_features_config.nf”. This
file contains the configuration for the chain features extraction step.
For these jobs a runtime limit of 1 hour and 10 Gb of memory will be
sufficient.</p></li>
<li><p>Create a file called “call_cesar_config_template.nf”. This
configuration file is for the jobs run by CESAR. It is <strong>highly
recommended</strong> that you request 24 hours of runtime. You do not
need to provide an exact amount of memory because TOGA will compute this
itself, so put in the placeholder “process.memory =”${<em>MEMORY</em>}G”
”</p></li>
</ol>
</div>
</div>
<div id="running" class="section level3">
<h3>Running</h3>
<div id="prepare-to-run" class="section level4">
<h4>Prepare to Run</h4>
<p>To run TOGA, we must first provide it with the appropriate input
files. These are:</p>
<ol style="list-style-type: decimal">
<li><p>A reference and query genomes in 2bit format (can be generated
from multi-fasta files using the <a
href="https://genome.ucsc.edu/goldenPath/help/twoBit.html">UCSC
twoBitToFa</a> command)</p></li>
<li><p>The coding gene annotation of the reference genome in bed-12
format (can by generated from genePred or gtf formats with the UCSC
utilities <a
href="https://genome.ucsc.edu/goldenPath/help/hubQuickStartSearch.html">genePredToBed</a>
or <a href="https://genome-store.ucsc.edu/">gtfToGenePred</a>)</p></li>
<li><p>A chain file containing chains of co-linear local alignments
between the reference and query genome.</p></li>
</ol>
<p>Optionally, you may also provide information about U12 introns, in
which noncanonical splice sites are common, as additional input.</p>
<details>
<summary>
Optimal Sample Genomes
</summary>
<p>In order to get the best results possible, the Hiller Lab suggests
including representative (principle) isoforms for each gene,
particularly ones that capture exon-intron structures, but exclude
isoforms that represent much shorter and likely more non-finctional
transcripts. It is also recommended to exclude transcripts that
represent fusion isoforms between 2 ancestral genes because including
these fusion transcripts interferes with the inference of the orthology
type.</p>
</details>
</div>
<div id="locally-1" class="section level4">
<h4>Locally</h4>
<p>To load TOGA and its dependencies on the Grace cluster, enter the
command</p>
<pre class="r"><code>ml GCC/8.3.0 OpenMPI/3.1.4 TOGA/2021.12.15-Python-3.7.4</code></pre>
<p>Then to launch TOGA use the command</p>
<pre class="r"><code>toga.py test_input/hg38.mm10.chr11.chain test_input/hg38.genCode27.chr11.bed test_input/hg38.2bit test_input/mm10.2bit --kt --pn test -i supply/hg38.wgEncodeGencodeCompV34.isoforms.txt --nc nextflow_config_files --cb 3,5 --cjn 500 --u12 supply/hg38.U12sites.tsv --ms --nextflow_dir nextflow_logs</code></pre>
</div>
<div id="on-a-cluster-1" class="section level4">
<h4>On a Cluster</h4>
<p>To run TOGA on a slurm-based cluster</p>
</div>
<div id="testing" class="section level4">
<h4>Testing</h4>
<p>The Hiller Lab provides a test <a
href="https://hgdownload.cse.ucsc.edu/goldenpath/hg38/bigZips/hg38.2bit">reference</a>
and <a
href="https://hgdownload.cse.ucsc.edu/goldenpath/mm10/bigZips/mm10.2bit">query</a>
genome, already in the required 2-bit formatting. You will also need to
download the <a
href="https://github.com/hillerlab/TOGA/tree/master/supply">supply</a>
folder in order to have the U12 data for the genomes. You can download
the whole folder at once by copy and pasting the URL into <a
href="https://download-directory.github.io/">this website</a>.</p>
<p>After this data has been downloaded &amp; placed in the correct
directory, call the following command:</p>
<pre class="r"><code>./toga.py test_input/hg38.mm10.chr11.chain test_input/hg38.genCode27.chr11.bed ${path_to_human_2bit} ${path_to_mouse_2bit} --kt --pn test -i supply/hg38.wgEncodeGencodeCompV34.isoforms.txt --nc ${path_to_nextflow_config_dir} --cb 3,5 --cjn 500 --u12 supply/hg38.U12sites.tsv --ms</code></pre>
<p>This will take about 20 minutes on a 500 core cluster.</p>
</div>
</div>
</div>
<div id="troubleshooting" class="section level2">
<h2>Troubleshooting</h2>
<div id="grace-permissionerror-errno-13" class="section level3">
<h3>Grace: PermissionError: [Errno 13]</h3>
<p>The full input and error message is here:</p>
<pre class="r"><code>  (base) [ecbaker7@c076 toga]$ toga.py test_input/hg38.mm10.chr11.chain test_input/hg38.genCode27.chr11.bed $/toga $/toga --kt --pn test -i supply/hg38.wgEncodeGencodeCompV34.isoforms.txt --nc $/scratch/group/songlab/emily/toga/nextflow_config_files --cb 3,5 --cjn 500 --u12 supply/hg38.U12sites.tsv --ms  
#### Initiating TOGA class ####
Checking dependencies...
check if binaries are compiled and libs are installed...
All dependencies found
/bin/sh: git: command not found
Traceback (most recent call last):
  File &quot;/sw/eb/sw/TOGA/2021.12.15-foss-2019b-Python-3.7.4/toga.py&quot;, line 1965, in &lt;module&gt;
    main()
  File &quot;/sw/eb/sw/TOGA/2021.12.15-foss-2019b-Python-3.7.4/toga.py&quot;, line 1960, in main
    toga_manager = Toga(args)
  File &quot;/sw/eb/sw/TOGA/2021.12.15-foss-2019b-Python-3.7.4/toga.py&quot;, line 88, in __init__
    self.nextflow_dir = self.__get_nf_dir(args.nextflow_dir)
  File &quot;/sw/eb/sw/TOGA/2021.12.15-foss-2019b-Python-3.7.4/toga.py&quot;, line 362, in __get_nf_dir
    os.mkdir(default_dir) if not os.path.isdir(default_dir) else None
PermissionError: [Errno 13] Permission denied: &#39;/sw/eb/sw/TOGA/2021.12.15-foss-2019b-Python-3.7.4/nextflow_logs&#39;</code></pre>
<p>I corrected this by creating a new directory called
<em>nextflow_logs</em> and then adding the flag –nextflow_dir or –nd
followed by the path to <em>nextflow_logs</em>. This gives Nextflow a
directory to operate from. The full command is:</p>
<pre class="r"><code>toga.py test_input/hg38.mm10.chr11.chain test_input/hg38.genCode27.chr11.bed test_input/hg38.2bit test_input/mm10.2bit --kt --pn test -i supply/hg38.wgEncodeGencodeCompV34.isoforms.txt --nc nextflow_config_files --cb 3,5 --cjn 500 --u12 supply/hg38.U12sites.tsv --ms --nextflow_dir nextflow_logs</code></pre>
</div>
<div id="error-on-toga-step-2-nextflow-jobs-wont-execute"
class="section level3">
<h3>Error on Toga step 2: Nextflow jobs won’t execute</h3>
<p>I’m having an error where I get to step 2 of TOGA but it can’t
complete any jobs and keeps retrying all of them. The error message I
get in my .out file is</p>
<pre class="r"><code>Traceback (most recent call last):
  File &quot;/var/spool/slurmd/job10522779/slurm_script&quot;, line 19, in &lt;module&gt;
    from modules.common import parts
ModuleNotFoundError: No module named &#39;modules&#39;</code></pre>
<p>I corrected this error by running the job through a .sh file called
<em>run-toga.sh</em> using sbatch. It contained the commands below:</p>
<pre class="r"><code>#!/bin/sh
ml GCC/8.3.0 OpenMPI/3.1.4 TOGA/2021.12.15-Python-3.7.4      # load all necessary modules

toga.py test_input/hg38.mm10.chr11.chain test_input/hg38.genCode27.chr11.bed test_input/hg38.2bit test_input/mm10.2bit --kt --pn test -i supply/hg38.wgEncodeGencodeCompV34.isoforms.txt --nc nextflow_config_files --cb 3,5 --cjn 500 --u12 supply/hg38.U12sites.tsv --ms --nextflow_dir nextflow_logs</code></pre>
<p>You can then run <em>run-toga.sh</em> on the cluster using the
command</p>
<pre class="r"><code>sbatch --ntasks=48 --mem=360G --nodes=1 --time=0-1:00:00 run-toga.sh</code></pre>
</div>
<div id="error-process-nextflow-died" class="section level3">
<h3>Error: Process nextflow died</h3>
<p>The full error code is here:</p>
<pre class="r"><code>Command executed:

  sbatch .command.run

Command exit status:
  0

Command output:
  sbatch: error: (from job_submit) either --ntasks, or --ntasks-per-node, or --nodes should be provided.
  sbatch: error: Batch job submission failed: Unspecified error
  

Work dir:
  /scratch/group/songlab/emily/toga/nextflow_logs/test_chain_feats_at_1718311976/work/16/a70dc9c3f7b04b94a742de26c47b86

Tip: you can try to figure out what&#39;s wrong by changing to the process work dir and showing the script file named `.command.sh`



Error! Process nextflow /sw/eb/sw/TOGA/2021.12.15-foss-2019b-Python-3.7.4/execute_joblist.nf --joblist /scratch/group/songlab/emily/toga/test/temp/chain_class_jobs_combined -c /scratch/group/songlab/emily/toga/nextflow_config_files/extract_chain_features_config.nf died
Program finished with exit code 1

Making gene_id: bed_data dict...</code></pre>
<p>IT believes it is an issue with the storage space on my home drive,
but this hasn’t seemed to have corrected the issue.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 22631)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.7.1

loaded via a namespace (and not attached):
 [1] vctrs_0.6.5       httr_1.4.7        cli_3.6.2         knitr_1.47       
 [5] rlang_1.1.3       xfun_0.44         stringi_1.8.4     processx_3.8.4   
 [9] promises_1.3.0    jsonlite_1.8.8    glue_1.7.0        rprojroot_2.0.4  
[13] git2r_0.33.0      htmltools_0.5.8.1 httpuv_1.6.15     ps_1.7.6         
[17] sass_0.4.9        rmarkdown_2.27    jquerylib_0.1.4   evaluate_0.23    
[21] fastmap_1.2.0     yaml_2.3.8        lifecycle_1.0.4   whisker_0.4.1    
[25] stringr_1.5.1     compiler_4.4.0    fs_1.6.4          Rcpp_1.0.12      
[29] rstudioapi_0.16.0 later_1.3.2       digest_0.6.35     R6_2.5.1         
[33] callr_3.7.6       magrittr_2.0.3    bslib_0.7.0       tools_4.4.0      
[37] cachem_1.1.0      getPass_0.2-4    </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
