<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Transcriptome Assembly with a Reference Genome</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="theme3.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Home</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-microscope"></span>
     
    Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Comparative Genomics of Schistocerca gregaria -- In prep</li>
    <li class="dropdown-header">Comparative Genomics of Mormon Cricket subgroups -- UNDER CONSTRUCTION</li>
    <li class="dropdown-header">Ensifera Ear Transcriptomics -- UNDER CONSTRUCTION</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-code-compare"></span>
     
    Comparative Genomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">FormicidaeMolecularEvolution - OrthoFinder Workflow for Longest Isoform</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="barkdull.html">TAMU Workflow</a>
        </li>
        <li>
          <a href="barkdull_asu.html">ASU Workflow</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">RepeatModeler &amp; RepeatMasker</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="rep-model-mask.html">TAMU Workflow</a>
        </li>
        <li>
          <a href="rep-model-mask_asu.html">ASU Workflow</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">PSMC Model Analysis</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="psmc.html">TAMU Workflow</a>
        </li>
        <li>
          <a href="psmc_asu.html">ASU Workflow</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-pen-to-square"></span>
     
    Transcriptomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="tf_mapping.html">Transcription Factor Family Mapping (ASU Only)</a>
    </li>
    <li>
      <a href="trans-to-ref.html">Transcriptome Data Assembly with a Reference Genome (ASU)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-flask"></span>
     
    Proteomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="alphafold.html">AlphaFold</a>
    </li>
    <li>
      <a href="chimerax.html">Chimera X</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-palette"></span>
     
    Visualization
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="kimura.html">Kimura Plots (Repeat Content)</a>
    </li>
    <li>
      <a href="genespace.html">GENESPACE (Synteny)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-book-open-reader"></span>
     
    Reference
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="slurm-comm.html">Linux and Slurm Commands</a>
    </li>
    <li>
      <a href="definitions.html">Glossary of Terminology</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Transcriptome Assembly with a Reference
Genome</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-10-22
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>song-lab/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240605code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240605)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240605code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240605)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomemilycbakersonglabtree3cfa3a5190e21364c43046ee3cafdb9934a675aatargetblank3cfa3a5a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/emily-cbaker/song-lab/tree/3cfa3a5190e21364c43046ee3cafdb9934a675aa" target="_blank">3cfa3a5</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomemilycbakersonglabtree3cfa3a5190e21364c43046ee3cafdb9934a675aatargetblank3cfa3a5a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/emily-cbaker/song-lab/tree/3cfa3a5190e21364c43046ee3cafdb9934a675aa" target="_blank">3cfa3a5</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/trans-to-ref.Rmd</code>) and HTML
(<code>docs/trans-to-ref.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/emily-cbaker/song-lab/blob/3cfa3a5190e21364c43046ee3cafdb9934a675aa/analysis/trans-to-ref.Rmd" target="_blank">3cfa3a5</a>
</td>
<td>
emilyc-baker
</td>
<td>
2025-10-22
</td>
<td>
wflow_publish("analysis/trans-to-ref.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/e7223293e2da2dc35d0b767f72ba87baa072d6d3/docs/trans-to-ref.html" target="_blank">e722329</a>
</td>
<td>
emilyc-baker
</td>
<td>
2025-10-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/emily-cbaker/song-lab/1e0f51a5cd0574f21b862d19aa2b2b5c57bfc224/docs/trans-to-ref.html" target="_blank">1e0f51a</a>
</td>
<td>
emilyc-baker
</td>
<td>
2025-10-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/emily-cbaker/song-lab/blob/dece38574257f32d55e11a499de9421101d7e75f/analysis/trans-to-ref.Rmd" target="_blank">dece385</a>
</td>
<td>
emilyc-baker
</td>
<td>
2025-10-22
</td>
<td>
wflow_publish(c("analysis/alphafold.Rmd", "analysis/barkdull.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="mapping-transcriptome-data" class="section level2 tabset">
<h2 class="tabset">Mapping Transcriptome Data</h2>
<div id="data-preprocessing" class="section level3">
<h3>Data Preprocessing</h3>
<div id="i-load-necessary-software" class="section level4">
<h4>i) Load Necessary Software</h4>
<div id="ia-load-r-modules" class="section level5">
<h5>ia) Load R Modules</h5>
<p>Before we begin processing our files, we need to load the following R
libraries in our R session. Load the R module using the command “module
load r-4.3.2-gcc-11.2.0” and copy and paste the following code into your
R session:</p>
<pre class="r"><code>library(&quot;knitr&quot;)
library(&quot;rmdformats&quot;)
library(&quot;tidyverse&quot;)
library(&quot;DT&quot;)  # for making interactive search table
library(&quot;plotly&quot;) # for interactive plots
library(&quot;ggthemes&quot;) # for theme_calc
library(&quot;reshape2&quot;)
library(&quot;readr&quot;)
library(&quot;ggplot2&quot;)

## Global options
options(max.print=&quot;10000&quot;)
knitr::opts_chunk$set(
    echo = TRUE,
    message = FALSE,
    warning = FALSE,
    cache = FALSE,
    comment = FALSE,
    prompt = FALSE,
    tidy = TRUE
)
opts_knit$set(width=75)</code></pre>
</div>
<div id="ib-create-mamba-environment-for-multiqc"
class="section level5">
<h5>ib) Create Mamba Environment for MultiQC</h5>
<p>The package MultiQC is not readily available on the Sol cluster, so
we will need to install it via a Mamba environment. Log into the Sol
cluster via ssh and follow the below commands to create a new Mamba
environment and install the MultiQC package into this environment.</p>
<pre class="r"><code>interactive -p htc -c 4 -t 30

module load mamba/latest

mamba create -n trans-map -c conda-forge -c bioconda python=3 multiqc trimgalore</code></pre>
<p>After it finishes loading, you will be prompted to answer yes or no
(Y/n), type Y to accept.</p>
<p>To then verify the creation of this environment, type the
command:</p>
<pre class="r"><code>source activate trans-map

python

import multiqc</code></pre>
<p>If you receive no errors, then installation is successful!</p>
</div>
</div>
<div id="ii-quality-control" class="section level4">
<h4>ii) Quality Control</h4>
<p>First, we will need to assess each .fastq file’s quality using
FASTQC. Create a file called <em>qual-control.sh</em> and copy and paste
the follwoing code into it:</p>
<pre class="r"><code>#!/bin/bash

##NECESSARY JOB SPECIFICATIONS
#SBATCH --job-name=quality-control       #Set the job name to &quot;psmc-referencegenome&quot;
#SBATCH --time=02:00:00         #Set the wall clock limit to 8 hours
#SBATCH --ntasks=1             #Request 1 task
#SBATCH --cpus-per-task=8       #Request 8 cpus
#SBATCH --mem=30G              #Request 30GB per node
#SBATCH --mail-user={USER@EMAIL.COM}
#SBATCH --output=stdout.%x.%j
#SBATCH --error=stderr.%x.%j

module load fastqc-0.11.3-gcc-12.1.0

mkdir fastqc_out

fastqc -t 8 ../data/raw/*.fastq -o ../data/fastqc_out/

module purge

module load mamba/latest 

source activate trans-map

multiqc --title &#39;MultiQC Report&#39; -v /scratch/ebaker48/greg_transcript/data/fastqc_out</code></pre>
<p>Edit the email in the slurm header to your email (if you wish to
receive email alerts when your job begins and ends, otherwise delete
that line), the path to your fastqc_out directory (your current
directory + /fastqc_out/), and optionally, the title of the MultiQC
report.</p>
<p>This produces FASTQC report files for each fastq file and the MULTIQC
is a compiles report summarizing all of the FASTQC reports.</p>
</div>
<div id="iii-trim-and-remove-adapters" class="section level4">
<h4>iii) Trim and Remove Adapters</h4>
<p>To perform this trimming step we will use trim_galore to remove the
contaminants from the files.</p>
<p>Create a file called <em>trim.sh</em> and copy and paste the
following code into it:</p>
<pre class="r"><code>#!/bin/bash
#SBATCH --job-name=trim_galore_all
#SBATCH --output=trim_galore_all_%j.log
#SBATCH --error=trim_galore_all_%j.err
#SBATCH --time=12:00:00
#SBATCH --mem=32G
#SBATCH -c 8

# Load Trim Galore via mamba environment (Step 0)
module purge
module load mamba/latest
source activate trans-map

INDIR=&quot;../data/raw&quot;
OUTDIR=&quot;../data/trimmed&quot;

# --------------------------------------------------
# Loop through paired-end FASTQ files
# --------------------------------------------------
for READ1 in &quot;${INDIR}&quot;/*_R1_001.fastq; do
    READ2=&quot;${READ1/_R1_001.fastq/_R2_001.fastq}&quot;

    if [[ -f &quot;$READ2&quot; ]]; then
        echo &quot;Processing: $READ1 and $READ2&quot;

        trim_galore --trim-n \
            --cores 8 \
            --quality 20 \
            --clip_R1 2 \
            --clip_R2 2 \
            --nextera \
            --fastqc \
            --output_dir &quot;$OUTDIR&quot; \
            --paired &quot;$READ1&quot; &quot;$READ2&quot;

    else
        echo &quot;Warning: Could not find pair for $READ1&quot;
    fi
done</code></pre>
</div>
<div id="iv-trimming-quality-check" class="section level4">
<h4>iv) Trimming Quality Check</h4>
<p>To ensure our trimming was not overly aggressive, we can perform one
last quality check of our trimmed files using the below script:</p>
<pre class="r"><code>#!/bin/bash
#SBATCH --job-name=fastqc_trimmed
#SBATCH --output=fastqc_trimmed_%j.log
#SBATCH --error=fastqc_trimmed_%j.err
#SBATCH --time=3:00:00
#SBATCH --mem=500M
#SBATCH --cpus-per-task=2

module load fastqc-0.11.3-gcc-12.1.0

fastqc ../data/trimmed/*_val_1.fq -o ../data/trimmed/
fastqc ../data/trimmed/*_val_2.fq -o ../data/trimmed/</code></pre>
</div>
</div>
<div id="transcript-mapping" class="section level3">
<h3>Transcript Mapping</h3>
<div id="i-indexing-genomes" class="section level4">
<h4>i) Indexing Genomes</h4>
<p>Now that our data has been processed, we can begin the mapping
process. To accomplish this, we will use STAR in the following
script:</p>
<pre class="r"><code>#!/bin/bash
#SBATCH --job-name=star_index
#SBATCH --time=2-00:00:00
#SBATCH --mem=256G
#SBATCH -c 12
#SBATCH --mail-type=ALL
#SBATCH --mail-user=ebaker48@asu.edu
#SBATCH --output=stdout.%x.%j
#SBATCH --error=stderr.%x.%j

# --------------------------------------------------
# Load STAR module
# --------------------------------------------------
module purge
module load star-2.7.10b-gcc-12.1.0

# --------------------------------------------------
# Define paths (edit these!)
# --------------------------------------------------
REFDIR=/scratch/ebaker48/greg_transcript/data/reference   # base reference directory
SPECIES=GCF_023897955.1_iqSchGreg1.2                       # species name
REF_GENOME=${REFDIR}/${SPECIES}_genomic.fna
ANNOTATION=${REFDIR}/${SPECIES}_genomic.gtf
OUTDIR=${REFDIR}/index/${SPECIES}/STAR

# --------------------------------------------------
# Make output directory if it doesn’t exist
# --------------------------------------------------
mkdir -p &quot;$OUTDIR&quot;

# --------------------------------------------------
# Run STAR genomeGenerate
# --------------------------------------------------

LIMIT_BYTES=$((256 * 1024 * 1024 * 1024))

STAR --runMode genomeGenerate \
     --runThreadN 12 \
     --genomeDir &quot;$OUTDIR&quot; \
     --genomeFastaFiles &quot;$REF_GENOME&quot; \
     --sjdbGTFfile &quot;$ANNOTATION&quot; \
     --alignIntronMax 2500000 \
     --sjdbOverhang 149
     --limitGenomeGenerateRAM $LIMIT_BYTES</code></pre>
<p>Please note the specialized memory specifications I’ve used in this
script. This is due to the Sol cluster killing the job for and out of
memory error, so I had ot juice my script up a bit for it to make it
through. You may adjust this as needed, but be sure to a) edit the
LIMIT_BYTES variable to match your #SBATCH –mem specification, b) remove
the variable entirely and simply replace the variable with your
requested –mem gigs, or c) delete the line from the STAR command
entirely.</p>
</div>
<div id="ii-alignment" class="section level4">
<h4>ii) Alignment</h4>
<p>Use the following script to produce the STAR alignment on a
cluster:</p>
<pre class="r"><code>#!/bin/bash
#SBATCH --job-name=star_align
#SBATCH --output=star_align_%j.log
#SBATCH --error=star_align_%j.err
#SBATCH --time=08:00:00
#SBATCH --mem=100G
#SBATCH -c 16

# --------------------------------------------------
# Load STAR
# --------------------------------------------------
module purge
module load star-2.7.10b-gcc-12.1.0

# --------------------------------------------------
# Define input/output paths (EDIT THESE!)
# --------------------------------------------------
REFDIR=/scratch/ebaker48/greg_transcript/data/reference
WORKDIR=/scratch/ebaker48/greg_transcript/data

SPECIES=GCF_023897955.1_iqSchGreg1.2
INDEX=${REFDIR}/index/${SPECIES}/STAR
ANNOTATION=${REFDIR}/${SPECIES}_genomic.gtf

TRIMDIR=${WORKDIR}/trimmed
OUTDIR=${WORKDIR}/02-${SPECIES}-star
COUNTDIR=${WORKDIR}/03-${SPECIES}-DESeq2

mkdir -p &quot;$OUTDIR&quot; &quot;$COUNTDIR&quot;

# --------------------------------------------------
# Loop through all R1 files in trimmed dir
# --------------------------------------------------
for R1 in ${TRIMDIR}/*_R1_001_val_1.fq; do
    R2=${R1/_R1_001_val_1.fq/_R2_001_val_2.fq}
    LOCUST=$(basename &quot;$R1&quot; _R1_001_val_1.fq)

    if [[ ! -f &quot;$R2&quot; ]]; then
        echo &quot;❌ Skipping $R1 (no matching R2 found)&quot;
        continue
    fi

    echo &quot;✅ Processing $LOCUST&quot;

    PREFIX=${OUTDIR}/${LOCUST}_
    BAM=${OUTDIR}/${LOCUST}_Aligned.sortedByCoord.out.bam
    READTABLE=${OUTDIR}/${LOCUST}_ReadsPerGene.out.tab
    COUNTS=${COUNTDIR}/${LOCUST}_counts.txt

    STAR --runThreadN 16 \
         --genomeDir &quot;$INDEX&quot; \
         --genomeLoad NoSharedMemory \
         --limitBAMsortRAM 32000000000 \
         --outSAMtype BAM SortedByCoordinate \
         --outSAMattrRGline ID:${LOCUST} SM:${LOCUST} LB:Stranded_Total_RNA_RiboZero PL:Illumina PU:NovaSeq6000 \
         --quantMode TranscriptomeSAM GeneCounts \
         --twopassMode Basic \
         --sjdbGTFfile &quot;$ANNOTATION&quot; \
         --sjdbOverhang 149 \
         --outSAMattributes NH HI AS NM MD \
         --alignIntronMax 2500000 \
         --outSAMunmapped Within \
         --readFilesIn &quot;$R1&quot; &quot;$R2&quot; \
         --outFileNamePrefix &quot;$PREFIX&quot;

    # --------------------------------------------------
    # Index BAM
    # --------------------------------------------------
    module purge
    module load samtools-1.21-gcc-12.1.0
    samtools index -c &quot;$BAM&quot;

    # --------------------------------------------------
    # Extract counts
    # --------------------------------------------------
    cut -f1,4 &quot;$READTABLE&quot; | grep -v &quot;_&quot; &gt; &quot;$COUNTS&quot;

    # reload STAR for next iteration
    module purge
    module load star-2.7.10b-gcc-12.1.0
done</code></pre>
<p>This will produce 2 directories in your working directory, one
containing the STAR alignment outputs and one containing the per-sample
count files for DESeq2.</p>
</div>
<div id="iii-quantification-using-genecount" class="section level4">
<h4>iii) Quantification using GeneCount</h4>
<p>The output file {locust}.ReadsPerGene.out.tab contains the read
counts per gene as follows:</p>
<table class="table table-striped table-hover table-condensed table-responsive" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
</caption>
<thead>
<tr>
<th style="text-align:left;font-weight: bold;background-color: rgba(191, 162, 176, 255) !important;">
Column_1
</th>
<th style="text-align:left;font-weight: bold;background-color: rgba(191, 162, 176, 255) !important;">
Column_2
</th>
<th style="text-align:left;font-weight: bold;background-color: rgba(191, 162, 176, 255) !important;">
Column_3
</th>
<th style="text-align:left;font-weight: bold;background-color: rgba(191, 162, 176, 255) !important;">
Column_4
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;width: 30em; ">
Gene ID
</td>
<td style="text-align:left;width: 30em; ">
Counts for unstranded RNA-seq
</td>
<td style="text-align:left;width: 30em; ">
Counts for 1st read strand aligned with RNA
</td>
<td style="text-align:left;width: 30em; ">
Counts for 2nd read strand aligned with RNA
</td>
</tr>
</tbody>
</table>
<p>We can use this to verify the data, as in a stranded library
preparation, there should be a large difference between the number of
reads mapped to known genes in forward versus reverse strands. We need
to rewrite this data so we can input it into DESeq2 for visualization to
verify the expression pattern is appropriate. We can do this using the
following script:</p>
<pre class="r"><code>#!/usr/bin/env bash
# This script extracts the 4th column (gene counts) from STAR ReadsPerGene.out.tab files
# and writes them into new files for DESeq2 input.

# Set directories
INPUT_DIR=&quot;/scratch/ebaker48/greg_transcript/data/02-GCF_023897955.1_iqSchGreg1.2-star&quot;
OUTPUT_DIR=&quot;/scratch/ebaker48/greg_transcript/data/02-GCF_023897955.1_iqSchGreg1.2-star/gene_count_quant&quot;

# Create output directory if it doesn’t exist
mkdir -p &quot;$OUTPUT_DIR&quot;

echo &quot;Extracting gene counts from files in $INPUT_DIR...&quot;

# Loop through all ReadsPerGene.out.tab files
for file in &quot;$INPUT_DIR&quot;/*ReadsPerGene.out.tab; do
    echo &quot;Processing: $file&quot;
    # Extract columns 1 and 4, remove rows with underscores, and write to new file
    base=$(basename &quot;$file&quot; ReadsPerGene.out.tab)
    output_file=&quot;${OUTPUT_DIR}/${base}counts.txt&quot;
    cut -f1,4 &quot;$file&quot; | grep -v &quot;_&quot; &gt; &quot;$output_file&quot;
    echo &quot;Saved: $output_file&quot;
done

echo &quot;✅ All files processed! Results are in: $OUTPUT_DIR&quot;</code></pre>
</div>
</div>
<div id="differential-gene-expression-analysis" class="section level3">
<h3>Differential Gene Expression Analysis</h3>
<div id="i-prepare-input-files" class="section level4">
<h4>i) Prepare input files</h4>
<p>To perform this analysis, we will need 4 files:</p>
<ol style="list-style-type: decimal">
<li><p>Raw count matrices These were produces during the STAR mapping
step.</p></li>
<li><p>Transcript to gene annotation file We have to reformat our .gtf
file for input into DESeq2 and edgeR. First, download your species of
interest’s .gtf annotation file from NCBI. Then run the following
command directly in your terminal:</p></li>
</ol>
<pre class="r"><code>zcat {SPECIES}.gtf.gz | awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;}{if($3==&quot;transcript&quot;){split($9, a, &quot;\&quot;&quot;); print a[4],a[2],a[8]}}&#39; &gt; tx2gene.{SPECIES}.csv</code></pre>
<p>This command outputs a .csv file containing the transcript ID, gene
ID, and gene symbol in the 3 co,umns respectively.</p>
<ol start="3" style="list-style-type: decimal">
<li>Sample sheet This file is a .txt file that is tab-delimited
containing necessary information about your samples. The format is as
follows:</li>
</ol>
<pre class="r"><code>SampleName  FileName  Tissue  ControlVariable</code></pre>
<p>For example, here is the file I created to analyze differential
expression between male and female <em>Schistocerca gregaria</em> in
crowded conditions:</p>
<pre class="r"><code>SampleName  FileName    Tissue  Sex
23475Son_SchgreF1_S54_L001  23475Son_SchgreF1_S54_L001_counts.txt   Ear Female
23475Son_SchgreF2_S55_L001  23475Son_SchgreF2_S55_L001_counts.txt   Ear Female
23475Son_SchgreF3_S56_L001  23475Son_SchgreF3_S56_L001_counts.txt   Ear Female
23475Son_SchgreM1_S51_L001  23475Son_SchgreM1_S51_L001_counts.txt   Ear Male
23475Son_SchgreM2_S52_L001  23475Son_SchgreM2_S52_L001_counts.txt   Ear Male
23475Son_SchgreM3_S53_L001  23475Son_SchgreM3_S53_L001_counts.txt   Ear Male</code></pre>
<p>Extra columns can be added to include information that could be
useful for statistical analysis, such as rearing condition, time points,
replicates, etc.</p>
<ol start="4" style="list-style-type: decimal">
<li>Reference State Finally, we will need the reference state for which
we will perform our differential expression analysis. In the example
above, I chose male as the reference state, as the reference genome for
<em>S. gregaria</em> on NCBI is from a male specimen.</li>
</ol>
</div>
<div id="ii-deseq2-analysis-from-star-output" class="section level4">
<h4>ii) DESeq2 Analysis from STAR Output</h4>
<p>We can use the following R script to generate the DEG analysis using
DESeq2. The modules necessary to run this script may not be readily
available on CRAN or Bioconductor, so be prepared to use devtools or
GitHub to install certain packages.</p>
<pre class="r"><code>library(&quot;DESeq2&quot;)
library(&quot;tximport&quot;)
library(&quot;txdbmaker&quot;)
library(&quot;knitr&quot;)
library(&quot;rmdformats&quot;)
library(&quot;tidyverse&quot;)
library(&quot;data.table&quot;)
library(&quot;DT&quot;)  # for making interactive search table
library(&quot;plotly&quot;) # for interactive plots
library(&quot;ggthemes&quot;) # for theme_calc
library(&quot;reshape2&quot;)
library(&quot;ComplexHeatmap&quot;)
library(&quot;RColorBrewer&quot;)
library(&quot;circlize&quot;)
library(&quot;apeglm&quot;)
library(&quot;ggpubr&quot;)
library(&quot;ggplot2&quot;)
library(&quot;ggrepel&quot;)
library(&quot;EnhancedVolcano&quot;)
library(&quot;SARTools&quot;)
library(&quot;pheatmap&quot;)
library(&quot;clusterProfiler&quot;)
library(&quot;sva&quot;)
library(&quot;cowplot&quot;)
library(&quot;ashr&quot;)
library(&quot;ggforce&quot;)
library(&quot;ggConvexHull&quot;)

###############################################################
### USER-DEFINED PARAMETERS (EDIT THESE ONLY)
###############################################################

homeDir &lt;- &quot;/path/to/folder&quot;                # e.g., your home project directory
workDir &lt;- &quot;path/to/working/directory&quot;  
projectName &lt;- &quot;Project Name&quot;
author &lt;- &quot;Author McScienceFace&quot;
species &lt;- &quot;Species name&quot;                   # MUST MATCH STAR RUN SPECIES NAME!
varInt &lt;- &quot;Condition&quot;                       # Experimental factor of interest
condRef &lt;- &quot;Control&quot;                        # Reference condition (baseline)
sampleFile &lt;- &quot;Sample File Name&quot;            # Name of your tab-delimited sample file

###############################################################
### AUTOMATIC PATHS AND INITIALIZATION
###############################################################

# Create working directory structure
dir.create(workDir, showWarnings = FALSE, recursive = TRUE)
setwd(workDir)

rawDir &lt;- file.path(workDir, paste0(&quot;03-&quot;, species, &quot;-DESeq2&quot;))
dir.create(rawDir, showWarnings = FALSE, recursive = TRUE)

# Create subdirectory for DESeq2 run
Dirname &lt;- paste0(&quot;DESeq2_&quot;, projectName)
dir.create(Dirname, showWarnings = FALSE)
setwd(Dirname)
workDir_DEseq2 &lt;- getwd()

# Define path to target file
targetFile &lt;- file.path(workDir, paste0(sampleFile))

###############################################################
### DESeq2 PARAMETERS
###############################################################

tresh_logfold &lt;- 1
tresh_padj &lt;- 0.05
alpha_DEseq2 &lt;- 0.05
pAdjustMethod_DEseq2 &lt;- &quot;BH&quot;
featuresToRemove &lt;- NULL
batch &lt;- NULL
fitType &lt;- &quot;parametric&quot;
cooksCutoff &lt;- TRUE
independentFiltering &lt;- TRUE
typeTrans &lt;- &quot;rlog&quot;
locfunc &lt;- &quot;median&quot;
colors &lt;- c(&quot;#B31B21&quot;, &quot;#1465AC&quot;)

###############################################################
### ANALYSIS PIPELINE
###############################################################

message(&quot;&gt;&gt;&gt; Starting DESeq2 analysis for project: &quot;, projectName)

# 1. Load target file ----
if (!file.exists(targetFile)) {
  stop(&quot;Target file not found: &quot;, targetFile)
}
target &lt;- loadTargetFile(targetFile = targetFile, varInt = varInt, condRef = condRef, batch = batch)

# 2. Load counts ----
counts &lt;- loadCountData(target = target, rawDir = rawDir, featuresToRemove = featuresToRemove)

# 3. Description plots ----
majSequences &lt;- descriptionPlots(counts = counts, group = target[, varInt], col = colors)

# 4. Run DESeq2 ----
out.DESeq2 &lt;- run.DESeq2(
  counts = counts, target = target, varInt = varInt, batch = batch,
  locfunc = locfunc, fitType = fitType, pAdjustMethod = pAdjustMethod_DEseq2,
  cooksCutoff = cooksCutoff, independentFiltering = independentFiltering,
  alpha = alpha_DEseq2
)

# 5. PCA + Clustering ----
exploreCounts(object = out.DESeq2$dds, group = target[, varInt], typeTrans = typeTrans, col = colors)

# 6. Summarize results ----
summaryResults &lt;- summarizeResults.DESeq2(
  out.DESeq2, group = target[, varInt], col = colors,
  independentFiltering = independentFiltering,
  cooksCutoff = cooksCutoff, alpha = alpha_DEseq2
)

# 7. Generate HTML report ----
writeReport.DESeq2(
  target = target, counts = counts, out.DESeq2 = out.DESeq2, summaryResults = summaryResults,
  majSequences = majSequences, workDir = workDir_DEseq2, projectName = projectName, author = author,
  targetFile = targetFile, rawDir = rawDir, featuresToRemove = featuresToRemove, varInt = varInt,
  condRef = condRef, batch = batch, fitType = fitType, cooksCutoff = cooksCutoff,
  independentFiltering = independentFiltering, alpha = alpha_DEseq2, pAdjustMethod = pAdjustMethod_DEseq2,
  typeTrans = typeTrans, locfunc = locfunc, colors = colors
)

message(&quot;&gt;&gt;&gt; DESeq2 analysis complete. Results saved to: &quot;, workDir_DEseq2)
###############################################################</code></pre>
</div>
</div>
<div id="troubleshooting" class="section level3">
<h3>Troubleshooting</h3>
<div id="lua-filter-skipped" class="section level4">
<h4>Lua Filter Skipped</h4>
<p>This error occurs when the pandoc version isn’t compatible with the
default Rmarkdown template system. I fixed this in my run by updating my
pandoc to the most recent version using the following steps:</p>
<ol style="list-style-type: decimal">
<li>Check if pandoc is updated to the most recent version (as of
10/22/25, 3.1.12.3 is the most recent).</li>
</ol>
<pre class="r"><code>module avail pandoc</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>If not the most recent version, then using the following code to
install a newer version. Since the pandoc module isn’t managed by Sol,
we have to update it ourselves.</li>
</ol>
<pre class="r"><code>cd ~
wget https://github.com/jgm/pandoc/releases/download/3.1.12.3/pandoc-3.1.12.3-linux-amd64.tar.gz

tar -xvzf pandoc-3.1.12.3-linux-amd64.tar.gz</code></pre>
<p>There should now be a folder named “pandoc-3.1.12.3”. We can now move
this file to our personal bin directory using the following code:</p>
<pre class="r"><code>mkdir -p ~/bin

mv pandoc-3.1.12.3/bin/pandoc ~/bin/</code></pre>
<p>Now we can add the new pandoc module to our path so the computer is
able to locate the module when we run our R script.</p>
<pre class="r"><code>echo &#39;export PATH=$HOME/bin:$PATH&#39; &gt;&gt; ~/.bashrc
source ~/.bashrc</code></pre>
<p>We can now verify our installation with</p>
<pre class="r"><code>pandoc --version</code></pre>
<p>This should match the version you downloaded.</p>
</div>
</div>
<div id="citations" class="section level3">
<h3>Citations</h3>
<p>Workflow developed by Maeva Techer</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.5.1 (2025-06-13)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.5-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.5-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.1

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Phoenix
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] kableExtra_1.4.0 knitr_1.50       workflowr_1.7.2 

loaded via a namespace (and not attached):
 [1] jsonlite_2.0.0     compiler_4.5.1     promises_1.3.3     Rcpp_1.1.0        
 [5] xml2_1.4.0         stringr_1.5.2      git2r_0.36.2       callr_3.7.6       
 [9] later_1.4.4        jquerylib_0.1.4    textshaping_1.0.4  systemfonts_1.3.1 
[13] scales_1.4.0       yaml_2.3.10        fastmap_1.2.0      R6_2.6.1          
[17] tibble_3.3.0       rprojroot_2.1.1    svglite_2.2.1      RColorBrewer_1.1-3
[21] bslib_0.9.0        pillar_1.11.1      rlang_1.1.6        cachem_1.1.0      
[25] stringi_1.8.7      httpuv_1.6.16      xfun_0.53          getPass_0.2-4     
[29] fs_1.6.6           sass_0.4.10        viridisLite_0.4.2  cli_3.6.5         
[33] magrittr_2.0.4     ps_1.9.1           digest_0.6.37      processx_3.8.6    
[37] rstudioapi_0.17.1  lifecycle_1.0.4    vctrs_0.6.5        evaluate_1.0.5    
[41] glue_1.8.0         farver_2.1.2       whisker_0.4.1      rmarkdown_2.30    
[45] httr_1.4.7         tools_4.5.1        pkgconfig_2.0.3    htmltools_0.5.8.1 </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
