---
title: "Differential Gene Expression"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
    toc_float: true
    toc_collapsed: false
    theme: default
    css: "theme3.css"
toc_depth: 1
number_sections: true
editor_options:
  chunk_output_type: console
---
## Analysis Steps & Troubleshooting {.tabset}

### Differential Gene Expression Analysis
#### i) Prepare input files
To perform this analysis, we will need 4 files:

1. Raw count matrices
These were produced during the STAR mapping step, assuming you followed the transcriptome data assembly process documented on this site. 

2. Transcript to gene annotation file
We have to reformat our .gtf file for input into DESeq2 and edgeR. First, download your species of interest's .gtf annotation file from NCBI. Then run the following command directly in your terminal:

```{r, eval=FALSE}

zcat {SPECIES}.gtf.gz | awk -F "\t" 'BEGIN{OFS="\t"}{if($3=="transcript"){split($9, a, "\""); print a[4],a[2],a[8]}}' > tx2gene.{SPECIES}.csv

```

This command outputs a .csv file containing the transcript ID, gene ID, and gene symbol in the 3 co,umns respectively.

3. Sample sheet
This file is a .txt file that is tab-delimited containing necessary information about your samples. The format is as follows:

```{r, eval=FALSE}

SampleName  FileName  Tissue  ControlVariable

```

For example, here is the file I created to analyze differential expression between male and female *Schistocerca gregaria* in crowded conditions:

```{r, eval=FALSE}

SampleName	FileName	Tissue	Sex
23475Son_SchgreF1_S54_L001	23475Son_SchgreF1_S54_L001_counts.txt	Ear Female
23475Son_SchgreF2_S55_L001	23475Son_SchgreF2_S55_L001_counts.txt	Ear	Female
23475Son_SchgreF3_S56_L001	23475Son_SchgreF3_S56_L001_counts.txt	Ear	Female
23475Son_SchgreM1_S51_L001	23475Son_SchgreM1_S51_L001_counts.txt	Ear	Male
23475Son_SchgreM2_S52_L001	23475Son_SchgreM2_S52_L001_counts.txt	Ear	Male
23475Son_SchgreM3_S53_L001	23475Son_SchgreM3_S53_L001_counts.txt	Ear	Male

```

Extra columns can be added to include information that could be useful for statistical analysis, such as rearing condition, time points, replicates, etc.

4. Reference State
Finally, we will need the reference state for which we will perform our differential expression analysis. In the example above, I chose male as the reference state, as the reference genome for *S. gregaria* on NCBI is from a male specimen. 

#### ii) DESeq2 Analysis from STAR Output

We can use the following R script to generate the DEG analysis using DESeq2. The modules necessary to run this script may not be readily available on CRAN or Bioconductor, so be prepared to use devtools or GitHub to install certain packages.

```{r, eval=FALSE}

library("DESeq2")
library("tximport")
library("txdbmaker")
library("knitr")
library("rmdformats")
library("tidyverse")
library("data.table")
library("DT")  # for making interactive search table
library("plotly") # for interactive plots
library("ggthemes") # for theme_calc
library("reshape2")
library("ComplexHeatmap")
library("RColorBrewer")
library("circlize")
library("apeglm")
library("ggpubr")
library("ggplot2")
library("ggrepel")
library("EnhancedVolcano")
library("SARTools")
library("pheatmap")
library("clusterProfiler")
library("sva")
library("cowplot")
library("ashr")
library("ggforce")
library("ggConvexHull")

###############################################################
### USER-DEFINED PARAMETERS (EDIT THESE ONLY)
###############################################################

homeDir <- "/path/to/folder"                # e.g., your home project directory
workDir <- "path/to/working/directory"  
projectName <- "Project Name"
author <- "Author McScienceFace"
species <- "Species name"                   # MUST MATCH STAR RUN SPECIES NAME!
varInt <- "Condition"                       # Experimental factor of interest
condRef <- "Control"                        # Reference condition (baseline)
sampleFile <- "Sample File Name"            # Name of your tab-delimited sample file

###############################################################
### AUTOMATIC PATHS AND INITIALIZATION
###############################################################

# Create working directory structure
dir.create(workDir, showWarnings = FALSE, recursive = TRUE)
setwd(workDir)

rawDir <- file.path(workDir, paste0("03-", species, "-DESeq2"))
dir.create(rawDir, showWarnings = FALSE, recursive = TRUE)

# Create subdirectory for DESeq2 run
Dirname <- paste0("DESeq2_", projectName)
dir.create(Dirname, showWarnings = FALSE)
setwd(Dirname)
workDir_DEseq2 <- getwd()

# Define path to target file
targetFile <- file.path(workDir, paste0(sampleFile))

###############################################################
### DESeq2 PARAMETERS
###############################################################

tresh_logfold <- 1
tresh_padj <- 0.05
alpha_DEseq2 <- 0.05
pAdjustMethod_DEseq2 <- "BH"
featuresToRemove <- NULL
batch <- NULL
fitType <- "parametric"
cooksCutoff <- TRUE
independentFiltering <- TRUE
typeTrans <- "rlog"
locfunc <- "median"
colors <- c("#B31B21", "#1465AC")

###############################################################
### ANALYSIS PIPELINE
###############################################################

message(">>> Starting DESeq2 analysis for project: ", projectName)

# 1. Load target file ----
if (!file.exists(targetFile)) {
  stop("Target file not found: ", targetFile)
}
target <- loadTargetFile(targetFile = targetFile, varInt = varInt, condRef = condRef, batch = batch)

# 2. Load counts ----
counts <- loadCountData(target = target, rawDir = rawDir, featuresToRemove = featuresToRemove)

# 3. Description plots ----
majSequences <- descriptionPlots(counts = counts, group = target[, varInt], col = colors)

# 4. Run DESeq2 ----
out.DESeq2 <- run.DESeq2(
  counts = counts, target = target, varInt = varInt, batch = batch,
  locfunc = locfunc, fitType = fitType, pAdjustMethod = pAdjustMethod_DEseq2,
  cooksCutoff = cooksCutoff, independentFiltering = independentFiltering,
  alpha = alpha_DEseq2
)

# 5. PCA + Clustering ----
exploreCounts(object = out.DESeq2$dds, group = target[, varInt], typeTrans = typeTrans, col = colors)

# 6. Summarize results ----
summaryResults <- summarizeResults.DESeq2(
  out.DESeq2, group = target[, varInt], col = colors,
  independentFiltering = independentFiltering,
  cooksCutoff = cooksCutoff, alpha = alpha_DEseq2
)

# 7. Generate HTML report ----
writeReport.DESeq2(
  target = target, counts = counts, out.DESeq2 = out.DESeq2, summaryResults = summaryResults,
  majSequences = majSequences, workDir = workDir_DEseq2, projectName = projectName, author = author,
  targetFile = targetFile, rawDir = rawDir, featuresToRemove = featuresToRemove, varInt = varInt,
  condRef = condRef, batch = batch, fitType = fitType, cooksCutoff = cooksCutoff,
  independentFiltering = independentFiltering, alpha = alpha_DEseq2, pAdjustMethod = pAdjustMethod_DEseq2,
  typeTrans = typeTrans, locfunc = locfunc, colors = colors
)

message(">>> DESeq2 analysis complete. Results saved to: ", workDir_DEseq2)
###############################################################


```











### Example Results
<div style="width:100%; height:0; padding-bottom:70%; position:relative; border:1px solid #ddd; border-radius:10px; box-shadow:0 0 10px rgba(0,0,0,0.1);">
  <iframe src="Gregaria_Transcriptome_report.html" 
          style="position:absolute; top:0; left:0; width:100%; height:100%; border:none;">
  </iframe>
</div>

### Troubleshooting
#### Lua Filter Skipped
This error occurs when the pandoc version isn't compatible with the default Rmarkdown template system. I fixed this in my run by updating my pandoc to the most recent version using the following steps:

1. Check if pandoc is updated to the most recent version (as of 10/22/25, 3.1.12.3 is the most recent).
```{r, eval=FALSE}

module avail pandoc

```

2. If not the most recent version, then using the following code to install a newer version. Since the pandoc module isn't managed by Sol, we have to update it ourselves. 

```{r, eval=FALSE}

cd ~
wget https://github.com/jgm/pandoc/releases/download/3.1.12.3/pandoc-3.1.12.3-linux-amd64.tar.gz

tar -xvzf pandoc-3.1.12.3-linux-amd64.tar.gz

```

There should now be a folder named "pandoc-3.1.12.3". We can now move this file to our personal bin directory using the following code:

```{r, eval=FALSE}

mkdir -p ~/bin

mv pandoc-3.1.12.3/bin/pandoc ~/bin/

```

Now we can add the new pandoc module to our path so the computer is able to locate the module when we run our R script.

```{r, eval=FALSE}

echo 'export PATH=$HOME/bin:$PATH' >> ~/.bashrc
source ~/.bashrc

```

We can now verify our installation with

```{r, eval=FALSE}

pandoc --version

```

This should match the version you downloaded.
