---
title: "GENESPACE"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
    toc_float: true
    toc_collapsed: false
    theme: default
    css: "theme3.css"
toc_depth: 1
number_sections: true
editor_options:
  chunk_output_type: console
---

## An R package for synteny- and orthology-constrained comparative genomics {.tabset}

### What is GENESPACE?

<div class="img-with-text">
<style>
p {
  text-indent: 2em; /* Adjust the value as needed to simulate a tab */
}
</style>


<div class="text">
  <p><b>
"The development of multiple chromosome-scale reference genome sequences in many taxonomic groups has yielded a high-resolution view of the patterns and processes of molecular evolution. Nonetheless, leveraging information across multiple genomes remains a significant challenge in nearly all eukaryotic systems... Here, we present GENESPACE, which addresses these challenges by integrating conserved gene order and orthology to define the expected physical position of all genes across multiple genomes."</b></p>
  
   -- Lovell <i> et. al.</i>, <i> GENESPACE tracks regions of interest and gene copy number variation across multiple genomes </i>

  
</div>

<img src="assets/riparian-example.png" alt="Riparian Plot Example">
</div>
<style>
.img-with-text {
  display: flex;
  margin-top: 5px;
  align-items: flex-start;
}

.img-with-text img {
  margin-right: 20px;
  margin-top: 10px;
  max-width: 550px;
  height: auto;
}

.img-with-text .text {
  flex: 3 0 400px;
  display: flex;
  flex-direction: column;
}
</style>




### Installation & Running

#### Install in a Conda Environment

These instructions are for installing GENESPACE within a conda environment on the TAMU Grace cluster.

1.Install OrthoFinder by setting
up a conda environment via the procedure described [here](https://hprc.tamu.edu/kb/Software/ANACONDA/)

After you've set up a conda environment, you should be able to install
orthofinder using the following command:

```{r, eval=FALSE}
conda install -c bioconda orthofinder
```

2. Install MCScanX using the following commands in your scratch
directory:

```{r, eval=FALSE}
git clone https://github.com/wyp1125/MCScanX
cd MCScanX
make
```

3. Finally, to install GENESPACE itself, you first need to load R and specify
path for local packages as described [here](https://hprc.tamu.edu/kb/Software/R-tamu/)

Here is an example of how to install R:

```{r, eval=FALSE}
module load GCC/12.2.0
module load OpenMPI/4.1.4
module load R_tamu/4.3.1

mkdir $SCRATCH/R_LIBS_USER/
export R_LIBS=$SCRATCH/R_LIBS_USER/
  ```

4. After you have installed R, run the following command in the conda environment
with orthofinder installed:

  ```{r, eval=FALSE}
R

if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install(c("Biostrings", "rtracklayer"))

if (!requireNamespace("devtools", quietly = TRUE))
    install.packages("devtools")
devtools::install_github("jtlovell/GENESPACE")
```

5. Verify that GENESPACE have been installed by running
```{r, eval=FALSE}
library(GENESPACE)
```
If no error message appears, then GENESPACE has been installed correctly.

#### Run a Test Case

The following commands can be edited afterwards to run other data, this is just the steps to create the example Riparian plot on the GENESPACE GitHub.

The code provided below downloads the files, unzips them, check that they're formatted correctly, and then runs GENESPACE in preparation for your analysis. You can remove the downloading parts of the script if yuo already have the data downloaded, just make sure your directories are set up correctly. If you look under [section 5.2 here](https://htmlpreview.github.io/?https://github.com/jtlovell/tutorials/blob/main/genespaceGuide.html) it will show the file structure you need to follow.

1. Create a file called *genespace-script.R* and copy and paste the following code into it:

```{r genespace-script, eval=FALSE}

#####################  CHANGE THESE TO YOUR DIRECTORIES  ####################### 

genomeRepo <- "~/path/to/store/rawGenomes"
wd <- "~/path/to/genespace/workingDirectory"
path2mcscanx <- "~/path/to/MCScanX/"

################################################################################



####################  CHANGE THESE TO YOUR GENOME URLS  ########################

urls <- c(
  human ="000/001/405/GCF_000001405.40_GRCh38.p14/GCF_000001405.40_GRCh38.p14_",
  mouse = "000/001/635/GCF_000001635.27_GRCm39/GCF_000001635.27_GRCm39_",
  platypus = "004/115/215/GCF_004115215.2_mOrnAna1.pri.v4/GCF_004115215.2_mOrnAna1.pri.v4_",
  chicken = "016/699/485/GCF_016699485.2_bGalGal1.mat.broiler.GRCg7b/GCF_016699485.2_bGalGal1.mat.broiler.GRCg7b_",
  sandLizard = "009/819/535/GCF_009819535.1_rLacAgi1.pri/GCF_009819535.1_rLacAgi1.pri_")

################################################################################




##############################  LEAVE THIS ALONE  ##############################

genomes2run <- names(urls)
urls <- file.path("https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF", urls)
translatedCDS <- sprintf("%stranslated_cds.faa.gz", urls)
geneGff <- sprintf("%sgenomic.gff.gz", urls)

names(translatedCDS) <- genomes2run
names(geneGff) <- genomes2run
writeDirs <- file.path(genomeRepo, genomes2run)
names(writeDirs) <- genomes2run
for(i in genomes2run){
  print(i)
  if(!dir.exists(writeDirs[i]))
    dir.create(writeDirs[i])
  download.file(
    url = geneGff[i], 
    destfile = file.path(writeDirs[i], basename(geneGff[i])))
  download.file(
    url = translatedCDS[i], 
    destfile = file.path(writeDirs[i], basename(translatedCDS[i])))
}
################################################################################



##########################  CHANGE TO YOUR GENOME NAMES  #######################

genomes2run <- c("human", "mouse", "platypus", "chicken", "sandLizard")
parsedPaths <- parse_annotations(
  rawGenomeRepo = genomeRepo,
  genomeDirs = genomes2run,
  genomeIDs = genomes2run,
  presets = "ncbi",
  genespaceWd = wd)

################################################################################



# Check if input data is okay
gpar <- init_genespace(
  wd = wd,
  path2mcscanx = path2mcscanx)


# Run GENESPACE
out <- run_genespace(gpar, overwrite = T)

### INSERT DESIRED FUNCTIONS HERE ###


```

2. You can then run this using the command
```{r, eval=FALSE}
source("genespace-script.R")
```



### Visualization Tools

#### Riparian Plots
<center>
![](assets/riparian-example.png){width=400px}
</center>

Riparian plots are used to visualize synteny between 2 or more species. The command to create these plots in GENESPACE is 

```{r, eval=FALSE}
ripd <- plot_riparian(
  gsParam = out,
  refGenome = "human", 
  useRegions = FALSE)
```

#### Dotplots
<center>
![](assets/dotplots.png){width=600px}
</center>

Dot plots are used to visualize global pattern of synteny between a pair of genomes. The following command produces 3 dot plots: (1) All hits above a score threshold, (2) All hits where the query and target genomes are in the same orthogroup, and (3) Syntenic anchor hits colored by syntenic block ID. The command to do this is

```{r, eval=FALSE}
hits <- read_allBlast(
  filepath = file.path(out$paths$syntenicHits, 
                       "mouse_vs_human.allBlast.txt.gz"))
ggdotplot(hits = hits, type = "all", verbose = FALSE)
```





### Troubleshooting
### Citations 
John T Lovell, Avinash Sreedasyam, M Eric Schranz, Melissa Wilson, Joseph W Carlson, Alex Harkess, David Emms, David M Goodstein, Jeremy Schmutz (2022) *GENESPACE tracks regions of interest and gene copy number variation across multiple genomes* eLife 11:e78526

    

