---
title: "GENESPACE"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
    toc_float: true
    toc_collapsed: false
    theme: default
    css: "theme3.css"
toc_depth: 1
number_sections: true
editor_options:
  chunk_output_type: console
---

## An R package for synteny- and orthology-constrained comparative genomics {.tabset}

### What is GENESPACE?

<div class="img-with-text">
<style>
p {
  text-indent: 2em; /* Adjust the value as needed to simulate a tab */
}
</style>


<div class="text">
  <p><b>
"The development of multiple chromosome-scale reference genome sequences in many taxonomic groups has yielded a high-resolution view of the patterns and processes of molecular evolution. Nonetheless, leveraging information across multiple genomes remains a significant challenge in nearly all eukaryotic systems... Here, we present GENESPACE, which addresses these challenges by integrating conserved gene order and orthology to define the expected physical position of all genes across multiple genomes."</b></p>
  
   -- Lovell <i> et. al.</i>, <i> GENESPACE tracks regions of interest and gene copy number variation across multiple genomes </i>

  
</div>

<img src="assets/riparian-example.png" alt="Riparian Plot Example">
</div>
<style>
.img-with-text {
  display: flex;
  margin-top: 5px;
  align-items: flex-start;
}

.img-with-text img {
  margin-right: 20px;
  margin-top: 10px;
  max-width: 550px;
  height: auto;
}

.img-with-text .text {
  flex: 3 0 400px;
  display: flex;
  flex-direction: column;
}
</style>


### Installation & Running

<mark> This page has been updated to run on the Grace cluster following the March 2025 update to the cluster.</mark>

#### Load packages & Prepare the R environment

These instructions are for installing GENESPACE within a conda environment on the TAMU Grace cluster.

To install GENESPACE itself, you first need to load R and specify
path for local packages as described [here](https://hprc.tamu.edu/kb/Software/R-tamu/)

Here is an example of how to install R:

```{r, eval=FALSE}
ml GCC/12.3.0
ml OpenMPI/4.1.5
ml R_tamu/4.3.2
ml OrthoFinder/2.5.5
ml MCScanX/2024.12.19
export R_LIBS=$SCRATCH/R_LIBS_USER/4.3.2-foss-2023a:$R_LIBS
```

3.  After you have installed R, run the following command in the R terminal:

```{r, eval=FALSE}
R # Launches R terminal 

if (!requireNamespace("devtools", quietly = TRUE))
    install.packages("devtools")
devtools::install_github("jtlovell/GENESPACE")
```

5. Verify that GENESPACE have been installed by running
```{r, eval=FALSE}
library(GENESPACE)
```
If no error message appears, then GENESPACE has been installed correctly. After this step, you may exit the R terminal with the command 

```{r quit-r, eval=FALSE}
q()
```

#### Running with your data

The code provided below downloads the files, unzips them, check that they're formatted correctly, and then runs the entire GENESPACE pipeline, including OrthoFinder and the production of all plots available in the GENESPACE program. You can remove the downloading parts of the script if you already have the data downloaded, just make sure your directories are set up correctly. If you look under [section 5.2 here](https://htmlpreview.github.io/?https://github.com/jtlovell/tutorials/blob/main/genespaceGuide.html) it will show the file structure you need to follow.

1. Create a file called *genespace-script.R* and copy and paste the following code into it:

```{r genespace-script, eval=FALSE}


library(GENESPACE)

genomeRepo <- "Path/to/genome/folder"
wd <- "Path/to/working/directory"
path2mcscanx <- "/sw/eb/sw/MCScanX/2024.12.19-foss-2023a/bin/" #If running on Grace, leave this alone. This will be the path to MCScanX after loading the module.

genomes2run <- c("species_1", "species_2", ...)

####################  CHANGE THESE TO YOUR GENOME URLS  ########################

# This script is meant to automatically download your genomes from the NCBI website. For each species, copy and paste the part of the URL following GCF and then duplication the GCF number for the species. 

#For example, the URL for Schistocerca gregaria is https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/023/897/955/GCF_023897955.1_iqSchGreg1.2/, but to download the information in this script, I would enter 023/897/955/GCF_023897955.1_iqSchGreg1.2/GCF_023897955.1_iqSchGreg1.2/ 

urls <- c(
  species_1 ="URL/formatted/for/species_1",
  species_2 = "URL/formatted/for/species_2", ...
  )

################################################################################



##########################  CHANGE TO YOUR GENOME NAMES  #######################

parsedPaths <- parse_annotations(
  rawGenomeRepo = genomeRepo,
  genomeDirs = genomes2run,
  genomeIDs = genomes2run,
  presets = "ncbi",
  genespaceWd = wd)


# This command prepares the pipeline to run the GENESPACE analysis
gpar <- init_genespace(
    wd = wd,
    path2mcscanx = path2mcscanx,
    rawOrthofinderDir = "/Path/to/OrthoFinder/Run" # This line is optional, it can just direct your GENESPACE run to take data from a previous OrthoFinder run. If you do not want to do this, simply delete this line & GENESPACE will automaitcally run OrthoFinder for you.
    )

out <- run_genespace(gpar) # This command then runs GENESPACE and produces all plots available on the GENESPACE pipeline
```

2. You can then run this using the command
```{r, eval=FALSE}
Rscript genespace-script.R
```



### Visualization Tools

#### Riparian Plots

Riparian plots are used to visualize synteny between 2 or more species. The command to create these plots in GENESPACE is 

```{r, eval=FALSE}
ripd <- plot_riparian(
  gsParam = out,
  refGenome = "human", 
  useRegions = FALSE)
```

This code produces a graph that looks like this: 

<center>
![](assets/riparian-example.png)
</center>

##### Customizing Riparian Plots

We can also generate riparian plots for regions of interest using the following commands:

To plot against the whole genome:

```{r plots-of-interest, eval = FALSE}
roibed <- roi[,c("genome", "chr")]
roibed$color <- c("pink", "cyan", "green")
ripd <- plot_riparian(
  gsParam = out, 
  useRegions = FALSE, 
  highlightBed = roibed)
```

This generates a plot that looks like this:

![](assets/riparian-whole-genome-regions-of-interest.png)

To plot against only the regions of interest:

```{r, eval=FALSE}

ripd <- plot_riparian(
  gsParam = out,  
  useRegions = FALSE,
  highlightBed = roibed, 
  backgroundColor = NULL)

```

This generates a plot that looks like this:

![](assets/riparian-only-regions-of-interest.png)


#### Dotplots


Dot plots are used to visualize global pattern of synteny between a pair of genomes. The following command produces 3 dot plots: (1) All hits above a score threshold, (2) All hits where the query and target genomes are in the same orthogroup, and (3) Syntenic anchor hits colored by syntenic block ID. The command to do this is

```{r, eval=FALSE}
hits <- read_allBlast(
  filepath = file.path(out$paths$syntenicHits, 
                       "mouse_vs_human.allBlast.txt.gz"))
ggdotplot(hits = hits, type = "all", verbose = FALSE)
```

This generates a plot that looks like this:

![](assets/dotplots.png)

<center> Image credits: Lovell et al. </center>



### Troubleshooting
### Citations 
John T Lovell, Avinash Sreedasyam, M Eric Schranz, Melissa Wilson, Joseph W Carlson, Alex Harkess, David Emms, David M Goodstein, Jeremy Schmutz (2022) *GENESPACE tracks regions of interest and gene copy number variation across multiple genomes* eLife 11:e78526

    

