---
title: "GENESPACE"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
    toc_float: true
    toc_collapsed: false
    theme: default
    css: "theme3.css"
toc_depth: 1
number_sections: true
editor_options:
  chunk_output_type: console
---

## An R package for synteny- and orthology-constrained comparative genomics {.tabset}

### What is GENESPACE?

<div class="img-with-text">
<style>
p {
  text-indent: 2em; /* Adjust the value as needed to simulate a tab */
}
</style>


<div class="text">
  <p><b>
"The development of multiple chromosome-scale reference genome sequences in many taxonomic groups has yielded a high-resolution view of the patterns and processes of molecular evolution. Nonetheless, leveraging information across multiple genomes remains a significant challenge in nearly all eukaryotic systems... Here, we present GENESPACE, which addresses these challenges by integrating conserved gene order and orthology to define the expected physical position of all genes across multiple genomes."</b></p>
  
   -- Lovell <i> et. al.</i>, <i> GENESPACE tracks regions of interest and gene copy number variation across multiple genomes </i>

  
</div>

<img src="assets/riparian-example.png" alt="Riparian Plot Example">
</div>
<style>
.img-with-text {
  display: flex;
  margin-top: 5px;
  align-items: flex-start;
}

.img-with-text img {
  margin-right: 20px;
  margin-top: 10px;
  max-width: 550px;
  height: auto;
}

.img-with-text .text {
  flex: 3 0 400px;
  display: flex;
  flex-direction: column;
}
</style>


### Installation & Running

<mark> This page has been updated to run on the Grace cluster following the March 2025 update, as well as the new GENESPACE 1.4 workflow.</mark>

#### Load packages & Prepare the R environment

These instructions are for installing GENESPACE within a conda environment on the TAMU Grace cluster.

**1.**To install GENESPACE itself, you first need to load R and specify
path for local packages as described [here](https://hprc.tamu.edu/kb/Software/R-tamu/)

```{r, eval=FALSE}
ml GCC/12.3.0
ml OpenMPI/4.1.5
ml R_tamu/4.3.2
ml OrthoFinder/2.5.5
ml MCScanX/2024.12.19
export R_LIBS=$SCRATCH/R_LIBS_USER/4.3.2-foss-2023a:$R_LIBS
```

**2.**  After you have installed R, run the following command in the R terminal:

```{r, eval=FALSE}
R # Launches R terminal 

if (!requireNamespace("devtools", quietly = TRUE))
    install.packages("devtools")
devtools::install_github("jtlovell/GENESPACE")
```

**3.** Verify that GENESPACE have been installed by running
```{r, eval=FALSE}
library(GENESPACE)
```
If no error message appears, then GENESPACE has been installed correctly. After this step, you may exit the R terminal with the command 

```{r quit-r, eval=FALSE}
q()
```

<br>
<hr style="border:2px solid lightgray">

#### Running with your data

The code provided below is meant to generate GENESPACE results from a previous OrthoFinder run. Slight modifications can be made to run OrthoFinder within the GENESPACE pipeline. 

**1.** Create the genome repository

GENESPACE expects a directory called the "genomeRepo" with a specific file structure for each species as follows:

```{ascii}
└── genomeRepo
    └── species_gene_id
        ├── species_gene_id_genomic_gff.gz
        └── species_gene_id_translated_cds.faa.gz
```

You must create a folder under the genomeRepo for each species you want to run GENESPACE on and **be sure to name the folder the *same* as the gene id you assigned to the species in OrthoFinder!**


**2.** Create a file called *genespace-script.R* and copy and paste the following code into it:

```{r genespace-script, eval=FALSE}


library(GENESPACE)


library(GENESPACE)

genomeRepo <- "/scratch/group/songlab/emily/comparative-genomics/genespace/mormon-ID/genomeRepo"
wd <- "/scratch/group/songlab/emily/comparative-genomics/SCMB/5_OrthoFinder/fasta/genespace"
path2mcscanx <- "/sw/eb/sw/MCScanX/2024.12.19-foss-2023a/bin/"

##########################  CHANGE TO YOUR GENOME NAMES  ####################### 

genomes2run <- c("Sgreg", "Scanc", "Samer", "Snite", "Spice", "Sscub", "Csecu", "Brsri", "Amel", "Sinvi", "Apisu", "Dmela", "Ansimp")

################################################################################



parsedPaths <- parse_annotations(
  rawGenomeRepo = genomeRepo,
  genomeDirs = genomes2run,
  genomeIDs = genomes2run,
  presets = "ncbi",
  genespaceWd = wd)



gpar <- init_genespace(
    wd = wd,
    path2mcscanx = path2mcscanx,
    rawOrthofinderDir = "/Path/to/OrthoFinder/run") # Note: If you do not have a previous OrthoFinder run you want to run GENESPACE on, 
                                                    # just leave this option and only include the wd and path2mcscanx options and 
                                                    # GENESPACE will automatically run OrthoFinder for you

out <- run_genespace(gpar)

```

**3.** You can then run this using the command
```{r, eval=FALSE}
Rscript genespace-script.R
```

This generates the riparian and dot plots from the GENESPACE pipeline.


### Visualization Tools

#### Riparian Plots

Riparian plots are used to visualize synteny between 2 or more species. The command to create these plots in GENESPACE is 

```{r, eval=FALSE}
ripd <- plot_riparian(
  gsParam = out,
  refGenome = "human", 
  useRegions = FALSE)
```

This code produces a graph that looks like this: 

<center>
![](assets/riparian-example.png)
</center>

##### Customizing Riparian Plots

We can also generate riparian plots for regions of interest using the following commands:

To plot against the whole genome:

```{r plots-of-interest, eval = FALSE}
roibed <- roi[,c("genome", "chr")]
roibed$color <- c("pink", "cyan", "green")
ripd <- plot_riparian(
  gsParam = out, 
  useRegions = FALSE, 
  highlightBed = roibed)
```

This generates a plot that looks like this:

![](assets/riparian-whole-genome-regions-of-interest.png)

To plot against only the regions of interest:

```{r, eval=FALSE}

ripd <- plot_riparian(
  gsParam = out,  
  useRegions = FALSE,
  highlightBed = roibed, 
  backgroundColor = NULL)

```

This generates a plot that looks like this:

![](assets/riparian-only-regions-of-interest.png)


#### Dotplots


Dot plots are used to visualize global pattern of synteny between a pair of genomes. The following command produces 3 dot plots: (1) All hits above a score threshold, (2) All hits where the query and target genomes are in the same orthogroup, and (3) Syntenic anchor hits colored by syntenic block ID. The command to do this is

```{r, eval=FALSE}
hits <- read_allBlast(
  filepath = file.path(out$paths$syntenicHits, 
                       "mouse_vs_human.allBlast.txt.gz"))
ggdotplot(hits = hits, type = "all", verbose = FALSE)
```

This generates a plot that looks like this:

![](assets/dotplots.png)

<center> Image credits: Lovell et al. </center>



### Troubleshooting
### Citations 
John T Lovell, Avinash Sreedasyam, M Eric Schranz, Melissa Wilson, Joseph W Carlson, Alex Harkess, David Emms, David M Goodstein, Jeremy Schmutz (2022) *GENESPACE tracks regions of interest and gene copy number variation across multiple genomes* eLife 11:e78526

    

