---
title: "Pairwise Sequentially Markovian Coalescent (PSMC) Model"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
    toc_float: true
    toc_collapsed: false
    theme: default
    css: "theme3.css"
toc_depth: 1
number_sections: true
editor_options:
  chunk_output_type: console
---

<!-- This code gets rid of the auto-indent -->
<style>
/* Reset margin/padding/indentation for text elements, but NOT code blocks */
body, p, h1, h2, h3, h4, h5, h6, ol, ul, li, blockquote {
  margin-left: 0 !important;
  padding-left: 0 !important;
  text-indent: 0 !important;
}
</style>


### Population Size History Inference {.tabset}

#### What is PSMC?
<center>
![](assets/gregaria_bootstrapped_plot.png){width=70%}
</center>

The Pairwise Sequentially Markovian Coalescent Model (PSMC) is a model that infers a species' population history using HiC data from the species in question. This process is motivated by "Coalescence Theory", which is a statistical theory that relates the size of a population to the coalescence times in the genealogy of sampled individuals (Mather et al.). This is done by mapping HiC reads back to the genome and searching for loci splits via variant calling and bootstrapping results to create a coalescence model. 

[Download HiC Data from NCBI](https://www.ncbi.nlm.nih.gov/sra)

#### Installation & Running

In the drop down below the full script to run the entire pipeline in one slurm script is available. Below this, each step is broken down for informational and troubleshooting purposes. You can run this procedure step-by-step or all in one script, but for the first run I would recommend doing a step-by-step run as it simplifies troubleshooting and familiarizes you with the procedure.

<details>
  <summary><b>Full PSMC Script</b></summary>
```{r, eval=FALSE}
#!/bin/bash

##NECESSARY JOB SPECIFICATIONS
#SBATCH --job-name=PSMC-SPECIES      	  # Set the job name to "PSMC-SPECIES" (change to your species name)
#SBATCH --time=7-00:00:00         			# Set the wall clock limit to 7 dyas
#SBATCH --ntasks=1             	  			# Request 1 task
#SBATCH --cpus-per-task=8         			# Request 8 cpus
#SBATCH --mem=30G                 			# Request 30GB per node
#SBATCH --mail-type=ALL						      # Email updates for all steps
#SBATCH --mail-user=user@@mail.com		  # Email to this address
#SBATCH --output=stdout.%x.%j				    # Name stdout file with run name and job Id
#SBATCH --error=stderr.%x.%j				    # Name stderr file with run name and job Id


########################### STEP 0: CHANGE THESE TO YOUR DATA #########################

SPECIES_NAME_ABBR = "Sgregaria"
SRR_NUM = "SRR17149718"
GENOME_FILE = "GCF_023897955.1_iqSchGreg1.2_genomic.fa"

#######################################################################################



################################ STEP 1: Align with BWA ###############################
module purge 
ml GCC/13.2.0 BWA/0.7.18 SAMtools/1.19.2 

bwa index -p "${SPECIES_NAME_ABBR}" "${GENOME_FILE}"

# assuming you have 8 threads, not that this .sam file will be huge, and we'll remove it at the end
bwa mem -t 8 {SPECIES_NAME_ABBR.} "${SRR_NUM}_1.fastq.gz" "${SRR_NUM}_2.fastq.gz" > ${base}.sam

# convert all .sam to .bam
samtools view -S -b "${SRR_NUM}.sam" > unsorted_alignment.bam

# Merge all .bam files into one
samtools merge unsorted_alignment.bam *.bam

#######################################################################################



################################ STEP 2: Sort .bam file ###############################
module purge
module load picard/2.25.1-Java-11 

java -jar $EBROOTPICARD/picard.jar SortSam \
      INPUT=unsorted_alignment.bam \
      OUTPUT=sorted.bam \
      SORT_ORDER=coordinate

#######################################################################################



################### STEP 3: Generate variant consensus fastq for PSMC #################
module purge
ml GCC/13.2.0 BCFtools/1.19

bcftools mpileup -Q 20 -q 20 -f GCF_023897955.1_iqSchGreg1.2_genomic.fa \
    sorted.bam --threads 8 | \
    bcftools call -c --threads 8 | \
    vcfutils.pl vcf2fq -d 1 -D 100 -Q 20 > variant_consensus.fq

module purge
ml GCC/13.2.0 psmc/0.6.5_20221121 parallel/20240322 gnuplot/6.0.1 texlive/20230313

#######################################################################################



######################### STEP 4: Generate PSMC analysis & plot #######################
# Convert .fq to .psmcfa
fq2psmcfa variant_consensus.fq > psmc_input.psmcfa

# Generate PSMC results
psmc -N25 -t15 -r5 -p "4+25*2+4+6" -o sample.psmc psmc_input.psmcfa

mkdir boot

# Bootstrap results and plot 
parallel 'psmc -N25 -t10 -r5 -b -p "8*1+30*2+4+6" \
	-o $SLURM_SUBMIT_DIR/boot/sample_r{}.psmc $SLURM_SUBMIT_DIR/psmc_input.psmcfa' \
	::: $(seq 50)

cat sample.psmc boot/sample_r*.psmc > sample_combined.psmc

psmc_plot.pl -p -g 2.2 -u 0.00000000506 bootstrapped_plot  sample_combined.psmc


#######################################################################################
  
```
</details> 
        
  
**1.** First, you must map the HiC reads back to the original genome using BWA. Doing this step on the (enormous) *S. gregaria* genome (8.3 Gb) took around 1-2 days. We do this using the following procedure:
  
```{r BWA-align-HiC, eval = FALSE}
#!/bin/bash

##NECESSARY JOB SPECIFICATIONS
#SBATCH --job-name=create-unsorted-alignment        #Set the job name to "psmc-referencegenome"
#SBATCH --time=3-00:00:00         #Set the wall clock limit to 2 hours
#SBATCH --ntasks=1             #Request 1 task
#SBATCH --cpus-per-task=8       #Request 8 cpus
#SBATCH --mem=30G              #Request 30GB per node
#SBATCH --output=stdout.%x.%j
#SBATCH --error=stderr.%x.%j

# Align with BWA
module purge 
ml GCC/13.2.0 BWA/0.7.18 SAMtools/1.19.2 

# bwa index -p Sgregaria GCF_023897955.1_iqSchGreg1.2_genomic.fa
# assuming you have 8 threads, not that this .sam file will be huge, and we'll remove it at the end

bwa mem -t 8 Sgregaria SRR17149718_1.fastq SRR17149718_2.fastq > SRR17149718.sam

# convert .sam to .bam
samtools view -S -b SRR17149718.sam > unsorted_alignment.bam

```

**2.** Sort the unsorted .bam file
```{r, eval=FALSE}

##NECESSARY JOB SPECIFICATIONS
#SBATCH --job-name=sort-alignment        #Set the job name to "psmc-referencegenome"
#SBATCH --time=1-00:00:00         #Set the wall clock limit to 2 hours
#SBATCH --ntasks=1             #Request 1 task
#SBATCH --cpus-per-task=8       #Request 8 cpus
#SBATCH --mem=30G              #Request 30GB per node
#SBATCH --output=stdout.%x.%j
#SBATCH --error=stderr.%x.%j


# sort the .bam file
module purge
module load picard/2.25.1-Java-11 

java -jar $EBROOTPICARD/picard.jar SortSam \
      INPUT=unsorted_alignment.bam \
      OUTPUT=sorted.bam \
      SORT_ORDER=coordinate

```

**3.** Generate a variant consensus file from the sorted .bam file
```{r, eval=FALSE}

#!/bin/bash

##NECESSARY JOB SPECIFICATIONS
#SBATCH --job-name=create-var-con                   # Set the job name to "create-var-con"
#SBATCH --time=2-00:00:00                           # Set the wall clock limit to 2 days
#SBATCH --ntasks=1                                  # Request 1 task
#SBATCH --cpus-per-task=8                           # Request 8 cpus
#SBATCH --mem=30G                                   # Request 30GB per node
#SBATCH --mail-type=ALL						                  # Email updates for all steps
#SBATCH --mail-user=user@mail.edu		                # Email to this address
#SBATCH --output=stdout.%x.%j                       # Name stdout file after job name and ID
#SBATCH --error=stderr.%x.%j                        # Name stderr file after job name and ID


module purge
ml GCC/13.2.0 BCFtools/1.19

# Generate consensus fastq for PSMC
bcftools mpileup -Q 20 -q 20 -f GCF_023897955.1_iqSchGreg1.2_genomic.fa \
    sorted.bam --threads 8 | \
    bcftools call -c --threads 8 | \
    vcfutils.pl vcf2fq -d 1 -D 100 -Q 20 > variant_consensus.fq
```

**4.** Generate PSMC results
```{r, eval=FALSE}

#!/bin/bash

##NECESSARY JOB SPECIFICATIONS
#SBATCH --job-name=psmc-input-and-plot
#SBATCH --time=4-00:00:00
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=8
#SBATCH --mem=120G
#SBATCH --mail-type=ALL
#SBATCH --mail-user=ecbaker7@tamu.edu
#SBATCH --output=stdout.%x.%j
#SBATCH --error=stderr.%x.%j

module purge
ml GCC/13.2.0 psmc/0.6.5_20221121 parallel/20240322 gnuplot/6.0.1 texlive/20230313

# Convert .fq to .psmcfa
fq2psmcfa variant_consensus.fq > psmc_input.psmcfa

# Generate PSMC results
psmc -N25 -t15 -r5 -p "4+25*2+4+6" -o sample.psmc psmc_input.psmcfa

mkdir boot

# Bootstrap results and plot 
parallel 'psmc -N25 -t10 -r5 -b -p "8*1+30*2+4+6" \
	-o $SLURM_SUBMIT_DIR/boot/sample_r{}.psmc $SLURM_SUBMIT_DIR/psmc_input.psmcfa' \
	::: $(seq 50)

cat sample.psmc boot/sample_r*.psmc > sample_combined.psmc

psmc_plot.pl -p -g 2.2 -u 0.00000000506 bootstrapped_plot  sample_combined.psmc
```


##### OPTIONAL: Adding a title to your plot

<center>
<b>[Example of a Titled Plot](orthop-genomes-project.html#psmc-analysis)</b>
</center>

There is no option to add a plot title in the psmc module, so if you want your plot to be named, you'll have to directly edit the gnuplot file. Search for a file in your PSMC dirctory ending in the extension *.gnuplot* or *.gp*. Then type 

```{r, eval=FALSE}
nano YOURFILE.gp
```
into your terminal to edit the gnuplot file. Then, find where the x and y labels are set. The file layout should look something like this:

```{r, eval=FALSE}
  set size 1, 0.8;
  set xran [10000:*];
  set log x;
  set format x "10^{%L}";
  set mxtics 10;
  set mytics 10;
  unset grid;
  set key off;
  set xtics font "Helvetica,16";
  set ytics nomirror font "Helvetica,16";
  set xlab "Years (g=2.2, {/Symbol m}=0.5x10^{-8})" font "Helvetica,16";
  set t po eps enhance so co "Helvetica,16";

  set yran [0:*];
  set ylab "Effective population size (x10^4)" font "Helvetica,16";
  set out "Schistocerca piceifrons.eps";
  set style line 1 lt 1 lc rgb "#FF0000" lw 4;
```

and add the line set title "Your Title" below the set x and y label commands. For example, my file looks like this for *Schistocerca piceifrons*:

```{r, eval=FALSE}
 set size 1, 0.8;
  set xran [10000:*];
  set log x;
  set format x "10^{%L}";
  set mxtics 10;
  set mytics 10;
  unset grid;
  set key off;
  set xtics font "Helvetica,16";
  set ytics nomirror font "Helvetica,16";
  set xlab "Years (g=2.2, {/Symbol m}=0.5x10^{-8})" font "Helvetica,16";
  set t po eps enhance so co "Helvetica,16";

  set yran [0:*];
  set ylab "Effective population size (x10^4)" font "Helvetica,16";

  set title "Schistocerca piceifrons" font "Helvetica-Italic,24";

  set out "Schistocerca piceifrons.eps";
  set style line 1 lt 1 lc rgb "#FF0000" lw 4;

```
#### Citations

Mather, Niklas, et al. “A Practical Introduction to Sequentially Markovian Coalescent Methods for Estimating Demographic History From Genomic Data.” Ecology and Evolution, vol. 10, no. 1, Dec. 2019, pp. 579–89, doi:10.1002/ece3.5888.

Unm-Carc. “QuickBytes/psmc_quickbyte.md at Master · UNM-CARC/QuickBytes.” GitHub, github.com/UNM-CARC/QuickBytes/blob/master/psmc_quickbyte.md.
