---
title: "Repeat Modeler and Masker"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: true
    toc_float: true
    toc_collapsed: false
    theme: default
    css: "theme3.css"
toc_depth: 1
number_sections: true
editor_options:
  chunk_output_type: console
---

## What is RepeatModeler & RepeatMasker?

## Step 0: File & Directory Creation

### Directories

Now we must create the necessary directories for this pipeline to function.

#### Working Directory

Create a new directory in your scratch folder (for example, you could put it in the following file path: scratch/user/YOUR_USERNAME/PROJECT_NAME). Call this new directory *rep-model-mask*. This is where you will add in the files we will create later.

#### Genome Directory

Create a directory in your scratch directory called *genomes*. This is where we will store our data for the repeat modeler and repeat masker runs.

### Files

The main problem with the currently available RepeatModeler and RepeatMasker software is that they create thousands of large "intermediary" files that do not get deleted once the pipeline is finished running. Devon J. Boland from the Texas A&M Institute of Genome Sciences and Society (TIGSS) developed this SnakeMake pipeline that automatically deletes these files, freeing up storage after the process has completed its run. To run this pipeline you will need to create the following 3 files:

#### i) Snakefile
  Create the file "Snakefile" in your rep-model-mask directory and copy and paste the below code into it.<mark> You will need to change lines 8 and 9 to the file paths of the work and genome folders you created in your scratch directory (see Step 0: Directories).</mark>
  
    ### =================================================================
    ### GENOME REPEAT MASKING PIPELINE (v. DECEMBER 2024)
    ### Project: Comparative Genomics Of Locusts
    ### Devon J. Boland, <devonjboland@tamu.edu>
    ### =================================================================
    
    ### SET DIRECTORY PATHS FOR REFERENCE AND OUTPUT DATA
    WORKDir = "/scratch/user/devonjboland/BPRI/Genomes/RepeatMasking"
    GENOMEDir = "/scratch/user/devonjboland/BPRI/Genomes"
    
    # For local testing
    #WORKDir = "/Users/devonjboland/Documents/TIGSS/Maeva_TECHER_SongLab/Repeat_Masking/SnakeMake"
    #GENOMEDir = "/Users/devonjboland/Documents/TIGSS/Maeva_TECHER_SongLab/Repeat_Masking/SnakeMake/Genomes"
    
    ### SET SAMPLES LIST
    INSECTS, = glob_wildcards(GENOMEDir + "/{insect}.fa")
    
    ### SET PATHS FOR STATIC FILES
    TRFSif = "/scratch/user/devonjboland/SIFs/trf_4.09.1--h031d066_4.sif"
    
    # SET PSEUDO-RULE
    
    rule all:
            input:
                    ###   GENERATING TRANSCRIPT MAPPING AND READS COUNTS
                    expand(WORKDir + "/{insect}/{insect}.fa.2.7.7.80.10.50.500.mask.masked", insect = INSECTS)
    
    ### =================================================================
    ###  Build TE Family Database Using RepeatModeler
    ### =================================================================
    
    rule repeat_modeler:
            input:
                    genome = GENOMEDir + "/{insect}.fa"
            threads: 48
            params:
                    genome_loc = GENOMEDir,
                    work_dir = WORKDir + "/{insect}",
                    short_name = "{insect}"
            output:
                    rm_lib = WORKDir + "/{insect}/{insect}-families.fa"
            shell:
                    """
                    set -euxo pipefail
                    ml purge
                    ml GCC/12.3.0  OpenMPI/4.1.5 RepeatModeler/2.0.6
    
                    cp {input.genome} $TMPDIR
                    cd $TMPDIR
    
                    # Create Blastn database of genome
                    BuildDatabase -name {params.short_name} {params.short_name}.fa
    
                    # Execute Repeat Modeler
                    RepeatModeler -database {params.short_name} -threads {threads} -LTRStruct
                    mv {params.short_name}* {params.work_dir}
                    mv {params.short_name}.fa {params.work_dir}
                    cd ${params.work_dir}
                    ml purge
    
                    """
    
    
    ### =================================================================
    ###  Identify Tandem Repeats Using TRF
    ### =================================================================
    
    rule run_trf:
            params:
                    genome = WORKDir + "/{insect}/{insect}.fa",
                    work_dir = WORKDir + "/{insect}",
                    sif = TRFSif
            output:
                    masked_trf = WORKDir + "/{insect}/{insect}.fa.2.7.7.80.10.50.500.mask"
            shell:
                    """
                    ml purge
                    cd {params.work_dir}
                    singularity exec {params.sif} trf {params.genome} 2 7 7 80 10 50 500 -f -d -m -h
    
                    """
    
    ### =================================================================
    ###  Mask TEs Using RepeatMasker on TRF Output with RM Input
    ### =================================================================
    
    rule repeat_masker:
            input:
                    genome_trf = WORKDir + "/{insect}/{insect}.fa.2.7.7.80.10.50.500.mask",
                    rm_lib = WORKDir + "/{insect}/{insect}-families.fa"
            threads: 48
            output:
                    final_mask = WORKDir + "/{insect}/{insect}.fa.2.7.7.80.10.50.500.mask.masked"
            shell:
                    """
                    set -euxo pipefail
                    ml purge
                    ml GCC/11.3.0  OpenMPI/4.1.4 RepeatMasker/4.1.4
    
                    RepeatMasker -lib {input.rm_lib} --pa ${threads} {input.genome_trf}
    
                    ml purge
                    """


#### ii) snakefile.slurm
  Create the file "snakefile.slurm" in your rep-model-mask directory and copy and paste the following code into it. This file specifies that the cluster should send a notification when any updates occurring the job process (job begin or end, next stage, job canceled, etc.) to the specified email. <mark> You will need to edit line 10 to email notifications to the appropriate email address. If you do not wish to receive the email updates, you can delete lines 9 and 10.</mark>
    
    #!/bin/bash
    #SBATCH --job-name=snakemake-master
    #SBATCH --time="7-00:00:00"
    #SBATCH --mem=100GB
    #SBATCH --cpus-per-task=1
    #SBATCH --ntasks=1
    
    ## OPTIONAL ##
    #SBATCH --mail-type=ALL              #Send email on all job events
    #SBATCH --mail-user=USEREMAIL@tamu.edu    #Send all emails
    
    ml GCC/12.2.0  OpenMPI/4.1.4 snakemake/7.32.3
    
    snakemake --latency-wait 1000 \
    --restart-times 1 \
    -j 30 -p \
    --max-jobs-per-second 1 \
    --cluster-config cluster.json \
    --cluster "sbatch -p {cluster.partition} --ntasks {cluster.ntasks} --cpus-per-task {cluster.cpus-per-task} -t {cluster.time} --mem {cluster.mem} --output {cluster.output} --error {cluster.error}" \
    --rerun-incomplete \
    --notemp \
    --nolock


#### iii) cluster.json
  Create a file called "cluster.json" in your rep-model-mask directory and copy and paste the following code into it. This file provides specifications for how the cluster should allocate memory and name the output files based on what task is being run. <mark> You do not need to edit any lines of code in this file prior to submitting the job. </mark>
    
    {
    "__default__" :
    {
        "cpus-per-task" : 12,
        "partition" : "medium",
        "ntasks": 1,
        "mem" : "40G",
        "time": "1-00:00:00",
        "out": "%j.out.txt"
    },
    "repeat_modeler":
    {
        "cpus-per-task" : 48,
        "partition" : "long",
        "ntasks": 1,
        "mem" : "320G",
        "time": "7-00:00:00",
        "output": "repeat_modeler.%j.stdout.txt",
        "error": "repeat_modeler.%j.stderr.txt"
    },
    "run_trf":
    {
        "cpus-per-task" : 48,
        "partition" : "long",
        "ntasks": 1,
        "mem" : "320G",
        "time": "3-00:00:00",
        "output": "run_trf.%j.stdout.txt",
        "error": "run_trf.%j.stderr.txt"
    },
    "repeat_masker":
    {
        "cpus-per-task" : 48,
        "partition" : "long",
        "ntasks": 1,
        "mem" : "320G",
        "time": "7-00:00:00",
        "output": "repeat_masker.%j.stdout.txt",
        "error": "repeat_masker.%j.stderr.txt"
    }
    }




## Step 1: Repeat Modeler

First we will run RepeatModeler on our genome of choice using the following code to submit the job to the cluster

```{r rep-mask-submit, eval=FALSE}

sbatch snakemake.slurm repeat_modeler

```

This job will launch jobs to run in parallel, so don't be alarmed if you see more jobs running than you've submitted.

## Step 2: Repeat Masker

Run RepeatMasker using the command

```{r rep-model-submit, eval=FALSE}

sbatch snakemake.slurm repeat_masker

```
