---
title: "Repeat Modeler and Masker"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: true
    toc_float: true
    toc_collapsed: false
    theme: default
    css: "theme3.css"
toc_depth: 1
number_sections: true
editor_options:
  chunk_output_type: console
---
## Identifying and Processing Repeat Content
<!-- ## What is RepeatModeler & RepeatMasker? -->

## Script

To run both Repeat Modeler and Masker, we used the following script created by Amanda Stahlke at Colorado Mesa University. 

```{r, eval=FALSE}

#!/bin/bash
#SBATCH --job-name=Repeat-SPECIES
#SBATCH --time=7-00:00:00
#SBATCH --nodes=1          # max 64 nodes for partition long
#SBATCH --ntasks-per-node=1
#SBATCH --cpus-per-task=48
#SBATCH --mem=360G
#SBATCH --mail-type=ALL
#SBATCH --mail-user=user@mail.com
#SBATCH --output=stdout.%x.%j
#SBATCH --error=stderr.%x.%j

ASM={Genome File Suffix}_genomic.fa
NM={Genome Name}

module purge
module load GCC/12.3.0 OpenMPI/4.1.5 RepeatModeler/2.0.6

printf "Built repeatmodler2 database on $ASM to build de novo custom library called $NM \n"
BuildDatabase -name $NM $ASM 
printf "Ran repeatmodler2 with LTRStruct on $ASM for  de novo custom library called $NM \n"
RepeatModeler -threads 40 -LTRStruct -database $NM 

module purge
module load GCC/11.3.0 OpenMPI/4.1.4 RepeatMasker/4.1.4

printf "Running RepeatMasker on $ASM"
RepeatMasker -pa 40 -a -gff -lib ${NM}-families.fa -dir denovo2 $ASM

```

Please note the use of the *-a* flag, which is used to generate an alignment file during the masking process. This is important for generating the Kimura plots mentioned later on this page

## Downstream Analysis
### Visualizing Repeat Content
#### For a single genome

![](assets/SchCanc_Kimura.png)

To generate a kimura repeat content plot like the one above, first you must run the script included in the RepeatMasker module called *calcDivergenceFromAlign.pl*. I did this using the script below on the TAMU Grace Cluster

```{r, eval=FALSE}

#!/bin/bash
#SBATCH --job-name=Generate-Dist
#SBATCH --time=03:00:00
#SBATCH --nodes=1          # max 64 nodes for partition long
#SBATCH --ntasks-per-node=1
#SBATCH --cpus-per-task=48
#SBATCH --mem=360G
#SBATCH --mail-type=ALL
#SBATCH --mail-user=user@mail.com
#SBATCH --output=stdout.%x.%j
#SBATCH --error=stderr.%x.%j

module purge
module load GCC/11.3.0 OpenMPI/4.1.4 RepeatMasker/4.1.4

/sw/eb/sw/RepeatMasker/4.1.4-foss-2022a/util/calcDivergenceFromAlign.pl -s {SpeciesName}.distance {SpeciesName}_genomic.fa.align

```

This will generate a file with a *.distance* suffix. I find it easier to just download this file and generate the figure locally, as we will need to edit this file as well. When you open your *species.distance* file you should see something like in the block below:

```{r, eval=FALSE, attr.source='.numberLines'}
Jukes/Cantor and Kimura subsitution levels adjusted for CpG sites
=================================================================
File: GCF_023864275.1_iqSchCanc2.1_genomic.fa.align
Weighted average Kimura divergence for each repeat family
Class	Repeat	absLen	wellCharLen	Kimura%
-----	------	------	-----------	-------
DNA	rnd-4_family-3901	2850838	2828840	9.37
DNA	rnd-5_family-7626	384874	375022	18.86
DNA/Academ-1	rnd-5_family-132	136309	134796	21.75
⋮
LINE	rnd-4_family-267	2208155	2149395	32.22
LINE	rnd-4_family-268	243936	240416	20.67
LINE	rnd-5_family-1066	674334	658113	22.96
⋮
LTR/Copia	rnd-5_family-1055	766978	760738	15.83
LTR/Copia	rnd-5_family-1977	149895	146399	25.27
⋮
RC/Helitron	rnd-4_family-1433	1591130	1508796	17.81
RC/Helitron	rnd-4_family-1977	1352003	1334534	17.66
⋮
Simple_repeat	combined	45799202	45402746	----
Unknown	rnd-1_family-1	141049217	140819533	5.19
⋮
tRNA	rnd-5_family-26180	476868	476755	3.98
tRNA	rnd-5_family-5841	2251189	2236163	10.50


Coverage for each repeat class and divergence (Kimura)
Div DNA DNA/Academ-1 ... LINE LINE/CR1 ... LTR/Copia ... RC/Helitron Simple_repeat Unknown tRNA
0 290 2240 223288 ... 214984 707342 328 ... 6524  711 0 56168370 340641 
⋮
```

But much, much larger. Copy and paste from where it starts with "Div", below the header for the Kimura scores. Create a new file named {SpeciesSuffix}.distance and copy and paste the text to this file. It should look like this:

```{r, eval=FALSE}
Div DNA DNA/Academ-1 ... LINE LINE/CR1 ... LTR/Copia ... RC/Helitron Simple_repeat Unknown tRNA
0 290 2240 223288 ... 214984 707342 328 ... 6524  711 0 56168370 340641 
⋮
```

Now we're ready to create our Kimura plot. Create a new R file called *kimura-plot-single-genome.r* and copy and paste the following code into it:

```{r, eval=FALSE}

# ------------------------------------------------------------------------------------------------ #
# Repeat annotation summary
#
# Simple figure to summarise repeat families within each genome assembly

# ------------------------------------------------------------------------------------------------ #
# Libraries
suppressPackageStartupMessages({
  library(tidyverse)
  library(magrittr)
  library(here)
  library(scales)
  library(ragg)
  library(readr)
})

# ------------------------------------------------------------------------------------------------ #

# ------------------- Edit for your species ------------------------------------------------- #

size = 8500000000 # Total genome size
sample_name = "Schistocerca cancellata"
file <- "song_assembly/SchCanc.distance"

# ------------------------------------------------------------------------------------------------ #

genome_sizes <- tibble(sample = sample_name, size = size)
pal <- colorRampPalette(RColorBrewer::brewer.pal(n = 7, 'Set3')) # Set color pallette
header <- read_lines(file=file, n_max=1)

df.kimura <- read_table(file, col_types = cols(.default = col_double())) # Read in data (1 species)
df.kimura <- df.kimura[, -ncol(df.kimura)] %>% # Remove last column created from whitespace on EOL
  mutate(sample = sample_name) %>% # Add in col for sample name, unnecessary?
  left_join(genome_sizes, by = "sample") %>% # Join df.kimura with genome.sizes by sample
   pivot_longer(cols = -c(sample, size, Div), names_to = "repeat_type", values_to = "value")  %>%# Make the width of the column the height
   separate(col = repeat_type, into = c("Family", "Sub-family"), sep = "/")  %>% # Seperate Family and Sub-family by "/"
   mutate(`Sub-family` = ifelse(is.na(`Sub-family`), Family, `Sub-family`)) %>% # If sub-family empty, put Family name
  mutate(
    percentage = (value/size) * 100 # calculate percentage composition of genome
    ) %>%
    mutate(`Sub-family` = ifelse(is.na(`Sub-family`), Family, `Sub-family`))

tmp <- df.kimura |>
  filter(!is.na(value), !str_detect(Family, '\\?')) |>
  group_by(sample, Div, Family) |>
  mutate(total_family = sum(value), percentage_family = (total_family/size) * 100) |>
  select(sample, Div, Family, percentage_family) |>
  ungroup() |>
  distinct()

tmp_filtered <- tmp |> filter(percentage_family > 0.01)
 # Plotting
kimura.gg <- ggplot(tmp_filtered, aes(x = Div, y = percentage_family, fill = Family)) +
  geom_bar(position = "stack", stat = "identity", alpha = 0.6) +
  scale_fill_manual( # associate colors with family names
    values = c(
      DNA = '#4eaf49',
      LINE = '#984f9f',
      LTR = '#397fb9',
      MITE = '#f7941d',
      Unknown = '#737e76'
    ),
    labels = c(
      DNA = "DNA",
      LINE = "LINEs",
      LTR = "LTRs",
      MITE = "MITEs",
      Unknown = "Unknown Repeats"
    )  ) +
  labs(
    x = '\nKimura substitution level',
    y = 'Genome percentage\n',
    fill = '\n'
  ) +
  scale_y_continuous(labels = scales::label_number(suffix = '%')) +
  scale_x_continuous(
    breaks = seq(0, 50, 5),
    limits = c(0, 55),
    expand = c(0, 0)
  ) +
  coord_cartesian(xlim = c(0, 51)) +
  theme_minimal() +
  theme(
    axis.title = element_text(size = 20, face = 'bold'),
    axis.text = element_text(size = 18),
    legend.position = 'right',
    legend.title = element_text(size = 18),
    legend.text = element_text(size = 18),
    strip.text = element_text(size = 20, face = 'bold.italic')
  ) +
  facet_wrap(~sample, nrow = 4)

# Save plot
pdf(
  file = paste0(sample_name, "-kimura.pdf"),
  width = 12,
  height = 9
)
print(kimura.gg)
invisible(dev.off())
```


#### For multiple genomes (stacked)

![](assets/stacked-kimura.png)

Follow the same procedure for each *.distance* file as described in the single-genome process. The only difference in this procedure is the script we will run to create the Kimura plots. Create a new R script called *kimura-plot-mult-genomes.R* and copy and paste the following script into it:

```{r, eval=FALSE}

COMING SOON

```